// auth0/dist/esm/management/api/resources/logStreams/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { normalizeClientOptionsWithAuth } from "../../../../BaseClient.mjs";
import * as core from "../../../../core/index.mjs";
import { mergeHeaders } from "../../../../core/headers.mjs";
import * as environments from "../../../../environments.mjs";
import { handleNonStatusCodeError } from "../../../../errors/handleNonStatusCodeError.mjs";
import * as errors from "../../../../errors/index.mjs";
import * as Management from "../../../index.mjs";
export class LogStreamsClient {
    constructor(options) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    /**
     * Retrieve details on [log streams](https://auth0.com/docs/logs/streams).
     *
     * **Sample Response**
     *
     * ```json
     * [{
     *   "id": "string",
     *   "name": "string",
     *   "type": "eventbridge",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "awsAccountId": "string",
     *     "awsRegion": "string",
     *     "awsPartnerEventSource": "string"
     *   }
     * }, {
     *   "id": "string",
     *   "name": "string",
     *   "type": "http",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "httpContentFormat": "JSONLINES|JSONARRAY",
     *     "httpContentType": "string",
     *     "httpEndpoint": "string",
     *     "httpAuthorization": "string"
     *   }
     * },
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "eventgrid",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "azureSubscriptionId": "string",
     *     "azureResourceGroup": "string",
     *     "azureRegion": "string",
     *     "azurePartnerTopic": "string"
     *   }
     * },
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "splunk",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "splunkDomain": "string",
     *     "splunkToken": "string",
     *     "splunkPort": "string",
     *     "splunkSecure": "boolean"
     *   }
     * },
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "sumo",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "sumoSourceAddress": "string"
     *   }
     * },
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "datadog",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "datadogRegion": "string",
     *     "datadogApiKey": "string"
     *   }
     * }]
     * ```
     *
     * @param {LogStreamsClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.NotFoundError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.logStreams.list()
     */
    list(requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__list(requestOptions));
    }
    __list(requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, "log-streams"),
                method: "GET",
                headers: _headers,
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 404:
                        throw new Management.NotFoundError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "GET", "/log-streams");
        });
    }
    /**
     * Create a log stream.
     *
     * **Log Stream Types**
     *
     * The `type` of log stream being created determines the properties required in the `sink` payload.
     *
     * **HTTP Stream**
     *
     * For an `http` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "http",
     *   "sink": {
     *     "httpEndpoint": "string",
     *     "httpContentType": "string",
     *     "httpContentFormat": "JSONLINES|JSONARRAY",
     *     "httpAuthorization": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "http",
     *   "status": "active",
     *   "sink": {
     *     "httpEndpoint": "string",
     *     "httpContentType": "string",
     *     "httpContentFormat": "JSONLINES|JSONARRAY",
     *     "httpAuthorization": "string"
     *   }
     * }
     * ```
     *
     * **Amazon EventBridge Stream**
     *
     * For an `eventbridge` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "eventbridge",
     *   "sink": {
     *     "awsRegion": "string",
     *     "awsAccountId": "string"
     *   }
     * }
     * ```
     *
     * The response will include an additional field `awsPartnerEventSource` in the `sink`:
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "eventbridge",
     *   "status": "active",
     *   "sink": {
     *     "awsAccountId": "string",
     *     "awsRegion": "string",
     *     "awsPartnerEventSource": "string"
     *   }
     * }
     * ```
     *
     * **Azure Event Grid Stream**
     *
     * For an `Azure Event Grid` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "eventgrid",
     *   "sink": {
     *     "azureSubscriptionId": "string",
     *     "azureResourceGroup": "string",
     *     "azureRegion": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "http",
     *   "status": "active",
     *   "sink": {
     *     "azureSubscriptionId": "string",
     *     "azureResourceGroup": "string",
     *     "azureRegion": "string",
     *     "azurePartnerTopic": "string"
     *   }
     * }
     * ```
     *
     * **Datadog Stream**
     *
     * For a `Datadog` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "datadog",
     *   "sink": {
     *     "datadogRegion": "string",
     *     "datadogApiKey": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "datadog",
     *   "status": "active",
     *   "sink": {
     *     "datadogRegion": "string",
     *     "datadogApiKey": "string"
     *   }
     * }
     * ```
     *
     * **Splunk Stream**
     *
     * For a `Splunk` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "splunk",
     *   "sink": {
     *     "splunkDomain": "string",
     *     "splunkToken": "string",
     *     "splunkPort": "string",
     *     "splunkSecure": "boolean"
     *   }
     * }
     * ```
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "splunk",
     *   "status": "active",
     *   "sink": {
     *     "splunkDomain": "string",
     *     "splunkToken": "string",
     *     "splunkPort": "string",
     *     "splunkSecure": "boolean"
     *   }
     * }
     * ```
     *
     * **Sumo Logic Stream**
     *
     * For a `Sumo Logic` Stream, the `sink` properties are listed in the payload below.
     *
     * **Request:**
     * ```json
     * {
     *   "name": "string",
     *   "type": "sumo",
     *   "sink": {
     *     "sumoSourceAddress": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "sumo",
     *   "status": "active",
     *   "sink": {
     *     "sumoSourceAddress": "string"
     *   }
     * }
     * ```
     *
     * @param {Management.CreateLogStreamRequestContent} request
     * @param {LogStreamsClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.ConflictError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.logStreams.create({
     *         type: "http",
     *         sink: {
     *             httpEndpoint: "httpEndpoint"
     *         }
     *     })
     */
    create(request, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__create(request, requestOptions));
    }
    __create(request, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, "log-streams"),
                method: "POST",
                headers: _headers,
                contentType: "application/json",
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                requestType: "json",
                body: request,
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return {
                    data: _response.body,
                    rawResponse: _response.rawResponse,
                };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 409:
                        throw new Management.ConflictError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "POST", "/log-streams");
        });
    }
    /**
     * Retrieve a log stream configuration and status.
     *
     * **Sample responses**
     *
     * **Amazon EventBridge Log Stream**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "eventbridge",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "awsAccountId": "string",
     *     "awsRegion": "string",
     *     "awsPartnerEventSource": "string"
     *   }
     * }
     * ```
     *
     * **HTTP Log Stream**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "http",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "httpContentFormat": "JSONLINES|JSONARRAY",
     *     "httpContentType": "string",
     *     "httpEndpoint": "string",
     *     "httpAuthorization": "string"
     *   }
     * }
     * ```
     *
     * **Datadog Log Stream**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "datadog",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "datadogRegion": "string",
     *     "datadogApiKey": "string"
     *   }
     * }
     * ```
     *
     * **Mixpanel**
     *
     * **Request:**
     *
     * ```json
     * {
     *   "name": "string",
     *   "type": "mixpanel",
     *   "sink": {
     *     "mixpanelRegion": "string",
     *     "mixpanelProjectId": "string",
     *     "mixpanelServiceAccountUsername": "string",
     *     "mixpanelServiceAccountPassword": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "mixpanel",
     *   "status": "active",
     *   "sink": {
     *     "mixpanelRegion": "string",
     *     "mixpanelProjectId": "string",
     *     "mixpanelServiceAccountUsername": "string",
     *     "mixpanelServiceAccountPassword": "string"
     *   }
     * }
     * ```
     *
     * **Segment**
     *
     * **Request:**
     *
     * ```json
     * {
     *   "name": "string",
     *   "type": "segment",
     *   "sink": {
     *     "segmentWriteKey": "string"
     *   }
     * }
     * ```
     *
     * **Response:**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "segment",
     *   "status": "active",
     *   "sink": {
     *     "segmentWriteKey": "string"
     *   }
     * }
     * ```
     *
     * **Splunk Log Stream**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "splunk",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "splunkDomain": "string",
     *     "splunkToken": "string",
     *     "splunkPort": "string",
     *     "splunkSecure": "boolean"
     *   }
     * }
     * ```
     *
     * **Sumo Logic Log Stream**
     *
     * ```json
     * {
     *   "id": "string",
     *   "name": "string",
     *   "type": "sumo",
     *   "status": "active|paused|suspended",
     *   "sink": {
     *     "sumoSourceAddress": "string"
     *   }
     * }
     * ```
     *
     * **Status**
     *
     * The `status` of a log stream maybe any of the following:
     *
     * 1. `active` - Stream is currently enabled.
     * 2. `paused` - Stream is currently user disabled and will not attempt log delivery.
     * 3. `suspended` - Stream is currently disabled because of errors and will not attempt log delivery.
     *
     * @param {string} id - The id of the log stream to get
     * @param {LogStreamsClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.NotFoundError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.logStreams.get("id")
     */
    get(id, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__get(id, requestOptions));
    }
    __get(id, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, `log-streams/${core.url.encodePathParam(id)}`),
                method: "GET",
                headers: _headers,
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return {
                    data: _response.body,
                    rawResponse: _response.rawResponse,
                };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 404:
                        throw new Management.NotFoundError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "GET", "/log-streams/{id}");
        });
    }
    /**
     * Delete a log stream.
     *
     * @param {string} id - The id of the log stream to delete
     * @param {LogStreamsClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.NotFoundError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.logStreams.delete("id")
     */
    delete(id, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__delete(id, requestOptions));
    }
    __delete(id, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, `log-streams/${core.url.encodePathParam(id)}`),
                method: "DELETE",
                headers: _headers,
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return { data: undefined, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 404:
                        throw new Management.NotFoundError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "DELETE", "/log-streams/{id}");
        });
    }
    /**
     * Update a log stream.
     *
     * **Examples of how to use the PATCH endpoint.**
     *
     * The following fields may be updated in a PATCH operation:
     *
     * - name
     * - status
     * - sink
     *
     * Note: For log streams of type `eventbridge` and `eventgrid`, updating the `sink` is not permitted.
     *
     * **Update the status of a log stream**
     *
     * ```json
     * {
     *   "status": "active|paused"
     * }
     * ```
     *
     * **Update the name of a log stream**
     *
     * ```json
     * {
     *   "name": "string"
     * }
     * ```
     *
     * **Update the sink properties of a stream of type `http`**
     *
     * ```json
     * {
     *   "sink": {
     *     "httpEndpoint": "string",
     *     "httpContentType": "string",
     *     "httpContentFormat": "JSONARRAY|JSONLINES",
     *     "httpAuthorization": "string"
     *   }
     * }
     * ```
     *
     * **Update the sink properties of a stream of type `datadog`**
     *
     * ```json
     * {
     *   "sink": {
     *     "datadogRegion": "string",
     *     "datadogApiKey": "string"
     *   }
     * }
     * ```
     *
     * **Update the sink properties of a stream of type `splunk`**
     *
     * ```json
     * {
     *   "sink": {
     *     "splunkDomain": "string",
     *     "splunkToken": "string",
     *     "splunkPort": "string",
     *     "splunkSecure": "boolean"
     *   }
     * }
     * ```
     *
     * **Update the sink properties of a stream of type `sumo`**
     *
     * ```json
     * {
     *   "sink": {
     *     "sumoSourceAddress": "string"
     *   }
     * }
     * ```
     *
     * @param {string} id - The id of the log stream to get
     * @param {Management.UpdateLogStreamRequestContent} request
     * @param {LogStreamsClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.logStreams.update("id")
     */
    update(id, request = {}, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__update(id, request, requestOptions));
    }
    __update(id_1) {
        return __awaiter(this, arguments, void 0, function* (id, request = {}, requestOptions) {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, `log-streams/${core.url.encodePathParam(id)}`),
                method: "PATCH",
                headers: _headers,
                contentType: "application/json",
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                requestType: "json",
                body: request,
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return {
                    data: _response.body,
                    rawResponse: _response.rawResponse,
                };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "PATCH", "/log-streams/{id}");
        });
    }
}
