// auth0/dist/esm/management/api/types/types.mjs
// This file was auto-generated by Fern from our API Definition.
export const OauthScope = {
    /**
     * Create Actions */
    CreateActions: "create:actions",
    /**
     * Read Actions */
    ReadActions: "read:actions",
    /**
     * Update Actions */
    UpdateActions: "update:actions",
    /**
     * Delete Actions */
    DeleteActions: "delete:actions",
    /**
     * Read Anomaly Blocks */
    ReadAnomalyBlocks: "read:anomaly_blocks",
    /**
     * Delete Anomaly Blocks */
    DeleteAnomalyBlocks: "delete:anomaly_blocks",
    /**
     * Read Attack Protection */
    ReadAttackProtection: "read:attack_protection",
    /**
     * Update Attack Protection */
    UpdateAttackProtection: "update:attack_protection",
    /**
     * Create Authentication Methods */
    CreateAuthenticationMethods: "create:authentication_methods",
    /**
     * Read Authentication Methods */
    ReadAuthenticationMethods: "read:authentication_methods",
    /**
     * Update Authentication Methods */
    UpdateAuthenticationMethods: "update:authentication_methods",
    /**
     * Delete Authentication Methods */
    DeleteAuthenticationMethods: "delete:authentication_methods",
    /**
     * Read Branding */
    ReadBranding: "read:branding",
    /**
     * Update Branding */
    UpdateBranding: "update:branding",
    /**
     * Delete Branding */
    DeleteBranding: "delete:branding",
    /**
     * Create Client Credentials */
    CreateClientCredentials: "create:client_credentials",
    /**
     * Read Client Credentials */
    ReadClientCredentials: "read:client_credentials",
    /**
     * Update Client Credentials */
    UpdateClientCredentials: "update:client_credentials",
    /**
     * Delete Client Credentials */
    DeleteClientCredentials: "delete:client_credentials",
    /**
     * Create Client Grants */
    CreateClientGrants: "create:client_grants",
    /**
     * Read Client Grants */
    ReadClientGrants: "read:client_grants",
    /**
     * Update Client Grants */
    UpdateClientGrants: "update:client_grants",
    /**
     * Delete Client Grants */
    DeleteClientGrants: "delete:client_grants",
    /**
     * Read Client Keys */
    ReadClientKeys: "read:client_keys",
    /**
     * Update Client Keys */
    UpdateClientKeys: "update:client_keys",
    /**
     * Read Client Summary */
    ReadClientSummary: "read:client_summary",
    /**
     * Update Client Token Vault Privileged Access */
    UpdateClientTokenVaultPrivilegedAccess: "update:client_token_vault_privileged_access",
    /**
     * Create Clients */
    CreateClients: "create:clients",
    /**
     * Read Clients */
    ReadClients: "read:clients",
    /**
     * Update Clients */
    UpdateClients: "update:clients",
    /**
     * Delete Clients */
    DeleteClients: "delete:clients",
    /**
     * Create Connection Profiles */
    CreateConnectionProfiles: "create:connection_profiles",
    /**
     * Read Connection Profiles */
    ReadConnectionProfiles: "read:connection_profiles",
    /**
     * Update Connection Profiles */
    UpdateConnectionProfiles: "update:connection_profiles",
    /**
     * Delete Connection Profiles */
    DeleteConnectionProfiles: "delete:connection_profiles",
    /**
     * Create Connections */
    CreateConnections: "create:connections",
    /**
     * Read Connections */
    ReadConnections: "read:connections",
    /**
     * Update Connections */
    UpdateConnections: "update:connections",
    /**
     * Delete Connections */
    DeleteConnections: "delete:connections",
    /**
     * Create Connections Keys */
    CreateConnectionsKeys: "create:connections_keys",
    /**
     * Read Connections Keys */
    ReadConnectionsKeys: "read:connections_keys",
    /**
     * Update Connections Keys */
    UpdateConnectionsKeys: "update:connections_keys",
    /**
     * Read Current User */
    ReadCurrentUser: "read:current_user",
    /**
     * Delete Current User */
    DeleteCurrentUser: "delete:current_user",
    /**
     * Create Current User Device Credentials */
    CreateCurrentUserDeviceCredentials: "create:current_user_device_credentials",
    /**
     * Delete Current User Device Credentials */
    DeleteCurrentUserDeviceCredentials: "delete:current_user_device_credentials",
    /**
     * Update Current User Identities */
    UpdateCurrentUserIdentities: "update:current_user_identities",
    /**
     * Update Current User Metadata */
    UpdateCurrentUserMetadata: "update:current_user_metadata",
    /**
     * Create Custom Domains */
    CreateCustomDomains: "create:custom_domains",
    /**
     * Read Custom Domains */
    ReadCustomDomains: "read:custom_domains",
    /**
     * Update Custom Domains */
    UpdateCustomDomains: "update:custom_domains",
    /**
     * Delete Custom Domains */
    DeleteCustomDomains: "delete:custom_domains",
    /**
     * Create Custom Signing Keys */
    CreateCustomSigningKeys: "create:custom_signing_keys",
    /**
     * Read Custom Signing Keys */
    ReadCustomSigningKeys: "read:custom_signing_keys",
    /**
     * Update Custom Signing Keys */
    UpdateCustomSigningKeys: "update:custom_signing_keys",
    /**
     * Delete Custom Signing Keys */
    DeleteCustomSigningKeys: "delete:custom_signing_keys",
    /**
     * Read Device Credentials */
    ReadDeviceCredentials: "read:device_credentials",
    /**
     * Delete Device Credentials */
    DeleteDeviceCredentials: "delete:device_credentials",
    /**
     * Create Directory Provisionings */
    CreateDirectoryProvisionings: "create:directory_provisionings",
    /**
     * Read Directory Provisionings */
    ReadDirectoryProvisionings: "read:directory_provisionings",
    /**
     * Update Directory Provisionings */
    UpdateDirectoryProvisionings: "update:directory_provisionings",
    /**
     * Delete Directory Provisionings */
    DeleteDirectoryProvisionings: "delete:directory_provisionings",
    /**
     * Create Email Provider */
    CreateEmailProvider: "create:email_provider",
    /**
     * Read Email Provider */
    ReadEmailProvider: "read:email_provider",
    /**
     * Update Email Provider */
    UpdateEmailProvider: "update:email_provider",
    /**
     * Delete Email Provider */
    DeleteEmailProvider: "delete:email_provider",
    /**
     * Create Email Templates */
    CreateEmailTemplates: "create:email_templates",
    /**
     * Read Email Templates */
    ReadEmailTemplates: "read:email_templates",
    /**
     * Update Email Templates */
    UpdateEmailTemplates: "update:email_templates",
    /**
     * Create Encryption Keys */
    CreateEncryptionKeys: "create:encryption_keys",
    /**
     * Read Encryption Keys */
    ReadEncryptionKeys: "read:encryption_keys",
    /**
     * Update Encryption Keys */
    UpdateEncryptionKeys: "update:encryption_keys",
    /**
     * Delete Encryption Keys */
    DeleteEncryptionKeys: "delete:encryption_keys",
    /**
     * Read Event Deliveries */
    ReadEventDeliveries: "read:event_deliveries",
    /**
     * Update Event Deliveries */
    UpdateEventDeliveries: "update:event_deliveries",
    /**
     * Create Event Streams */
    CreateEventStreams: "create:event_streams",
    /**
     * Read Event Streams */
    ReadEventStreams: "read:event_streams",
    /**
     * Update Event Streams */
    UpdateEventStreams: "update:event_streams",
    /**
     * Delete Event Streams */
    DeleteEventStreams: "delete:event_streams",
    /**
     * Read Events */
    ReadEvents: "read:events",
    /**
     * Read Federated Connections Tokens */
    ReadFederatedConnectionsTokens: "read:federated_connections_tokens",
    /**
     * Delete Federated Connections Tokens */
    DeleteFederatedConnectionsTokens: "delete:federated_connections_tokens",
    /**
     * Create Flows */
    CreateFlows: "create:flows",
    /**
     * Read Flows */
    ReadFlows: "read:flows",
    /**
     * Update Flows */
    UpdateFlows: "update:flows",
    /**
     * Delete Flows */
    DeleteFlows: "delete:flows",
    /**
     * Read Flows Executions */
    ReadFlowsExecutions: "read:flows_executions",
    /**
     * Delete Flows Executions */
    DeleteFlowsExecutions: "delete:flows_executions",
    /**
     * Create Flows Vault Connections */
    CreateFlowsVaultConnections: "create:flows_vault_connections",
    /**
     * Read Flows Vault Connections */
    ReadFlowsVaultConnections: "read:flows_vault_connections",
    /**
     * Update Flows Vault Connections */
    UpdateFlowsVaultConnections: "update:flows_vault_connections",
    /**
     * Delete Flows Vault Connections */
    DeleteFlowsVaultConnections: "delete:flows_vault_connections",
    /**
     * Create Forms */
    CreateForms: "create:forms",
    /**
     * Read Forms */
    ReadForms: "read:forms",
    /**
     * Update Forms */
    UpdateForms: "update:forms",
    /**
     * Delete Forms */
    DeleteForms: "delete:forms",
    /**
     * Read Grants */
    ReadGrants: "read:grants",
    /**
     * Delete Grants */
    DeleteGrants: "delete:grants",
    /**
     * Read Group Members */
    ReadGroupMembers: "read:group_members",
    /**
     * Read Groups */
    ReadGroups: "read:groups",
    /**
     * Delete Groups */
    DeleteGroups: "delete:groups",
    /**
     * Create Guardian Enrollment Tickets */
    CreateGuardianEnrollmentTickets: "create:guardian_enrollment_tickets",
    /**
     * Read Guardian Enrollments */
    ReadGuardianEnrollments: "read:guardian_enrollments",
    /**
     * Delete Guardian Enrollments */
    DeleteGuardianEnrollments: "delete:guardian_enrollments",
    /**
     * Read Guardian Factors */
    ReadGuardianFactors: "read:guardian_factors",
    /**
     * Update Guardian Factors */
    UpdateGuardianFactors: "update:guardian_factors",
    /**
     * Create Hooks */
    CreateHooks: "create:hooks",
    /**
     * Read Hooks */
    ReadHooks: "read:hooks",
    /**
     * Update Hooks */
    UpdateHooks: "update:hooks",
    /**
     * Delete Hooks */
    DeleteHooks: "delete:hooks",
    /**
     * Create Log Streams */
    CreateLogStreams: "create:log_streams",
    /**
     * Read Log Streams */
    ReadLogStreams: "read:log_streams",
    /**
     * Update Log Streams */
    UpdateLogStreams: "update:log_streams",
    /**
     * Delete Log Streams */
    DeleteLogStreams: "delete:log_streams",
    /**
     * Read Logs */
    ReadLogs: "read:logs",
    /**
     * Read Logs Users */
    ReadLogsUsers: "read:logs_users",
    /**
     * Read Mfa Policies */
    ReadMfaPolicies: "read:mfa_policies",
    /**
     * Update Mfa Policies */
    UpdateMfaPolicies: "update:mfa_policies",
    /**
     * Create Network Acls */
    CreateNetworkAcls: "create:network_acls",
    /**
     * Read Network Acls */
    ReadNetworkAcls: "read:network_acls",
    /**
     * Update Network Acls */
    UpdateNetworkAcls: "update:network_acls",
    /**
     * Delete Network Acls */
    DeleteNetworkAcls: "delete:network_acls",
    /**
     * Create Organization Client Grants */
    CreateOrganizationClientGrants: "create:organization_client_grants",
    /**
     * Read Organization Client Grants */
    ReadOrganizationClientGrants: "read:organization_client_grants",
    /**
     * Delete Organization Client Grants */
    DeleteOrganizationClientGrants: "delete:organization_client_grants",
    /**
     * Create Organization Connections */
    CreateOrganizationConnections: "create:organization_connections",
    /**
     * Read Organization Connections */
    ReadOrganizationConnections: "read:organization_connections",
    /**
     * Update Organization Connections */
    UpdateOrganizationConnections: "update:organization_connections",
    /**
     * Delete Organization Connections */
    DeleteOrganizationConnections: "delete:organization_connections",
    /**
     * Create Organization Discovery Domains */
    CreateOrganizationDiscoveryDomains: "create:organization_discovery_domains",
    /**
     * Read Organization Discovery Domains */
    ReadOrganizationDiscoveryDomains: "read:organization_discovery_domains",
    /**
     * Update Organization Discovery Domains */
    UpdateOrganizationDiscoveryDomains: "update:organization_discovery_domains",
    /**
     * Delete Organization Discovery Domains */
    DeleteOrganizationDiscoveryDomains: "delete:organization_discovery_domains",
    /**
     * Create Organization Invitations */
    CreateOrganizationInvitations: "create:organization_invitations",
    /**
     * Read Organization Invitations */
    ReadOrganizationInvitations: "read:organization_invitations",
    /**
     * Delete Organization Invitations */
    DeleteOrganizationInvitations: "delete:organization_invitations",
    /**
     * Create Organization Member Roles */
    CreateOrganizationMemberRoles: "create:organization_member_roles",
    /**
     * Read Organization Member Roles */
    ReadOrganizationMemberRoles: "read:organization_member_roles",
    /**
     * Delete Organization Member Roles */
    DeleteOrganizationMemberRoles: "delete:organization_member_roles",
    /**
     * Create Organization Members */
    CreateOrganizationMembers: "create:organization_members",
    /**
     * Read Organization Members */
    ReadOrganizationMembers: "read:organization_members",
    /**
     * Delete Organization Members */
    DeleteOrganizationMembers: "delete:organization_members",
    /**
     * Create Organizations */
    CreateOrganizations: "create:organizations",
    /**
     * Read Organizations */
    ReadOrganizations: "read:organizations",
    /**
     * Update Organizations */
    UpdateOrganizations: "update:organizations",
    /**
     * Delete Organizations */
    DeleteOrganizations: "delete:organizations",
    /**
     * Read Organizations Summary */
    ReadOrganizationsSummary: "read:organizations_summary",
    /**
     * Create Phone Providers */
    CreatePhoneProviders: "create:phone_providers",
    /**
     * Read Phone Providers */
    ReadPhoneProviders: "read:phone_providers",
    /**
     * Update Phone Providers */
    UpdatePhoneProviders: "update:phone_providers",
    /**
     * Delete Phone Providers */
    DeletePhoneProviders: "delete:phone_providers",
    /**
     * Create Phone Templates */
    CreatePhoneTemplates: "create:phone_templates",
    /**
     * Read Phone Templates */
    ReadPhoneTemplates: "read:phone_templates",
    /**
     * Update Phone Templates */
    UpdatePhoneTemplates: "update:phone_templates",
    /**
     * Delete Phone Templates */
    DeletePhoneTemplates: "delete:phone_templates",
    /**
     * Read Prompts */
    ReadPrompts: "read:prompts",
    /**
     * Update Prompts */
    UpdatePrompts: "update:prompts",
    /**
     * Read Refresh Tokens */
    ReadRefreshTokens: "read:refresh_tokens",
    /**
     * Update Refresh Tokens */
    UpdateRefreshTokens: "update:refresh_tokens",
    /**
     * Delete Refresh Tokens */
    DeleteRefreshTokens: "delete:refresh_tokens",
    /**
     * Create Resource Servers */
    CreateResourceServers: "create:resource_servers",
    /**
     * Read Resource Servers */
    ReadResourceServers: "read:resource_servers",
    /**
     * Update Resource Servers */
    UpdateResourceServers: "update:resource_servers",
    /**
     * Delete Resource Servers */
    DeleteResourceServers: "delete:resource_servers",
    /**
     * Create Role Members */
    CreateRoleMembers: "create:role_members",
    /**
     * Read Role Members */
    ReadRoleMembers: "read:role_members",
    /**
     * Delete Role Members */
    DeleteRoleMembers: "delete:role_members",
    /**
     * Create Roles */
    CreateRoles: "create:roles",
    /**
     * Read Roles */
    ReadRoles: "read:roles",
    /**
     * Update Roles */
    UpdateRoles: "update:roles",
    /**
     * Delete Roles */
    DeleteRoles: "delete:roles",
    /**
     * Create Rules */
    CreateRules: "create:rules",
    /**
     * Read Rules */
    ReadRules: "read:rules",
    /**
     * Update Rules */
    UpdateRules: "update:rules",
    /**
     * Delete Rules */
    DeleteRules: "delete:rules",
    /**
     * Read Rules Configs */
    ReadRulesConfigs: "read:rules_configs",
    /**
     * Update Rules Configs */
    UpdateRulesConfigs: "update:rules_configs",
    /**
     * Delete Rules Configs */
    DeleteRulesConfigs: "delete:rules_configs",
    /**
     * Create Scim Config */
    CreateScimConfig: "create:scim_config",
    /**
     * Read Scim Config */
    ReadScimConfig: "read:scim_config",
    /**
     * Update Scim Config */
    UpdateScimConfig: "update:scim_config",
    /**
     * Delete Scim Config */
    DeleteScimConfig: "delete:scim_config",
    /**
     * Create Scim Token */
    CreateScimToken: "create:scim_token",
    /**
     * Read Scim Token */
    ReadScimToken: "read:scim_token",
    /**
     * Delete Scim Token */
    DeleteScimToken: "delete:scim_token",
    /**
     * Read Self Service Profile Custom Texts */
    ReadSelfServiceProfileCustomTexts: "read:self_service_profile_custom_texts",
    /**
     * Update Self Service Profile Custom Texts */
    UpdateSelfServiceProfileCustomTexts: "update:self_service_profile_custom_texts",
    /**
     * Create Self Service Profiles */
    CreateSelfServiceProfiles: "create:self_service_profiles",
    /**
     * Read Self Service Profiles */
    ReadSelfServiceProfiles: "read:self_service_profiles",
    /**
     * Update Self Service Profiles */
    UpdateSelfServiceProfiles: "update:self_service_profiles",
    /**
     * Delete Self Service Profiles */
    DeleteSelfServiceProfiles: "delete:self_service_profiles",
    /**
     * Read Sessions */
    ReadSessions: "read:sessions",
    /**
     * Update Sessions */
    UpdateSessions: "update:sessions",
    /**
     * Delete Sessions */
    DeleteSessions: "delete:sessions",
    /**
     * Create Signing Keys */
    CreateSigningKeys: "create:signing_keys",
    /**
     * Read Signing Keys */
    ReadSigningKeys: "read:signing_keys",
    /**
     * Update Signing Keys */
    UpdateSigningKeys: "update:signing_keys",
    /**
     * Create Sso Access Tickets */
    CreateSsoAccessTickets: "create:sso_access_tickets",
    /**
     * Delete Sso Access Tickets */
    DeleteSsoAccessTickets: "delete:sso_access_tickets",
    /**
     * Read Stats */
    ReadStats: "read:stats",
    /**
     * Read Tenant Settings */
    ReadTenantSettings: "read:tenant_settings",
    /**
     * Update Tenant Settings */
    UpdateTenantSettings: "update:tenant_settings",
    /**
     * Create Token Exchange Profiles */
    CreateTokenExchangeProfiles: "create:token_exchange_profiles",
    /**
     * Read Token Exchange Profiles */
    ReadTokenExchangeProfiles: "read:token_exchange_profiles",
    /**
     * Update Token Exchange Profiles */
    UpdateTokenExchangeProfiles: "update:token_exchange_profiles",
    /**
     * Delete Token Exchange Profiles */
    DeleteTokenExchangeProfiles: "delete:token_exchange_profiles",
    /**
     * Create User Attribute Profiles */
    CreateUserAttributeProfiles: "create:user_attribute_profiles",
    /**
     * Read User Attribute Profiles */
    ReadUserAttributeProfiles: "read:user_attribute_profiles",
    /**
     * Update User Attribute Profiles */
    UpdateUserAttributeProfiles: "update:user_attribute_profiles",
    /**
     * Delete User Attribute Profiles */
    DeleteUserAttributeProfiles: "delete:user_attribute_profiles",
    /**
     * Read User Idp Tokens */
    ReadUserIdpTokens: "read:user_idp_tokens",
    /**
     * Create User Tickets */
    CreateUserTickets: "create:user_tickets",
    /**
     * Create Users */
    CreateUsers: "create:users",
    /**
     * Read Users */
    ReadUsers: "read:users",
    /**
     * Update Users */
    UpdateUsers: "update:users",
    /**
     * Delete Users */
    DeleteUsers: "delete:users",
    /**
     * Update Users App Metadata */
    UpdateUsersAppMetadata: "update:users_app_metadata",
    /**
     * Create Vdcs Templates */
    CreateVdcsTemplates: "create:vdcs_templates",
    /**
     * Read Vdcs Templates */
    ReadVdcsTemplates: "read:vdcs_templates",
    /**
     * Update Vdcs Templates */
    UpdateVdcsTemplates: "update:vdcs_templates",
    /**
     * Delete Vdcs Templates */
    DeleteVdcsTemplates: "delete:vdcs_templates",
};
/** How the action is being referred to: `action_id` or `action_name`. */
export const ActionBindingRefTypeEnum = {
    BindingId: "binding_id",
    ActionId: "action_id",
    ActionName: "action_name",
};
/** In order to execute an Action, it must be bound to a trigger using a binding. `trigger-bound` means that bindings are managed by the tenant. `entity-bound` means that the bindings are automatically managed by Auth0 and other internal resouces will control those bindings. Tenants cannot manage `entity-bound` bindings. */
export const ActionBindingTypeEnum = {
    TriggerBound: "trigger-bound",
    EntityBound: "entity-bound",
};
/** The build status of this action. */
export const ActionBuildStatusEnum = {
    Pending: "pending",
    Building: "building",
    Packaged: "packaged",
    Built: "built",
    Retrying: "retrying",
    Failed: "failed",
};
/** The overall status of an execution. */
export const ActionExecutionStatusEnum = {
    Unspecified: "unspecified",
    Pending: "pending",
    Final: "final",
    Partial: "partial",
    Canceled: "canceled",
    Suspended: "suspended",
};
/** An actions extensibility point. */
export const ActionTriggerTypeEnum = {
    PostLogin: "post-login",
    CredentialsExchange: "credentials-exchange",
    PreUserRegistration: "pre-user-registration",
    PostUserRegistration: "post-user-registration",
    PostChangePassword: "post-change-password",
    SendPhoneMessage: "send-phone-message",
    CustomPhoneProvider: "custom-phone-provider",
    CustomEmailProvider: "custom-email-provider",
    PasswordResetPostChallenge: "password-reset-post-challenge",
    CustomTokenExchange: "custom-token-exchange",
    EventStream: "event-stream",
    PasswordHashMigration: "password-hash-migration",
    LoginPostIdentifier: "login-post-identifier",
    SignupPostIdentifier: "signup-post-identifier",
};
/** The build status of this specific version. */
export const ActionVersionBuildStatusEnum = {
    Pending: "pending",
    Building: "building",
    Packaged: "packaged",
    Built: "built",
    Retrying: "retrying",
    Failed: "failed",
};
/** Static context values */
export const AculContextEnum = {
    BrandingSettings: "branding.settings",
    BrandingThemesDefault: "branding.themes.default",
    ClientLogoUri: "client.logo_uri",
    ClientDescription: "client.description",
    OrganizationDisplayName: "organization.display_name",
    OrganizationBranding: "organization.branding",
    ScreenTexts: "screen.texts",
    TenantName: "tenant.name",
    TenantFriendlyName: "tenant.friendly_name",
    TenantLogoUrl: "tenant.logo_url",
    TenantEnabledLocales: "tenant.enabled_locales",
    UntrustedDataSubmittedFormData: "untrusted_data.submitted_form_data",
    UntrustedDataAuthorizationParamsLoginHint: "untrusted_data.authorization_params.login_hint",
    UntrustedDataAuthorizationParamsScreenHint: "untrusted_data.authorization_params.screen_hint",
    UntrustedDataAuthorizationParamsUiLocales: "untrusted_data.authorization_params.ui_locales",
    UserOrganizations: "user.organizations",
    TransactionCustomDomainDomain: "transaction.custom_domain.domain",
};
/** Type of match to apply */
export const AculMatchTypeEnum = {
    IncludesAny: "includes_any",
    ExcludesAny: "excludes_any",
};
/** Rendering mode to filter by */
export const AculRenderingModeEnum = {
    Advanced: "advanced",
    Standard: "standard",
};
/** Assessors to clear. */
export const AssessorsTypeEnum = {
    NewDevice: "new-device",
};
export const AsyncApprovalNotificationsChannelsEnum = {
    GuardianPush: "guardian-push",
    Email: "email",
};
/** The id of the active provider for the CAPTCHA. */
export const AttackProtectionCaptchaProviderId = {
    Arkose: "arkose",
    AuthChallenge: "auth_challenge",
    FriendlyCaptcha: "friendly_captcha",
    Hcaptcha: "hcaptcha",
    RecaptchaV2: "recaptcha_v2",
    RecaptchaEnterprise: "recaptcha_enterprise",
    SimpleCaptcha: "simple_captcha",
};
export const AuthenticationMethodTypeEnum = {
    RecoveryCode: "recovery-code",
    Totp: "totp",
    Push: "push",
    Phone: "phone",
    Email: "email",
    EmailVerification: "email-verification",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
    Guardian: "guardian",
    Passkey: "passkey",
    Password: "password",
};
export const AuthenticationTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
};
export var BadRequestSchema;
(function (BadRequestSchema) {
    BadRequestSchema.Error_ = {
        BadRequest: "Bad Request",
    };
})(BadRequestSchema || (BadRequestSchema = {}));
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordResetFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordlessFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The level of bot detection sensitivity */
export const BotDetectionLevelEnum = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
/** Controls identifier input presentation on the login flow. Defaults to "unified" for legacy tenants, "separate" for tenants created post-GA of this feature. */
export const BrandingLoginDisplayEnum = {
    Unified: "unified",
    Separate: "separate",
};
/** Controls the format used when displaying phone numbers. */
export const BrandingPhoneFormattingEnum = {
    Regional: "regional",
    International: "international",
};
/** Controls how phone numbers are masked when displayed back to users. */
export const BrandingPhoneMaskingEnum = {
    ShowAll: "show_all",
    HideCountryCode: "hide_country_code",
    MaskDigits: "mask_digits",
};
/** Buttons style */
export const BrandingThemeBordersButtonsStyleEnum = {
    Pill: "pill",
    Rounded: "rounded",
    Sharp: "sharp",
};
/** Inputs style */
export const BrandingThemeBordersInputsStyleEnum = {
    Pill: "pill",
    Rounded: "rounded",
    Sharp: "sharp",
};
/** Captcha Widget Theme */
export const BrandingThemeColorsCaptchaWidgetThemeEnum = {
    Auto: "auto",
    Dark: "dark",
    Light: "light",
};
/** Links style */
export const BrandingThemeFontLinksStyleEnum = {
    Normal: "normal",
    Underlined: "underlined",
};
/** Page Layout */
export const BrandingThemePageBackgroundPageLayoutEnum = {
    Center: "center",
    Left: "left",
    Right: "right",
};
/** Header text alignment */
export const BrandingThemeWidgetHeaderTextAlignmentEnum = {
    Center: "center",
    Left: "left",
    Right: "right",
};
/** Logo position */
export const BrandingThemeWidgetLogoPositionEnum = {
    Center: "center",
    Left: "left",
    None: "none",
    Right: "right",
};
/** Social buttons layout */
export const BrandingThemeWidgetSocialButtonsLayoutEnum = {
    Bottom: "bottom",
    Top: "top",
};
export const BreachedPasswordDetectionAdminNotificationFrequencyEnum = {
    Immediately: "immediately",
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/**
 * The subscription level for breached password detection methods. Use "enhanced" to enable Credential Guard.
 *         Possible values: <code>standard</code>, <code>enhanced</code>.
 */
export const BreachedPasswordDetectionMethodEnum = {
    Standard: "standard",
    Enhanced: "enhanced",
};
export const BreachedPasswordDetectionPreChangePasswordShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
export const BreachedPasswordDetectionPreUserRegistrationShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
export const BreachedPasswordDetectionShieldsEnum = {
    Block: "block",
    UserNotification: "user_notification",
    AdminNotification: "admin_notification",
};
/**
 * Account Lockout: Determines whether or not IP address is used when counting failed attempts.
 *           Possible values: <code>count_per_identifier_and_ip</code>, <code>count_per_identifier</code>.
 */
export const BruteForceProtectionModeEnum = {
    CountPerIdentifierAndIp: "count_per_identifier_and_ip",
    CountPerIdentifier: "count_per_identifier",
};
export const BruteForceProtectionShieldsEnum = {
    Block: "block",
    UserNotification: "user_notification",
};
export const CertificateSubjectDnCredentialTypeEnum = {
    CertSubjectDn: "cert_subject_dn",
};
/** The type of application this client represents */
export const ClientAppTypeEnum = {
    Native: "native",
    Spa: "spa",
    RegularWeb: "regular_web",
    NonInteractive: "non_interactive",
    ResourceServer: "resource_server",
    ExpressConfiguration: "express_configuration",
    Rms: "rms",
    Box: "box",
    Cloudbees: "cloudbees",
    Concur: "concur",
    Dropbox: "dropbox",
    Mscrm: "mscrm",
    Echosign: "echosign",
    Egnyte: "egnyte",
    Newrelic: "newrelic",
    Office365: "office365",
    Salesforce: "salesforce",
    Sentry: "sentry",
    Sharepoint: "sharepoint",
    Slack: "slack",
    Springcm: "springcm",
    Zendesk: "zendesk",
    Zoom: "zoom",
    SsoIntegration: "sso_integration",
    Oag: "oag",
};
/** Defines the compliance level for this client, which may restrict it's capabilities */
export const ClientComplianceLevelEnum = {
    None: "none",
    Fapi1AdvPkjPar: "fapi1_adv_pkj_par",
    Fapi1AdvMtlsPar: "fapi1_adv_mtls_par",
    Fapi2SpPkjMtls: "fapi2_sp_pkj_mtls",
    Fapi2SpMtlsMtls: "fapi2_sp_mtls_mtls",
};
/** Algorithm which will be used with the credential. Supported algorithms: RS256,RS384,PS256 */
export const ClientCredentialAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Ps256: "PS256",
};
/** The type of credential. */
export const ClientCredentialTypeEnum = {
    PublicKey: "public_key",
    CertSubjectDn: "cert_subject_dn",
    X509Cert: "x509_cert",
};
export const ClientDefaultOrganizationFlowsEnum = {
    ClientCredentials: "client_credentials",
};
/** Indicates who created the external metadata client. The value <code>admin</code> indicates the client was registered via the Management API. The value <code>client</code> indicates the client was registered dynamically. This field is only present when external_metadata_type is set. */
export const ClientExternalMetadataCreatedByEnum = {
    Admin: "admin",
    Client: "client",
};
/** Indicates the type of external metadata used to register the client. This field is omitted for regular clients. The value <code>cimd</code> identifies clients registered via a Client ID Metadata Document. */
export const ClientExternalMetadataTypeEnum = {
    Cimd: "cimd",
};
/** Applies this client grant as the default for all clients in the specified group. The only accepted value is <a href="https://auth0.com/docs/get-started/applications/application-access-to-apis-client-grants#default-permissions-for-third-party-applications">`third_party_clients`</a>, which applies the grant to all third-party clients. Per-client grants for the same audience take precedence. Mutually exclusive with `client_id`. */
export const ClientGrantDefaultForEnum = {
    ThirdPartyClients: "third_party_clients",
};
/** Controls how organizations may be used with this grant */
export const ClientGrantOrganizationNullableUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Defines whether organizations can be used with client credentials exchanges for this grant. */
export const ClientGrantOrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** The type of application access the client grant allows. */
export const ClientGrantSubjectTypeEnum = {
    Client: "client",
    User: "user",
};
/** The allowed connection strategy values for the My Organization Configuration. */
export const ClientMyOrganizationConfigurationAllowedStrategiesEnum = {
    Pingfederate: "pingfederate",
    Adfs: "adfs",
    Waad: "waad",
    GoogleApps: "google-apps",
    Okta: "okta",
    Oidc: "oidc",
    Samlp: "samlp",
};
/** The deletion behavior for this client. */
export const ClientMyOrganizationDeletionBehaviorEnum = {
    Allow: "allow",
    AllowIfEmpty: "allow_if_empty",
};
/** The `selected_initiators` property contains the list of initiators to be enabled for the given application. */
export const ClientOidcBackchannelLogoutInitiatorsEnum = {
    RpLogout: "rp-logout",
    IdpLogout: "idp-logout",
    PasswordChanged: "password-changed",
    SessionExpired: "session-expired",
    SessionRevoked: "session-revoked",
    AccountDeleted: "account-deleted",
    EmailIdentifierChanged: "email-identifier-changed",
    MfaPhoneUnenrolled: "mfa-phone-unenrolled",
    AccountDeactivated: "account-deactivated",
};
/** The `mode` property determines the configuration method for enabling initiators. `custom` enables only the initiators listed in the selected_initiators array, `all` enables all current and future initiators. */
export const ClientOidcBackchannelLogoutInitiatorsModeEnum = {
    Custom: "custom",
    All: "all",
};
/** Method for discovering organizations during the `pre_login_prompt`. `email` allows users to find their organization by entering their email address and performing domain matching, while `organization_name` requires users to enter the organization name directly. These methods can be combined. */
export const ClientOrganizationDiscoveryEnum = {
    Email: "email",
    OrganizationName: "organization_name",
};
/** Defines how to proceed during an authentication transaction when `client.organization_usage: 'require'`. Can be `no_prompt` (default), `pre_login_prompt` or `post_login_prompt`. `post_login_prompt` requires `oidc_conformant: true`. */
export const ClientOrganizationRequireBehaviorEnum = {
    NoPrompt: "no_prompt",
    PreLoginPrompt: "pre_login_prompt",
    PostLoginPrompt: "post_login_prompt",
};
/** Defines how to proceed during an authentication transaction when `client.organization_usage: 'require'`. Can be `no_prompt` (default), `pre_login_prompt` or `post_login_prompt`. `post_login_prompt` requires `oidc_conformant: true`. */
export const ClientOrganizationRequireBehaviorPatchEnum = {
    NoPrompt: "no_prompt",
    PreLoginPrompt: "pre_login_prompt",
    PostLoginPrompt: "post_login_prompt",
};
/** Defines how to proceed during an authentication transaction with regards an organization. Can be `deny` (default), `allow` or `require`. */
export const ClientOrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Defines how to proceed during an authentication transaction with regards an organization. Can be `deny` (default), `allow` or `require`. */
export const ClientOrganizationUsagePatchEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Controls whether Auth0 redirects users to the application's callback URL on authentication errors or in email verification flows. `open_redirect_protection` shows an error page instead of redirecting, and hides the callback domain from email templates. `allow_always` enables standard redirect behavior. Defaults to `open_redirect_protection` for third-party clients. Only applies when `is_first_party` is `false` and `third_party_security_mode` is `strict`. To learn more, read <a href="https://auth0.com/docs/get-started/applications/third-party-applications/security-controls#redirect-protection">Redirect protection</a>. */
export const ClientRedirectionPolicyEnum = {
    AllowAlways: "allow_always",
    OpenRedirectProtection: "open_redirect_protection",
};
export const ClientSessionTransferAllowedAuthenticationMethodsEnum = {
    Cookie: "cookie",
    Query: "query",
};
/** Indicates the device binding enforcement for delegation (impersonation) access. If set to 'ip', device binding is enforced by IP. If set to 'asn', device binding is enforced by ASN. Default value is `ip`. */
export const ClientSessionTransferDelegationDeviceBindingEnum = {
    Ip: "ip",
    Asn: "asn",
};
/** Indicates whether device binding security should be enforced for the app. If set to 'ip', the app will enforce device binding by IP, meaning that consumption of Session Transfer Token must be done from the same IP of the issuer. Likewise, if set to 'asn', device binding is enforced by ASN, meaning consumption of Session Transfer Token must be done from the same ASN as the issuer. If set to 'none', device binding is not enforced. Usually configured in the web application. Default value is `ip`. */
export const ClientSessionTransferDeviceBindingEnum = {
    Ip: "ip",
    Asn: "asn",
    None: "none",
};
/** Security mode for third-party clients. `strict` enforces <a href="https://auth0.com/docs/get-started/applications/third-party-applications/security-controls">enhanced security controls</a>: OAuth 2.1 alignment, explicit API authorization, and a curated set of supported features. `permissive` preserves <a href="https://auth0.com/docs/get-started/applications/third-party-applications/permissive-mode">pre-existing behavior</a> and is only available to tenants with prior third-party client usage. Set on creation and cannot be modified. */
export const ClientThirdPartySecurityModeEnum = {
    Strict: "strict",
    Permissive: "permissive",
};
/** Defines the requested authentication method for the token endpoint. Can be `none` (public client without a client secret), `client_secret_post` (client uses HTTP POST parameters), or `client_secret_basic` (client uses HTTP Basic). */
export const ClientTokenEndpointAuthMethodEnum = {
    None: "none",
    ClientSecretPost: "client_secret_post",
    ClientSecretBasic: "client_secret_basic",
};
/** Defines the requested authentication method for the token endpoint. Can be `none` (public client without a client secret), `client_secret_post` (client uses HTTP POST parameters), or `client_secret_basic` (client uses HTTP Basic). */
export const ClientTokenEndpointAuthMethodOrNullEnum = {
    None: "none",
    ClientSecretPost: "client_secret_post",
    ClientSecretBasic: "client_secret_basic",
};
/** Token exchange type. `on_behalf_of_token_exchange`: enables On-Behalf-Of token exchange (Generally Available). `custom_authentication`: enables custom token exchange profiles (Early Access, requires entitlement). */
export const ClientTokenExchangeTypeEnum = {
    CustomAuthentication: "custom_authentication",
    OnBehalfOfTokenExchange: "on_behalf_of_token_exchange",
};
/** The access type for the connected account. */
export const ConnectedAccountAccessTypeEnum = {
    Offline: "offline",
};
/** Specifies the API behavior for password authentication */
export const ConnectionApiBehaviorEnum = {
    Required: "required",
    Optional: "optional",
};
/** The algorithm profile to use for decrypting SAML assertions. */
export const ConnectionAssertionDecryptionAlgorithmProfileEnum = {
    V20261: "v2026-1",
};
/** PKCE configuration. */
export const ConnectionConnectionSettingsPkceEnum = {
    Auto: "auto",
    S256: "S256",
    Plain: "plain",
    Disabled: "disabled",
};
/** Algorithm used for computing digest values when signing SAML requests and logout requests. Defaults to 'sha256'. */
export const ConnectionDigestAlgorithmEnumSaml = {
    Sha1: "sha1",
    Sha256: "sha256",
};
/** Algorithm used for DPoP proof JWT signing. */
export const ConnectionDpopSigningAlgEnum = {
    Es256: "ES256",
    Ed25519: "Ed25519",
};
export var ConnectionEmailEmail;
(function (ConnectionEmailEmail) {
    /** Email template syntax type */
    ConnectionEmailEmail.Syntax = {
        Liquid: "liquid",
    };
})(ConnectionEmailEmail || (ConnectionEmailEmail = {}));
/** Algorithm allowed to verify the ID tokens. */
export const ConnectionIdTokenSignedResponseAlgEnum = {
    Es256: "ES256",
    Es384: "ES384",
    Ps256: "PS256",
    Ps384: "PS384",
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
};
/** Order of precedence for attribute types */
export const ConnectionIdentifierPrecedenceEnum = {
    Email: "email",
    PhoneNumber: "phone_number",
    Username: "username",
};
/** Identity API version to use */
export const ConnectionIdentityApiEnumAzureAd = {
    MicrosoftIdentityPlatformV20: "microsoft-identity-platform-v2.0",
    AzureActiveDirectoryV10: "azure-active-directory-v1.0",
};
/** The identity provider identifier for the connection */
export const ConnectionIdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** Signing key use, whether for encryption or signing */
export const ConnectionKeyUseEnum = {
    Encryption: "encryption",
    Signing: "signing",
};
/** Method used to map incoming claims when strategy=oidc. */
export const ConnectionMappingModeEnumOidc = {
    BindAll: "bind_all",
    UseMap: "use_map",
};
/** Method used to map incoming claims when strategy=okta. */
export const ConnectionMappingModeEnumOkta = {
    BasicProfile: "basic_profile",
    UseMap: "use_map",
};
/** The response protocol used to communicate with the default application. */
export const ConnectionOptionsIdpInitiatedClientProtocolEnumSaml = {
    Oidc: "oidc",
    Samlp: "samlp",
    Wsfed: "wsfed",
};
/** Allowed protocol identifiers for Twitter connections. Use oauth1 for the legacy v1.1 OAuth flow or oauth2 for the newer OAuth 2.0 flow. */
export const ConnectionOptionsProtocolEnumTwitter = {
    Oauth1: "oauth1",
    Oauth2: "oauth2",
};
/** Controls the UI used to challenge the user for their passkey. */
export const ConnectionPasskeyChallengeUiEnum = {
    Both: "both",
    Autofill: "autofill",
    Button: "button",
};
/** Password strength level */
export const ConnectionPasswordPolicyEnum = {
    None: "none",
    Low: "low",
    Fair: "fair",
    Good: "good",
    Excellent: "excellent",
};
/** Indicates if membership should be assigned on login. */
export const ConnectionProfileOrganizationAssignMembershipOnLoginEnum = {
    None: "none",
    Optional: "optional",
    Required: "required",
};
/** Indicates if the organization should be shown as a button. */
export const ConnectionProfileOrganizationShowAsButtonEnum = {
    None: "none",
    Optional: "optional",
    Required: "required",
};
/** Valid SAML protocol bindings */
export const ConnectionProtocolBindingEnumSaml = {
    UrnOasisNamesTcSaml20BindingsHttpPost: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
    UrnOasisNamesTcSaml20BindingsHttpRedirect: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
};
/** SMS provider values. Use 'sms_gateway' to use a custom SMS gateway instead of Twilio. */
export const ConnectionProviderEnumSms = {
    SmsGateway: "sms_gateway",
    Twilio: "twilio",
};
export var ConnectionResponseContentAd;
(function (ConnectionResponseContentAd) {
    ConnectionResponseContentAd.Strategy = {
        Ad: "ad",
    };
})(ConnectionResponseContentAd || (ConnectionResponseContentAd = {}));
export var ConnectionResponseContentAdfs;
(function (ConnectionResponseContentAdfs) {
    ConnectionResponseContentAdfs.Strategy = {
        Adfs: "adfs",
    };
})(ConnectionResponseContentAdfs || (ConnectionResponseContentAdfs = {}));
export var ConnectionResponseContentAmazon;
(function (ConnectionResponseContentAmazon) {
    ConnectionResponseContentAmazon.Strategy = {
        Amazon: "amazon",
    };
})(ConnectionResponseContentAmazon || (ConnectionResponseContentAmazon = {}));
export var ConnectionResponseContentApple;
(function (ConnectionResponseContentApple) {
    ConnectionResponseContentApple.Strategy = {
        Apple: "apple",
    };
})(ConnectionResponseContentApple || (ConnectionResponseContentApple = {}));
export var ConnectionResponseContentAuth0;
(function (ConnectionResponseContentAuth0) {
    ConnectionResponseContentAuth0.Strategy = {
        Auth0: "auth0",
    };
})(ConnectionResponseContentAuth0 || (ConnectionResponseContentAuth0 = {}));
export var ConnectionResponseContentAuth0Oidc;
(function (ConnectionResponseContentAuth0Oidc) {
    ConnectionResponseContentAuth0Oidc.Strategy = {
        Auth0Oidc: "auth0-oidc",
    };
})(ConnectionResponseContentAuth0Oidc || (ConnectionResponseContentAuth0Oidc = {}));
export var ConnectionResponseContentAzureAd;
(function (ConnectionResponseContentAzureAd) {
    ConnectionResponseContentAzureAd.Strategy = {
        Waad: "waad",
    };
})(ConnectionResponseContentAzureAd || (ConnectionResponseContentAzureAd = {}));
export var ConnectionResponseContentBaidu;
(function (ConnectionResponseContentBaidu) {
    ConnectionResponseContentBaidu.Strategy = {
        Baidu: "baidu",
    };
})(ConnectionResponseContentBaidu || (ConnectionResponseContentBaidu = {}));
export var ConnectionResponseContentBitbucket;
(function (ConnectionResponseContentBitbucket) {
    ConnectionResponseContentBitbucket.Strategy = {
        Bitbucket: "bitbucket",
    };
})(ConnectionResponseContentBitbucket || (ConnectionResponseContentBitbucket = {}));
export var ConnectionResponseContentBitly;
(function (ConnectionResponseContentBitly) {
    ConnectionResponseContentBitly.Strategy = {
        Bitly: "bitly",
    };
})(ConnectionResponseContentBitly || (ConnectionResponseContentBitly = {}));
export var ConnectionResponseContentBox;
(function (ConnectionResponseContentBox) {
    ConnectionResponseContentBox.Strategy = {
        Box: "box",
    };
})(ConnectionResponseContentBox || (ConnectionResponseContentBox = {}));
export var ConnectionResponseContentCustom;
(function (ConnectionResponseContentCustom) {
    ConnectionResponseContentCustom.Strategy = {
        Custom: "custom",
    };
})(ConnectionResponseContentCustom || (ConnectionResponseContentCustom = {}));
export var ConnectionResponseContentDaccount;
(function (ConnectionResponseContentDaccount) {
    ConnectionResponseContentDaccount.Strategy = {
        Daccount: "daccount",
    };
})(ConnectionResponseContentDaccount || (ConnectionResponseContentDaccount = {}));
export var ConnectionResponseContentDropbox;
(function (ConnectionResponseContentDropbox) {
    ConnectionResponseContentDropbox.Strategy = {
        Dropbox: "dropbox",
    };
})(ConnectionResponseContentDropbox || (ConnectionResponseContentDropbox = {}));
export var ConnectionResponseContentDwolla;
(function (ConnectionResponseContentDwolla) {
    ConnectionResponseContentDwolla.Strategy = {
        Dwolla: "dwolla",
    };
})(ConnectionResponseContentDwolla || (ConnectionResponseContentDwolla = {}));
export var ConnectionResponseContentEmail;
(function (ConnectionResponseContentEmail) {
    ConnectionResponseContentEmail.Strategy = {
        Email: "email",
    };
})(ConnectionResponseContentEmail || (ConnectionResponseContentEmail = {}));
export var ConnectionResponseContentEvernote;
(function (ConnectionResponseContentEvernote) {
    ConnectionResponseContentEvernote.Strategy = {
        Evernote: "evernote",
    };
})(ConnectionResponseContentEvernote || (ConnectionResponseContentEvernote = {}));
export var ConnectionResponseContentEvernoteSandbox;
(function (ConnectionResponseContentEvernoteSandbox) {
    ConnectionResponseContentEvernoteSandbox.Strategy = {
        EvernoteSandbox: "evernote-sandbox",
    };
})(ConnectionResponseContentEvernoteSandbox || (ConnectionResponseContentEvernoteSandbox = {}));
export var ConnectionResponseContentExact;
(function (ConnectionResponseContentExact) {
    ConnectionResponseContentExact.Strategy = {
        Exact: "exact",
    };
})(ConnectionResponseContentExact || (ConnectionResponseContentExact = {}));
export var ConnectionResponseContentFacebook;
(function (ConnectionResponseContentFacebook) {
    ConnectionResponseContentFacebook.Strategy = {
        Facebook: "facebook",
    };
})(ConnectionResponseContentFacebook || (ConnectionResponseContentFacebook = {}));
export var ConnectionResponseContentFitbit;
(function (ConnectionResponseContentFitbit) {
    ConnectionResponseContentFitbit.Strategy = {
        Fitbit: "fitbit",
    };
})(ConnectionResponseContentFitbit || (ConnectionResponseContentFitbit = {}));
export var ConnectionResponseContentGitHub;
(function (ConnectionResponseContentGitHub) {
    ConnectionResponseContentGitHub.Strategy = {
        Github: "github",
    };
})(ConnectionResponseContentGitHub || (ConnectionResponseContentGitHub = {}));
export var ConnectionResponseContentGoogleApps;
(function (ConnectionResponseContentGoogleApps) {
    ConnectionResponseContentGoogleApps.Strategy = {
        GoogleApps: "google-apps",
    };
})(ConnectionResponseContentGoogleApps || (ConnectionResponseContentGoogleApps = {}));
export var ConnectionResponseContentGoogleOAuth2;
(function (ConnectionResponseContentGoogleOAuth2) {
    ConnectionResponseContentGoogleOAuth2.Strategy = {
        GoogleOauth2: "google-oauth2",
    };
})(ConnectionResponseContentGoogleOAuth2 || (ConnectionResponseContentGoogleOAuth2 = {}));
export var ConnectionResponseContentIp;
(function (ConnectionResponseContentIp) {
    ConnectionResponseContentIp.Strategy = {
        Ip: "ip",
    };
})(ConnectionResponseContentIp || (ConnectionResponseContentIp = {}));
export var ConnectionResponseContentInstagram;
(function (ConnectionResponseContentInstagram) {
    ConnectionResponseContentInstagram.Strategy = {
        Instagram: "instagram",
    };
})(ConnectionResponseContentInstagram || (ConnectionResponseContentInstagram = {}));
export var ConnectionResponseContentLine;
(function (ConnectionResponseContentLine) {
    ConnectionResponseContentLine.Strategy = {
        Line: "line",
    };
})(ConnectionResponseContentLine || (ConnectionResponseContentLine = {}));
export var ConnectionResponseContentLinkedin;
(function (ConnectionResponseContentLinkedin) {
    ConnectionResponseContentLinkedin.Strategy = {
        Linkedin: "linkedin",
    };
})(ConnectionResponseContentLinkedin || (ConnectionResponseContentLinkedin = {}));
export var ConnectionResponseContentOAuth1;
(function (ConnectionResponseContentOAuth1) {
    ConnectionResponseContentOAuth1.Strategy = {
        Oauth1: "oauth1",
    };
})(ConnectionResponseContentOAuth1 || (ConnectionResponseContentOAuth1 = {}));
export var ConnectionResponseContentOAuth2;
(function (ConnectionResponseContentOAuth2) {
    ConnectionResponseContentOAuth2.Strategy = {
        Oauth2: "oauth2",
    };
})(ConnectionResponseContentOAuth2 || (ConnectionResponseContentOAuth2 = {}));
export var ConnectionResponseContentOidc;
(function (ConnectionResponseContentOidc) {
    ConnectionResponseContentOidc.Strategy = {
        Oidc: "oidc",
    };
})(ConnectionResponseContentOidc || (ConnectionResponseContentOidc = {}));
export var ConnectionResponseContentOffice365;
(function (ConnectionResponseContentOffice365) {
    ConnectionResponseContentOffice365.Strategy = {
        Office365: "office365",
    };
})(ConnectionResponseContentOffice365 || (ConnectionResponseContentOffice365 = {}));
export var ConnectionResponseContentOkta;
(function (ConnectionResponseContentOkta) {
    ConnectionResponseContentOkta.Strategy = {
        Okta: "okta",
    };
})(ConnectionResponseContentOkta || (ConnectionResponseContentOkta = {}));
export var ConnectionResponseContentPaypal;
(function (ConnectionResponseContentPaypal) {
    ConnectionResponseContentPaypal.Strategy = {
        Paypal: "paypal",
    };
})(ConnectionResponseContentPaypal || (ConnectionResponseContentPaypal = {}));
export var ConnectionResponseContentPaypalSandbox;
(function (ConnectionResponseContentPaypalSandbox) {
    ConnectionResponseContentPaypalSandbox.Strategy = {
        PaypalSandbox: "paypal-sandbox",
    };
})(ConnectionResponseContentPaypalSandbox || (ConnectionResponseContentPaypalSandbox = {}));
export var ConnectionResponseContentPingFederate;
(function (ConnectionResponseContentPingFederate) {
    ConnectionResponseContentPingFederate.Strategy = {
        Pingfederate: "pingfederate",
    };
})(ConnectionResponseContentPingFederate || (ConnectionResponseContentPingFederate = {}));
export var ConnectionResponseContentPlanningCenter;
(function (ConnectionResponseContentPlanningCenter) {
    ConnectionResponseContentPlanningCenter.Strategy = {
        Planningcenter: "planningcenter",
    };
})(ConnectionResponseContentPlanningCenter || (ConnectionResponseContentPlanningCenter = {}));
export var ConnectionResponseContentSaml;
(function (ConnectionResponseContentSaml) {
    ConnectionResponseContentSaml.Strategy = {
        Samlp: "samlp",
    };
})(ConnectionResponseContentSaml || (ConnectionResponseContentSaml = {}));
export var ConnectionResponseContentSms;
(function (ConnectionResponseContentSms) {
    ConnectionResponseContentSms.Strategy = {
        Sms: "sms",
    };
})(ConnectionResponseContentSms || (ConnectionResponseContentSms = {}));
export var ConnectionResponseContentSalesforce;
(function (ConnectionResponseContentSalesforce) {
    ConnectionResponseContentSalesforce.Strategy = {
        Salesforce: "salesforce",
    };
})(ConnectionResponseContentSalesforce || (ConnectionResponseContentSalesforce = {}));
export var ConnectionResponseContentSalesforceCommunity;
(function (ConnectionResponseContentSalesforceCommunity) {
    ConnectionResponseContentSalesforceCommunity.Strategy = {
        SalesforceCommunity: "salesforce-community",
    };
})(ConnectionResponseContentSalesforceCommunity || (ConnectionResponseContentSalesforceCommunity = {}));
export var ConnectionResponseContentSalesforceSandbox;
(function (ConnectionResponseContentSalesforceSandbox) {
    ConnectionResponseContentSalesforceSandbox.Strategy = {
        SalesforceSandbox: "salesforce-sandbox",
    };
})(ConnectionResponseContentSalesforceSandbox || (ConnectionResponseContentSalesforceSandbox = {}));
export var ConnectionResponseContentSharepoint;
(function (ConnectionResponseContentSharepoint) {
    ConnectionResponseContentSharepoint.Strategy = {
        Sharepoint: "sharepoint",
    };
})(ConnectionResponseContentSharepoint || (ConnectionResponseContentSharepoint = {}));
export var ConnectionResponseContentShop;
(function (ConnectionResponseContentShop) {
    ConnectionResponseContentShop.Strategy = {
        Shop: "shop",
    };
})(ConnectionResponseContentShop || (ConnectionResponseContentShop = {}));
export var ConnectionResponseContentShopify;
(function (ConnectionResponseContentShopify) {
    ConnectionResponseContentShopify.Strategy = {
        Shopify: "shopify",
    };
})(ConnectionResponseContentShopify || (ConnectionResponseContentShopify = {}));
export var ConnectionResponseContentSoundcloud;
(function (ConnectionResponseContentSoundcloud) {
    ConnectionResponseContentSoundcloud.Strategy = {
        Soundcloud: "soundcloud",
    };
})(ConnectionResponseContentSoundcloud || (ConnectionResponseContentSoundcloud = {}));
export var ConnectionResponseContentThirtySevenSignals;
(function (ConnectionResponseContentThirtySevenSignals) {
    ConnectionResponseContentThirtySevenSignals.Strategy = {
        Thirtysevensignals: "thirtysevensignals",
    };
})(ConnectionResponseContentThirtySevenSignals || (ConnectionResponseContentThirtySevenSignals = {}));
export var ConnectionResponseContentTwitter;
(function (ConnectionResponseContentTwitter) {
    ConnectionResponseContentTwitter.Strategy = {
        Twitter: "twitter",
    };
})(ConnectionResponseContentTwitter || (ConnectionResponseContentTwitter = {}));
export var ConnectionResponseContentUntappd;
(function (ConnectionResponseContentUntappd) {
    ConnectionResponseContentUntappd.Strategy = {
        Untappd: "untappd",
    };
})(ConnectionResponseContentUntappd || (ConnectionResponseContentUntappd = {}));
export var ConnectionResponseContentVkontakte;
(function (ConnectionResponseContentVkontakte) {
    ConnectionResponseContentVkontakte.Strategy = {
        Vkontakte: "vkontakte",
    };
})(ConnectionResponseContentVkontakte || (ConnectionResponseContentVkontakte = {}));
export var ConnectionResponseContentWeibo;
(function (ConnectionResponseContentWeibo) {
    ConnectionResponseContentWeibo.Strategy = {
        Weibo: "weibo",
    };
})(ConnectionResponseContentWeibo || (ConnectionResponseContentWeibo = {}));
export var ConnectionResponseContentWindowsLive;
(function (ConnectionResponseContentWindowsLive) {
    ConnectionResponseContentWindowsLive.Strategy = {
        Windowslive: "windowslive",
    };
})(ConnectionResponseContentWindowsLive || (ConnectionResponseContentWindowsLive = {}));
export var ConnectionResponseContentWordpress;
(function (ConnectionResponseContentWordpress) {
    ConnectionResponseContentWordpress.Strategy = {
        Wordpress: "wordpress",
    };
})(ConnectionResponseContentWordpress || (ConnectionResponseContentWordpress = {}));
export var ConnectionResponseContentYahoo;
(function (ConnectionResponseContentYahoo) {
    ConnectionResponseContentYahoo.Strategy = {
        Yahoo: "yahoo",
    };
})(ConnectionResponseContentYahoo || (ConnectionResponseContentYahoo = {}));
export var ConnectionResponseContentYandex;
(function (ConnectionResponseContentYandex) {
    ConnectionResponseContentYandex.Strategy = {
        Yandex: "yandex",
    };
})(ConnectionResponseContentYandex || (ConnectionResponseContentYandex = {}));
/** When using an external IdP, this flag determines  whether 'name', 'given_name', 'family_name', 'nickname', and 'picture' attributes are updated. In addition, it also determines whether the user is created when user doesnt exist previously. Possible values are 'on_each_login' (default value, it configures the connection to automatically create the user if necessary and update the root attributes from the external IdP with each user login. When this setting is used, root attributes cannot be independently updated), 'on_first_login' (configures the connection to create the user and set the root attributes on first login only, allowing them to be independently updated thereafter), and 'never_on_login' (configures the connection not to create the user and not to set the root attributes from the external IdP, allowing them to be independently updated). */
export const ConnectionSetUserRootAttributesEnum = {
    OnEachLogin: "on_each_login",
    OnFirstLogin: "on_first_login",
    NeverOnLogin: "never_on_login",
};
/** Choose how Auth0 sets the email_verified field in the user profile. */
export const ConnectionShouldTrustEmailVerifiedConnectionEnum = {
    NeverSetEmailsAsVerified: "never_set_emails_as_verified",
    AlwaysSetEmailsAsVerified: "always_set_emails_as_verified",
};
/** Valid SAML signature algorithms */
export const ConnectionSignatureAlgorithmEnumSaml = {
    RsaSha1: "rsa-sha1",
    RsaSha256: "rsa-sha256",
};
/** OAuth 1.0a signature algorithm used when signing request-token and access-token calls for this connection (RSA-SHA1 only). */
export const ConnectionSignatureMethodOAuth1 = {
    RsaSha1: "RSA-SHA1",
};
/** Specifies the signup behavior for password authentication */
export const ConnectionSignupBehaviorEnum = {
    Allow: "allow",
    Block: "block",
};
export const ConnectionStrategyEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
    Auth0Adldap: "auth0-adldap",
};
/** SMS template syntax type. Set to 'md_with_macros' to enable macro processing in templates. */
export const ConnectionTemplateSyntaxEnumSms = {
    Liquid: "liquid",
    MdWithMacros: "md_with_macros",
};
/** Authentication method used at the identity provider's token endpoint. 'client_secret_post' sends credentials in the request body; 'private_key_jwt' uses a signed JWT assertion for enhanced security. */
export const ConnectionTokenEndpointAuthMethodEnum = {
    ClientSecretPost: "client_secret_post",
    PrivateKeyJwt: "private_key_jwt",
};
/** Algorithm used to sign client_assertions. */
export const ConnectionTokenEndpointAuthSigningAlgEnum = {
    Es256: "ES256",
    Es384: "ES384",
    Ps256: "PS256",
    Ps384: "PS384",
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
};
/** Specifies the format of the aud (audience) claim included in the JWT used for client authentication at the token endpoint. Accepted values are: 'issuer' (the aud claim is set to the OIDC issuer URL) or 'token_endpoint' (the aud claim is set to the token endpoint URL). */
export const ConnectionTokenEndpointJwtcaAudFormatEnumOidc = {
    Issuer: "issuer",
    TokenEndpoint: "token_endpoint",
};
/** OIDC communication channel type. 'back_channel' (confidential client) exchanges tokens server-side for stronger security; 'front_channel' handles responses in the browser. */
export const ConnectionTypeEnumOidc = {
    BackChannel: "back_channel",
    FrontChannel: "front_channel",
};
/** Connection type */
export const ConnectionTypeEnumOkta = {
    BackChannel: "back_channel",
};
export const ConnectionUpstreamAliasEnum = {
    AcrValues: "acr_values",
    Audience: "audience",
    ClientId: "client_id",
    Display: "display",
    IdTokenHint: "id_token_hint",
    LoginHint: "login_hint",
    MaxAge: "max_age",
    Prompt: "prompt",
    Resource: "resource",
    ResponseMode: "response_mode",
    ResponseType: "response_type",
    UiLocales: "ui_locales",
};
/** User ID attribute to use. Only applies when waad_protocol=openid-connect */
export const ConnectionUseridAttributeEnumAzureAd = {
    Oid: "oid",
    Sub: "sub",
};
/** Available WAAD protocols */
export const ConnectionWaadProtocolEnumAzureAd = {
    WsFederation: "ws-federation",
    OpenidConnect: "openid-connect",
};
export var CreateConnectionRequestContentAd;
(function (CreateConnectionRequestContentAd) {
    CreateConnectionRequestContentAd.Strategy = {
        Ad: "ad",
    };
})(CreateConnectionRequestContentAd || (CreateConnectionRequestContentAd = {}));
export var CreateConnectionRequestContentAdfs;
(function (CreateConnectionRequestContentAdfs) {
    CreateConnectionRequestContentAdfs.Strategy = {
        Adfs: "adfs",
    };
})(CreateConnectionRequestContentAdfs || (CreateConnectionRequestContentAdfs = {}));
export var CreateConnectionRequestContentAmazon;
(function (CreateConnectionRequestContentAmazon) {
    CreateConnectionRequestContentAmazon.Strategy = {
        Amazon: "amazon",
    };
})(CreateConnectionRequestContentAmazon || (CreateConnectionRequestContentAmazon = {}));
export var CreateConnectionRequestContentApple;
(function (CreateConnectionRequestContentApple) {
    CreateConnectionRequestContentApple.Strategy = {
        Apple: "apple",
    };
})(CreateConnectionRequestContentApple || (CreateConnectionRequestContentApple = {}));
export var CreateConnectionRequestContentAuth0;
(function (CreateConnectionRequestContentAuth0) {
    CreateConnectionRequestContentAuth0.Strategy = {
        Auth0: "auth0",
    };
})(CreateConnectionRequestContentAuth0 || (CreateConnectionRequestContentAuth0 = {}));
export var CreateConnectionRequestContentAuth0Oidc;
(function (CreateConnectionRequestContentAuth0Oidc) {
    CreateConnectionRequestContentAuth0Oidc.Strategy = {
        Auth0Oidc: "auth0-oidc",
    };
})(CreateConnectionRequestContentAuth0Oidc || (CreateConnectionRequestContentAuth0Oidc = {}));
export var CreateConnectionRequestContentAzureAd;
(function (CreateConnectionRequestContentAzureAd) {
    CreateConnectionRequestContentAzureAd.Strategy = {
        Waad: "waad",
    };
})(CreateConnectionRequestContentAzureAd || (CreateConnectionRequestContentAzureAd = {}));
export var CreateConnectionRequestContentBaidu;
(function (CreateConnectionRequestContentBaidu) {
    CreateConnectionRequestContentBaidu.Strategy = {
        Baidu: "baidu",
    };
})(CreateConnectionRequestContentBaidu || (CreateConnectionRequestContentBaidu = {}));
export var CreateConnectionRequestContentBitbucket;
(function (CreateConnectionRequestContentBitbucket) {
    CreateConnectionRequestContentBitbucket.Strategy = {
        Bitbucket: "bitbucket",
    };
})(CreateConnectionRequestContentBitbucket || (CreateConnectionRequestContentBitbucket = {}));
export var CreateConnectionRequestContentBitly;
(function (CreateConnectionRequestContentBitly) {
    CreateConnectionRequestContentBitly.Strategy = {
        Bitly: "bitly",
    };
})(CreateConnectionRequestContentBitly || (CreateConnectionRequestContentBitly = {}));
export var CreateConnectionRequestContentBox;
(function (CreateConnectionRequestContentBox) {
    CreateConnectionRequestContentBox.Strategy = {
        Box: "box",
    };
})(CreateConnectionRequestContentBox || (CreateConnectionRequestContentBox = {}));
export var CreateConnectionRequestContentCustom;
(function (CreateConnectionRequestContentCustom) {
    CreateConnectionRequestContentCustom.Strategy = {
        Custom: "custom",
    };
})(CreateConnectionRequestContentCustom || (CreateConnectionRequestContentCustom = {}));
export var CreateConnectionRequestContentDaccount;
(function (CreateConnectionRequestContentDaccount) {
    CreateConnectionRequestContentDaccount.Strategy = {
        Daccount: "daccount",
    };
})(CreateConnectionRequestContentDaccount || (CreateConnectionRequestContentDaccount = {}));
export var CreateConnectionRequestContentDropbox;
(function (CreateConnectionRequestContentDropbox) {
    CreateConnectionRequestContentDropbox.Strategy = {
        Dropbox: "dropbox",
    };
})(CreateConnectionRequestContentDropbox || (CreateConnectionRequestContentDropbox = {}));
export var CreateConnectionRequestContentDwolla;
(function (CreateConnectionRequestContentDwolla) {
    CreateConnectionRequestContentDwolla.Strategy = {
        Dwolla: "dwolla",
    };
})(CreateConnectionRequestContentDwolla || (CreateConnectionRequestContentDwolla = {}));
export var CreateConnectionRequestContentEmail;
(function (CreateConnectionRequestContentEmail) {
    CreateConnectionRequestContentEmail.Strategy = {
        Email: "email",
    };
})(CreateConnectionRequestContentEmail || (CreateConnectionRequestContentEmail = {}));
export var CreateConnectionRequestContentEvernote;
(function (CreateConnectionRequestContentEvernote) {
    CreateConnectionRequestContentEvernote.Strategy = {
        Evernote: "evernote",
    };
})(CreateConnectionRequestContentEvernote || (CreateConnectionRequestContentEvernote = {}));
export var CreateConnectionRequestContentEvernoteSandbox;
(function (CreateConnectionRequestContentEvernoteSandbox) {
    CreateConnectionRequestContentEvernoteSandbox.Strategy = {
        EvernoteSandbox: "evernote-sandbox",
    };
})(CreateConnectionRequestContentEvernoteSandbox || (CreateConnectionRequestContentEvernoteSandbox = {}));
export var CreateConnectionRequestContentExact;
(function (CreateConnectionRequestContentExact) {
    CreateConnectionRequestContentExact.Strategy = {
        Exact: "exact",
    };
})(CreateConnectionRequestContentExact || (CreateConnectionRequestContentExact = {}));
export var CreateConnectionRequestContentFacebook;
(function (CreateConnectionRequestContentFacebook) {
    CreateConnectionRequestContentFacebook.Strategy = {
        Facebook: "facebook",
    };
})(CreateConnectionRequestContentFacebook || (CreateConnectionRequestContentFacebook = {}));
export var CreateConnectionRequestContentFitbit;
(function (CreateConnectionRequestContentFitbit) {
    CreateConnectionRequestContentFitbit.Strategy = {
        Fitbit: "fitbit",
    };
})(CreateConnectionRequestContentFitbit || (CreateConnectionRequestContentFitbit = {}));
export var CreateConnectionRequestContentGitHub;
(function (CreateConnectionRequestContentGitHub) {
    CreateConnectionRequestContentGitHub.Strategy = {
        Github: "github",
    };
})(CreateConnectionRequestContentGitHub || (CreateConnectionRequestContentGitHub = {}));
export var CreateConnectionRequestContentGoogleApps;
(function (CreateConnectionRequestContentGoogleApps) {
    CreateConnectionRequestContentGoogleApps.Strategy = {
        GoogleApps: "google-apps",
    };
})(CreateConnectionRequestContentGoogleApps || (CreateConnectionRequestContentGoogleApps = {}));
export var CreateConnectionRequestContentGoogleOAuth2;
(function (CreateConnectionRequestContentGoogleOAuth2) {
    CreateConnectionRequestContentGoogleOAuth2.Strategy = {
        GoogleOauth2: "google-oauth2",
    };
})(CreateConnectionRequestContentGoogleOAuth2 || (CreateConnectionRequestContentGoogleOAuth2 = {}));
export var CreateConnectionRequestContentIp;
(function (CreateConnectionRequestContentIp) {
    CreateConnectionRequestContentIp.Strategy = {
        Ip: "ip",
    };
})(CreateConnectionRequestContentIp || (CreateConnectionRequestContentIp = {}));
export var CreateConnectionRequestContentInstagram;
(function (CreateConnectionRequestContentInstagram) {
    CreateConnectionRequestContentInstagram.Strategy = {
        Instagram: "instagram",
    };
})(CreateConnectionRequestContentInstagram || (CreateConnectionRequestContentInstagram = {}));
export var CreateConnectionRequestContentLine;
(function (CreateConnectionRequestContentLine) {
    CreateConnectionRequestContentLine.Strategy = {
        Line: "line",
    };
})(CreateConnectionRequestContentLine || (CreateConnectionRequestContentLine = {}));
export var CreateConnectionRequestContentLinkedin;
(function (CreateConnectionRequestContentLinkedin) {
    CreateConnectionRequestContentLinkedin.Strategy = {
        Linkedin: "linkedin",
    };
})(CreateConnectionRequestContentLinkedin || (CreateConnectionRequestContentLinkedin = {}));
export var CreateConnectionRequestContentOAuth1;
(function (CreateConnectionRequestContentOAuth1) {
    CreateConnectionRequestContentOAuth1.Strategy = {
        Oauth1: "oauth1",
    };
})(CreateConnectionRequestContentOAuth1 || (CreateConnectionRequestContentOAuth1 = {}));
export var CreateConnectionRequestContentOAuth2;
(function (CreateConnectionRequestContentOAuth2) {
    CreateConnectionRequestContentOAuth2.Strategy = {
        Oauth2: "oauth2",
    };
})(CreateConnectionRequestContentOAuth2 || (CreateConnectionRequestContentOAuth2 = {}));
export var CreateConnectionRequestContentOidc;
(function (CreateConnectionRequestContentOidc) {
    CreateConnectionRequestContentOidc.Strategy = {
        Oidc: "oidc",
    };
})(CreateConnectionRequestContentOidc || (CreateConnectionRequestContentOidc = {}));
export var CreateConnectionRequestContentOffice365;
(function (CreateConnectionRequestContentOffice365) {
    CreateConnectionRequestContentOffice365.Strategy = {
        Office365: "office365",
    };
})(CreateConnectionRequestContentOffice365 || (CreateConnectionRequestContentOffice365 = {}));
export var CreateConnectionRequestContentOkta;
(function (CreateConnectionRequestContentOkta) {
    CreateConnectionRequestContentOkta.Strategy = {
        Okta: "okta",
    };
})(CreateConnectionRequestContentOkta || (CreateConnectionRequestContentOkta = {}));
export var CreateConnectionRequestContentPaypal;
(function (CreateConnectionRequestContentPaypal) {
    CreateConnectionRequestContentPaypal.Strategy = {
        Paypal: "paypal",
    };
})(CreateConnectionRequestContentPaypal || (CreateConnectionRequestContentPaypal = {}));
export var CreateConnectionRequestContentPaypalSandbox;
(function (CreateConnectionRequestContentPaypalSandbox) {
    CreateConnectionRequestContentPaypalSandbox.Strategy = {
        PaypalSandbox: "paypal-sandbox",
    };
})(CreateConnectionRequestContentPaypalSandbox || (CreateConnectionRequestContentPaypalSandbox = {}));
export var CreateConnectionRequestContentPingFederate;
(function (CreateConnectionRequestContentPingFederate) {
    CreateConnectionRequestContentPingFederate.Strategy = {
        Pingfederate: "pingfederate",
    };
})(CreateConnectionRequestContentPingFederate || (CreateConnectionRequestContentPingFederate = {}));
export var CreateConnectionRequestContentPlanningCenter;
(function (CreateConnectionRequestContentPlanningCenter) {
    CreateConnectionRequestContentPlanningCenter.Strategy = {
        Planningcenter: "planningcenter",
    };
})(CreateConnectionRequestContentPlanningCenter || (CreateConnectionRequestContentPlanningCenter = {}));
export var CreateConnectionRequestContentSaml;
(function (CreateConnectionRequestContentSaml) {
    CreateConnectionRequestContentSaml.Strategy = {
        Samlp: "samlp",
    };
})(CreateConnectionRequestContentSaml || (CreateConnectionRequestContentSaml = {}));
export var CreateConnectionRequestContentSms;
(function (CreateConnectionRequestContentSms) {
    CreateConnectionRequestContentSms.Strategy = {
        Sms: "sms",
    };
})(CreateConnectionRequestContentSms || (CreateConnectionRequestContentSms = {}));
export var CreateConnectionRequestContentSalesforce;
(function (CreateConnectionRequestContentSalesforce) {
    CreateConnectionRequestContentSalesforce.Strategy = {
        Salesforce: "salesforce",
    };
})(CreateConnectionRequestContentSalesforce || (CreateConnectionRequestContentSalesforce = {}));
export var CreateConnectionRequestContentSalesforceCommunity;
(function (CreateConnectionRequestContentSalesforceCommunity) {
    CreateConnectionRequestContentSalesforceCommunity.Strategy = {
        SalesforceCommunity: "salesforce-community",
    };
})(CreateConnectionRequestContentSalesforceCommunity || (CreateConnectionRequestContentSalesforceCommunity = {}));
export var CreateConnectionRequestContentSalesforceSandbox;
(function (CreateConnectionRequestContentSalesforceSandbox) {
    CreateConnectionRequestContentSalesforceSandbox.Strategy = {
        SalesforceSandbox: "salesforce-sandbox",
    };
})(CreateConnectionRequestContentSalesforceSandbox || (CreateConnectionRequestContentSalesforceSandbox = {}));
export var CreateConnectionRequestContentSharepoint;
(function (CreateConnectionRequestContentSharepoint) {
    CreateConnectionRequestContentSharepoint.Strategy = {
        Sharepoint: "sharepoint",
    };
})(CreateConnectionRequestContentSharepoint || (CreateConnectionRequestContentSharepoint = {}));
export var CreateConnectionRequestContentShop;
(function (CreateConnectionRequestContentShop) {
    CreateConnectionRequestContentShop.Strategy = {
        Shop: "shop",
    };
})(CreateConnectionRequestContentShop || (CreateConnectionRequestContentShop = {}));
export var CreateConnectionRequestContentShopify;
(function (CreateConnectionRequestContentShopify) {
    CreateConnectionRequestContentShopify.Strategy = {
        Shopify: "shopify",
    };
})(CreateConnectionRequestContentShopify || (CreateConnectionRequestContentShopify = {}));
export var CreateConnectionRequestContentSoundcloud;
(function (CreateConnectionRequestContentSoundcloud) {
    CreateConnectionRequestContentSoundcloud.Strategy = {
        Soundcloud: "soundcloud",
    };
})(CreateConnectionRequestContentSoundcloud || (CreateConnectionRequestContentSoundcloud = {}));
export var CreateConnectionRequestContentThirtySevenSignals;
(function (CreateConnectionRequestContentThirtySevenSignals) {
    CreateConnectionRequestContentThirtySevenSignals.Strategy = {
        Thirtysevensignals: "thirtysevensignals",
    };
})(CreateConnectionRequestContentThirtySevenSignals || (CreateConnectionRequestContentThirtySevenSignals = {}));
export var CreateConnectionRequestContentTwitter;
(function (CreateConnectionRequestContentTwitter) {
    CreateConnectionRequestContentTwitter.Strategy = {
        Twitter: "twitter",
    };
})(CreateConnectionRequestContentTwitter || (CreateConnectionRequestContentTwitter = {}));
export var CreateConnectionRequestContentUntappd;
(function (CreateConnectionRequestContentUntappd) {
    CreateConnectionRequestContentUntappd.Strategy = {
        Untappd: "untappd",
    };
})(CreateConnectionRequestContentUntappd || (CreateConnectionRequestContentUntappd = {}));
export var CreateConnectionRequestContentVkontakte;
(function (CreateConnectionRequestContentVkontakte) {
    CreateConnectionRequestContentVkontakte.Strategy = {
        Vkontakte: "vkontakte",
    };
})(CreateConnectionRequestContentVkontakte || (CreateConnectionRequestContentVkontakte = {}));
export var CreateConnectionRequestContentWeibo;
(function (CreateConnectionRequestContentWeibo) {
    CreateConnectionRequestContentWeibo.Strategy = {
        Weibo: "weibo",
    };
})(CreateConnectionRequestContentWeibo || (CreateConnectionRequestContentWeibo = {}));
export var CreateConnectionRequestContentWindowsLive;
(function (CreateConnectionRequestContentWindowsLive) {
    CreateConnectionRequestContentWindowsLive.Strategy = {
        Windowslive: "windowslive",
    };
})(CreateConnectionRequestContentWindowsLive || (CreateConnectionRequestContentWindowsLive = {}));
export var CreateConnectionRequestContentWordpress;
(function (CreateConnectionRequestContentWordpress) {
    CreateConnectionRequestContentWordpress.Strategy = {
        Wordpress: "wordpress",
    };
})(CreateConnectionRequestContentWordpress || (CreateConnectionRequestContentWordpress = {}));
export var CreateConnectionRequestContentYahoo;
(function (CreateConnectionRequestContentYahoo) {
    CreateConnectionRequestContentYahoo.Strategy = {
        Yahoo: "yahoo",
    };
})(CreateConnectionRequestContentYahoo || (CreateConnectionRequestContentYahoo = {}));
export var CreateConnectionRequestContentYandex;
(function (CreateConnectionRequestContentYandex) {
    CreateConnectionRequestContentYandex.Strategy = {
        Yandex: "yandex",
    };
})(CreateConnectionRequestContentYandex || (CreateConnectionRequestContentYandex = {}));
/** Type of the encryption key to be created. */
export const CreateEncryptionKeyType = {
    CustomerProvidedRootKey: "customer-provided-root-key",
    TenantEncryptionKey: "tenant-encryption-key",
};
export const CreatedAuthenticationMethodTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
    WebauthnRoaming: "webauthn-roaming",
};
export const CreatedUserAuthenticationMethodTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
    WebauthnRoaming: "webauthn-roaming",
    Passkey: "passkey",
};
/** Applies to passkeys only. The kind of device the credential is stored on as defined by backup eligibility. "single_device" credentials cannot be backed up and synced to another device, "multi_device" credentials can be backed up if enabled by the end-user. */
export const CredentialDeviceTypeEnum = {
    SingleDevice: "single_device",
    MultiDevice: "multi_device",
};
/** The HTTP header to fetch the client's IP address */
export const CustomDomainCustomClientIpHeaderEnum = {
    TrueClientIp: "true-client-ip",
    CfConnectingIp: "cf-connecting-ip",
    XForwardedFor: "x-forwarded-for",
    XAzureClientip: "x-azure-clientip",
    Empty: "",
};
/** Custom domain provisioning type. Must be `auth0_managed_certs` or `self_managed_certs`. */
export const CustomDomainProvisioningTypeEnum = {
    Auth0ManagedCerts: "auth0_managed_certs",
    SelfManagedCerts: "self_managed_certs",
};
/** Custom domain configuration status. Can be `failed`, `pending_verification`, or `ready`. */
export const CustomDomainStatusFilterEnum = {
    PendingVerification: "pending_verification",
    Ready: "ready",
    Failed: "failed",
};
/** Custom domain TLS policy. Must be `recommended`, includes TLS 1.2. */
export const CustomDomainTlsPolicyEnum = {
    Recommended: "recommended",
};
/** Custom domain provisioning type. Can be `auth0_managed_certs` or `self_managed_certs`. */
export const CustomDomainTypeEnum = {
    Auth0ManagedCerts: "auth0_managed_certs",
    SelfManagedCerts: "self_managed_certs",
};
/** Custom domain verification method. Must be `txt`. */
export const CustomDomainVerificationMethodEnum = {
    Txt: "txt",
};
export const CustomProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
/** Key algorithm */
export const CustomSigningKeyAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Es256: "ES256",
    Es384: "ES384",
    Es512: "ES512",
    Ps256: "PS256",
    Ps384: "PS384",
    Ps512: "PS512",
};
/** Curve */
export const CustomSigningKeyCurveEnum = {
    P256: "P-256",
    P384: "P-384",
    P521: "P-521",
};
export const CustomSigningKeyOperationEnum = {
    Verify: "verify",
};
/** Key type */
export const CustomSigningKeyTypeEnum = {
    Ec: "EC",
    Rsa: "RSA",
};
/** Key use */
export const CustomSigningKeyUseEnum = {
    Sig: "sig",
};
/** Default authentication method for email identifier */
export const DefaultMethodEmailIdentifierEnum = {
    Password: "password",
    EmailOtp: "email_otp",
};
/** Type of credential. Must be `public_key`. */
export const DeviceCredentialPublicKeyTypeEnum = {
    PublicKey: "public_key",
};
/** Type of credentials to retrieve. Must be `public_key`, `refresh_token` or `rotating_refresh_token`. The property will default to `refresh_token` when paging is requested */
export const DeviceCredentialTypeEnum = {
    PublicKey: "public_key",
    RefreshToken: "refresh_token",
    RotatingRefreshToken: "rotating_refresh_token",
};
/** The Certificate Authority issued the certificate. */
export const DomainCertificateAuthorityEnum = {
    Letsencrypt: "letsencrypt",
    Googletrust: "googletrust",
};
/** The provisioning status of the certificate. */
export const DomainCertificateStatusEnum = {
    Provisioning: "provisioning",
    ProvisioningFailed: "provisioning_failed",
    Provisioned: "provisioned",
    RenewingFailed: "renewing_failed",
};
/** Domain verification method. */
export const DomainVerificationMethodNameEnum = {
    Cname: "cname",
    Txt: "txt",
};
/** The DNS record verification status. This field is relevant only for Custom Domains with Auth0-Managed Certificates. */
export const DomainVerificationStatusEnum = {
    Verified: "verified",
    Pending: "pending",
    Failed: "failed",
};
/** Set to <code>eu</code> if your domain is provisioned to use Mailgun's EU region. Otherwise, set to <code>null</code>. */
export const EmailMailgunRegionEnum = {
    Eu: "eu",
};
/** Name of the email provider. Can be `mailgun`, `mandrill`, `sendgrid`, `resend`, `ses`, `sparkpost`, `smtp`, `azure_cs`, `ms365`, or `custom`. */
export const EmailProviderNameEnum = {
    Mailgun: "mailgun",
    Mandrill: "mandrill",
    Sendgrid: "sendgrid",
    Resend: "resend",
    Ses: "ses",
    Sparkpost: "sparkpost",
    Smtp: "smtp",
    AzureCs: "azure_cs",
    Ms365: "ms365",
    Custom: "custom",
};
/** Set to <code>eu</code> to use SparkPost service hosted in Western Europe. To use SparkPost hosted in North America, set it to <code>null</code>. */
export const EmailSparkPostRegionEnum = {
    Eu: "eu",
};
/** Template name. Can be `verify_email`, `verify_email_by_code`, `reset_email`, `reset_email_by_code`, `welcome_email`, `blocked_account`, `stolen_credentials`, `enrollment_email`, `mfa_oob_code`, `user_invitation`, `async_approval`, `change_password` (legacy), or `password_reset` (legacy). */
export const EmailTemplateNameEnum = {
    VerifyEmail: "verify_email",
    VerifyEmailByCode: "verify_email_by_code",
    ResetEmail: "reset_email",
    ResetEmailByCode: "reset_email_by_code",
    WelcomeEmail: "welcome_email",
    BlockedAccount: "blocked_account",
    StolenCredentials: "stolen_credentials",
    EnrollmentEmail: "enrollment_email",
    MfaOobCode: "mfa_oob_code",
    UserInvitation: "user_invitation",
    ChangePassword: "change_password",
    PasswordReset: "password_reset",
    AsyncApproval: "async_approval",
};
/** Enum for enabled features. */
export const EnabledFeaturesEnum = {
    Scim: "scim",
    UniversalLogout: "universal_logout",
};
/** Encryption algorithm that shall be used to wrap your key material */
export const EncryptionKeyPublicWrappingAlgorithm = {
    CkmRsaAesKeyWrap: "CKM_RSA_AES_KEY_WRAP",
};
/** Key state */
export const EncryptionKeyState = {
    PreActivation: "pre-activation",
    Active: "active",
    Deactivated: "deactivated",
    Destroyed: "destroyed",
};
/** Key type */
export const EncryptionKeyType = {
    CustomerProvidedRootKey: "customer-provided-root-key",
    EnvironmentRootKey: "environment-root-key",
    TenantMasterKey: "tenant-master-key",
    TenantEncryptionKey: "tenant-encryption-key",
};
export const EventStreamActionDestinationTypeEnum = {
    Action: "action",
};
/**
 * The purpose of this event. This field will typically appear only in special cases
 * such as sending a test event. For normal events, this field will be omitted.
 */
export const EventStreamCloudEventA0PurposeEnum = {
    Test: "test",
};
/** Machine-readable error code. */
export const EventStreamCloudEventErrorCodeEnum = {
    InvalidCursor: "invalid_cursor",
    CursorExpired: "cursor_expired",
    Timeout: "timeout",
    PayloadTooLarge: "payload_too_large",
    ProcessingError: "processing_error",
    ConnectionTimeout: "connection_timeout",
};
/** Identifies this as an error message (injected from the SSE event field). */
export const EventStreamCloudEventErrorMessageTypeEnum = {
    Error: "error",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupCreatedCloudEventTypeEnum = {
    GroupCreated: "group.created",
};
/** The type of the group. */
export const EventStreamCloudEventGroupCreatedObject0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupCreatedObject1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupCreatedObject2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupCreatedTypeEnum = {
    GroupCreated: "group.created",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupDeletedCloudEventTypeEnum = {
    GroupDeleted: "group.deleted",
};
/** The type of the group. */
export const EventStreamCloudEventGroupDeletedObject0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupDeletedObject1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupDeletedObject2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupDeletedTypeEnum = {
    GroupDeleted: "group.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupMemberAddedCloudEventTypeEnum = {
    GroupMemberAdded: "group.member.added",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberAddedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberAddedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberAddedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** Type discriminator for user members */
export const EventStreamCloudEventGroupMemberAddedObjectMember0MemberTypeEnum = {
    User: "user",
};
/** Type discriminator for connection members */
export const EventStreamCloudEventGroupMemberAddedObjectMember1MemberTypeEnum = {
    Connection: "connection",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupMemberAddedTypeEnum = {
    GroupMemberAdded: "group.member.added",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupMemberDeletedCloudEventTypeEnum = {
    GroupMemberDeleted: "group.member.deleted",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberDeletedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberDeletedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupMemberDeletedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** Type discriminator for user members */
export const EventStreamCloudEventGroupMemberDeletedObjectMember0MemberTypeEnum = {
    User: "user",
};
/** Type discriminator for connection members */
export const EventStreamCloudEventGroupMemberDeletedObjectMember1MemberTypeEnum = {
    Connection: "connection",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupMemberDeletedTypeEnum = {
    GroupMemberDeleted: "group.member.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupRoleAssignedCloudEventTypeEnum = {
    GroupRoleAssigned: "group.role.assigned",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleAssignedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleAssignedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleAssignedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupRoleAssignedTypeEnum = {
    GroupRoleAssigned: "group.role.assigned",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupRoleDeletedCloudEventTypeEnum = {
    GroupRoleDeleted: "group.role.deleted",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleDeletedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleDeletedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupRoleDeletedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupRoleDeletedTypeEnum = {
    GroupRoleDeleted: "group.role.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventGroupUpdatedCloudEventTypeEnum = {
    GroupUpdated: "group.updated",
};
/** The type of the group. */
export const EventStreamCloudEventGroupUpdatedObject0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventGroupUpdatedObject1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventGroupUpdatedObject2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventGroupUpdatedTypeEnum = {
    GroupUpdated: "group.updated",
};
/** Identifies this as an offset-only heartbeat message (injected from the SSE event field). */
export const EventStreamCloudEventOffsetOnlyMessageTypeEnum = {
    OffsetOnly: "offset-only",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgConnectionAddedCloudEventTypeEnum = {
    OrganizationConnectionAdded: "organization.connection.added",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgConnectionAddedTypeEnum = {
    OrganizationConnectionAdded: "organization.connection.added",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgConnectionRemovedCloudEventTypeEnum = {
    OrganizationConnectionRemoved: "organization.connection.removed",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgConnectionRemovedTypeEnum = {
    OrganizationConnectionRemoved: "organization.connection.removed",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgConnectionUpdatedCloudEventTypeEnum = {
    OrganizationConnectionUpdated: "organization.connection.updated",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgConnectionUpdatedTypeEnum = {
    OrganizationConnectionUpdated: "organization.connection.updated",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgCreatedCloudEventTypeEnum = {
    OrganizationCreated: "organization.created",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgCreatedTypeEnum = {
    OrganizationCreated: "organization.created",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgDeletedCloudEventTypeEnum = {
    OrganizationDeleted: "organization.deleted",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgDeletedTypeEnum = {
    OrganizationDeleted: "organization.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgGroupRoleAssignedCloudEventTypeEnum = {
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleAssignedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleAssignedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleAssignedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgGroupRoleAssignedTypeEnum = {
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgGroupRoleDeletedCloudEventTypeEnum = {
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleDeletedObjectGroup0TypeEnum = {
    Connection: "connection",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleDeletedObjectGroup1TypeEnum = {
    Organization: "organization",
};
/** The type of the group. */
export const EventStreamCloudEventOrgGroupRoleDeletedObjectGroup2TypeEnum = {
    Tenant: "tenant",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgGroupRoleDeletedTypeEnum = {
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgMemberAddedCloudEventTypeEnum = {
    OrganizationMemberAdded: "organization.member.added",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgMemberAddedTypeEnum = {
    OrganizationMemberAdded: "organization.member.added",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgMemberDeletedCloudEventTypeEnum = {
    OrganizationMemberDeleted: "organization.member.deleted",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgMemberDeletedTypeEnum = {
    OrganizationMemberDeleted: "organization.member.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgMemberRoleAssignedCloudEventTypeEnum = {
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgMemberRoleAssignedTypeEnum = {
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgMemberRoleDeletedCloudEventTypeEnum = {
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgMemberRoleDeletedTypeEnum = {
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventOrgUpdatedCloudEventTypeEnum = {
    OrganizationUpdated: "organization.updated",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventOrgUpdatedTypeEnum = {
    OrganizationUpdated: "organization.updated",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventUserCreatedCloudEventTypeEnum = {
    UserCreated: "user.created",
};
/** List of custom identity providers. */
export const EventStreamCloudEventUserCreatedObjectIdentitiesItemCustomProviderEnum = {
    Custom: "custom",
};
/** List of database identity providers. */
export const EventStreamCloudEventUserCreatedObjectIdentitiesItemDatabaseProviderEnum = {
    Auth0: "auth0",
};
/** List of enterprise identity providers. */
export const EventStreamCloudEventUserCreatedObjectIdentitiesItemEnterpriseProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    GoogleApps: "google-apps",
    Ip: "ip",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Pingfederate: "pingfederate",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Waad: "waad",
};
/** List of passwordless identity providers. */
export const EventStreamCloudEventUserCreatedObjectIdentitiesItemPasswordlessProviderEnum = {
    Email: "email",
    Sms: "sms",
};
/** List of social identity providers. */
export const EventStreamCloudEventUserCreatedObjectIdentitiesItemSocialProviderEnum = {
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Daccount: "daccount",
    Dwolla: "dwolla",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Shopify: "shopify",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untapped: "untapped",
    Vkontakte: "vkontakte",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventUserCreatedTypeEnum = {
    UserCreated: "user.created",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventUserDeletedCloudEventTypeEnum = {
    UserDeleted: "user.deleted",
};
/** List of custom identity providers. */
export const EventStreamCloudEventUserDeletedObjectIdentitiesItemCustomProviderEnum = {
    Custom: "custom",
};
/** List of database identity providers. */
export const EventStreamCloudEventUserDeletedObjectIdentitiesItemDatabaseProviderEnum = {
    Auth0: "auth0",
};
/** List of enterprise identity providers. */
export const EventStreamCloudEventUserDeletedObjectIdentitiesItemEnterpriseProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    GoogleApps: "google-apps",
    Ip: "ip",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Pingfederate: "pingfederate",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Waad: "waad",
};
/** List of passwordless identity providers. */
export const EventStreamCloudEventUserDeletedObjectIdentitiesItemPasswordlessProviderEnum = {
    Email: "email",
    Sms: "sms",
};
/** List of social identity providers. */
export const EventStreamCloudEventUserDeletedObjectIdentitiesItemSocialProviderEnum = {
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Daccount: "daccount",
    Dwolla: "dwolla",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Shopify: "shopify",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untapped: "untapped",
    Vkontakte: "vkontakte",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventUserDeletedTypeEnum = {
    UserDeleted: "user.deleted",
};
/** The type of the event which has happened. */
export const EventStreamCloudEventUserUpdatedCloudEventTypeEnum = {
    UserUpdated: "user.updated",
};
/** List of custom identity providers. */
export const EventStreamCloudEventUserUpdatedObjectIdentitiesItemCustomProviderEnum = {
    Custom: "custom",
};
/** List of database identity providers. */
export const EventStreamCloudEventUserUpdatedObjectIdentitiesItemDatabaseProviderEnum = {
    Auth0: "auth0",
};
/** List of enterprise identity providers. */
export const EventStreamCloudEventUserUpdatedObjectIdentitiesItemEnterpriseProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    GoogleApps: "google-apps",
    Ip: "ip",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Pingfederate: "pingfederate",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Waad: "waad",
};
/** List of passwordless identity providers. */
export const EventStreamCloudEventUserUpdatedObjectIdentitiesItemPasswordlessProviderEnum = {
    Email: "email",
    Sms: "sms",
};
/** List of social identity providers. */
export const EventStreamCloudEventUserUpdatedObjectIdentitiesItemSocialProviderEnum = {
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Daccount: "daccount",
    Dwolla: "dwolla",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Shopify: "shopify",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untapped: "untapped",
    Vkontakte: "vkontakte",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** The event type (injected from the SSE event field). */
export const EventStreamCloudEventUserUpdatedTypeEnum = {
    UserUpdated: "user.updated",
};
/** Type of event */
export const EventStreamDeliveryEventTypeEnum = {
    GroupCreated: "group.created",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
    GroupRoleAssigned: "group.role.assigned",
    GroupRoleDeleted: "group.role.deleted",
    GroupUpdated: "group.updated",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionRemoved: "organization.connection.removed",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationCreated: "organization.created",
    OrganizationDeleted: "organization.deleted",
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationUpdated: "organization.updated",
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
};
/** Delivery status */
export const EventStreamDeliveryStatusEnum = {
    Failed: "failed",
};
/** AWS Region for EventBridge destination. */
export const EventStreamEventBridgeAwsRegionEnum = {
    AfSouth1: "af-south-1",
    ApEast1: "ap-east-1",
    ApEast2: "ap-east-2",
    ApNortheast1: "ap-northeast-1",
    ApNortheast2: "ap-northeast-2",
    ApNortheast3: "ap-northeast-3",
    ApSouth1: "ap-south-1",
    ApSouth2: "ap-south-2",
    ApSoutheast1: "ap-southeast-1",
    ApSoutheast2: "ap-southeast-2",
    ApSoutheast3: "ap-southeast-3",
    ApSoutheast4: "ap-southeast-4",
    ApSoutheast5: "ap-southeast-5",
    ApSoutheast6: "ap-southeast-6",
    ApSoutheast7: "ap-southeast-7",
    CaCentral1: "ca-central-1",
    CaWest1: "ca-west-1",
    EuCentral1: "eu-central-1",
    EuCentral2: "eu-central-2",
    EuNorth1: "eu-north-1",
    EuSouth1: "eu-south-1",
    EuSouth2: "eu-south-2",
    EuWest1: "eu-west-1",
    EuWest2: "eu-west-2",
    EuWest3: "eu-west-3",
    IlCentral1: "il-central-1",
    MeCentral1: "me-central-1",
    MeSouth1: "me-south-1",
    MxCentral1: "mx-central-1",
    SaEast1: "sa-east-1",
    UsGovEast1: "us-gov-east-1",
    UsGovWest1: "us-gov-west-1",
    UsEast1: "us-east-1",
    UsEast2: "us-east-2",
    UsWest1: "us-west-1",
    UsWest2: "us-west-2",
};
export const EventStreamEventBridgeDestinationTypeEnum = {
    Eventbridge: "eventbridge",
};
export const EventStreamEventTypeEnum = {
    GroupCreated: "group.created",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
    GroupRoleAssigned: "group.role.assigned",
    GroupRoleDeleted: "group.role.deleted",
    GroupUpdated: "group.updated",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionRemoved: "organization.connection.removed",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationCreated: "organization.created",
    OrganizationDeleted: "organization.deleted",
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationUpdated: "organization.updated",
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
};
/** Indicates whether the event stream is actively forwarding events. */
export const EventStreamStatusEnum = {
    Enabled: "enabled",
    Disabled: "disabled",
};
export const EventStreamSubscribeEventsEventTypeEnum = {
    GroupCreated: "group.created",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
    GroupRoleAssigned: "group.role.assigned",
    GroupRoleDeleted: "group.role.deleted",
    GroupUpdated: "group.updated",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionRemoved: "organization.connection.removed",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationCreated: "organization.created",
    OrganizationDeleted: "organization.deleted",
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationUpdated: "organization.updated",
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
};
/** The type of event this test event represents. */
export const EventStreamTestEventTypeEnum = {
    GroupCreated: "group.created",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
    GroupRoleAssigned: "group.role.assigned",
    GroupRoleDeleted: "group.role.deleted",
    GroupUpdated: "group.updated",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionRemoved: "organization.connection.removed",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationCreated: "organization.created",
    OrganizationDeleted: "organization.deleted",
    OrganizationGroupRoleAssigned: "organization.group.role.assigned",
    OrganizationGroupRoleDeleted: "organization.group.role.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationUpdated: "organization.updated",
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
};
/** Type of authorization. */
export const EventStreamWebhookBasicAuthMethodEnum = {
    Basic: "basic",
};
/** Type of authorization. */
export const EventStreamWebhookBearerAuthMethodEnum = {
    Bearer: "bearer",
};
/** Type of authorization. */
export const EventStreamWebhookCustomHeaderAuthMethodEnum = {
    CustomHeader: "custom_header",
};
export const EventStreamWebhookDestinationTypeEnum = {
    Webhook: "webhook",
};
export var FlowActionActivecampaignListContacts;
(function (FlowActionActivecampaignListContacts) {
    FlowActionActivecampaignListContacts.Type = {
        Activecampaign: "ACTIVECAMPAIGN",
    };
    FlowActionActivecampaignListContacts.Action = {
        ListContacts: "LIST_CONTACTS",
    };
})(FlowActionActivecampaignListContacts || (FlowActionActivecampaignListContacts = {}));
export var FlowActionActivecampaignUpsertContact;
(function (FlowActionActivecampaignUpsertContact) {
    FlowActionActivecampaignUpsertContact.Type = {
        Activecampaign: "ACTIVECAMPAIGN",
    };
    FlowActionActivecampaignUpsertContact.Action = {
        UpsertContact: "UPSERT_CONTACT",
    };
})(FlowActionActivecampaignUpsertContact || (FlowActionActivecampaignUpsertContact = {}));
export var FlowActionAirtableCreateRecord;
(function (FlowActionAirtableCreateRecord) {
    FlowActionAirtableCreateRecord.Type = {
        Airtable: "AIRTABLE",
    };
    FlowActionAirtableCreateRecord.Action = {
        CreateRecord: "CREATE_RECORD",
    };
})(FlowActionAirtableCreateRecord || (FlowActionAirtableCreateRecord = {}));
export var FlowActionAirtableListRecords;
(function (FlowActionAirtableListRecords) {
    FlowActionAirtableListRecords.Type = {
        Airtable: "AIRTABLE",
    };
    FlowActionAirtableListRecords.Action = {
        ListRecords: "LIST_RECORDS",
    };
})(FlowActionAirtableListRecords || (FlowActionAirtableListRecords = {}));
export var FlowActionAirtableUpdateRecord;
(function (FlowActionAirtableUpdateRecord) {
    FlowActionAirtableUpdateRecord.Type = {
        Airtable: "AIRTABLE",
    };
    FlowActionAirtableUpdateRecord.Action = {
        UpdateRecord: "UPDATE_RECORD",
    };
})(FlowActionAirtableUpdateRecord || (FlowActionAirtableUpdateRecord = {}));
export var FlowActionAuth0CreateUser;
(function (FlowActionAuth0CreateUser) {
    FlowActionAuth0CreateUser.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0CreateUser.Action = {
        CreateUser: "CREATE_USER",
    };
})(FlowActionAuth0CreateUser || (FlowActionAuth0CreateUser = {}));
export var FlowActionAuth0GetUser;
(function (FlowActionAuth0GetUser) {
    FlowActionAuth0GetUser.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0GetUser.Action = {
        GetUser: "GET_USER",
    };
})(FlowActionAuth0GetUser || (FlowActionAuth0GetUser = {}));
export var FlowActionAuth0MakeCall;
(function (FlowActionAuth0MakeCall) {
    FlowActionAuth0MakeCall.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0MakeCall.Action = {
        MakeCall: "MAKE_CALL",
    };
})(FlowActionAuth0MakeCall || (FlowActionAuth0MakeCall = {}));
export var FlowActionAuth0SendEmail;
(function (FlowActionAuth0SendEmail) {
    FlowActionAuth0SendEmail.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0SendEmail.Action = {
        SendEmail: "SEND_EMAIL",
    };
})(FlowActionAuth0SendEmail || (FlowActionAuth0SendEmail = {}));
export var FlowActionAuth0SendRequest;
(function (FlowActionAuth0SendRequest) {
    FlowActionAuth0SendRequest.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0SendRequest.Action = {
        SendRequest: "SEND_REQUEST",
    };
})(FlowActionAuth0SendRequest || (FlowActionAuth0SendRequest = {}));
export var FlowActionAuth0SendRequestParams;
(function (FlowActionAuth0SendRequestParams) {
    FlowActionAuth0SendRequestParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
        Patch: "PATCH",
        Delete: "DELETE",
    };
})(FlowActionAuth0SendRequestParams || (FlowActionAuth0SendRequestParams = {}));
export var FlowActionAuth0SendSms;
(function (FlowActionAuth0SendSms) {
    FlowActionAuth0SendSms.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0SendSms.Action = {
        SendSms: "SEND_SMS",
    };
})(FlowActionAuth0SendSms || (FlowActionAuth0SendSms = {}));
export var FlowActionAuth0UpdateUser;
(function (FlowActionAuth0UpdateUser) {
    FlowActionAuth0UpdateUser.Type = {
        Auth0: "AUTH0",
    };
    FlowActionAuth0UpdateUser.Action = {
        UpdateUser: "UPDATE_USER",
    };
})(FlowActionAuth0UpdateUser || (FlowActionAuth0UpdateUser = {}));
export var FlowActionBigqueryInsertRows;
(function (FlowActionBigqueryInsertRows) {
    FlowActionBigqueryInsertRows.Type = {
        Bigquery: "BIGQUERY",
    };
    FlowActionBigqueryInsertRows.Action = {
        InsertRows: "INSERT_ROWS",
    };
})(FlowActionBigqueryInsertRows || (FlowActionBigqueryInsertRows = {}));
export var FlowActionClearbitFindCompany;
(function (FlowActionClearbitFindCompany) {
    FlowActionClearbitFindCompany.Type = {
        Clearbit: "CLEARBIT",
    };
    FlowActionClearbitFindCompany.Action = {
        FindCompany: "FIND_COMPANY",
    };
})(FlowActionClearbitFindCompany || (FlowActionClearbitFindCompany = {}));
export var FlowActionClearbitFindPerson;
(function (FlowActionClearbitFindPerson) {
    FlowActionClearbitFindPerson.Type = {
        Clearbit: "CLEARBIT",
    };
    FlowActionClearbitFindPerson.Action = {
        FindPerson: "FIND_PERSON",
    };
})(FlowActionClearbitFindPerson || (FlowActionClearbitFindPerson = {}));
export var FlowActionEmailVerifyEmail;
(function (FlowActionEmailVerifyEmail) {
    FlowActionEmailVerifyEmail.Type = {
        Email: "EMAIL",
    };
    FlowActionEmailVerifyEmail.Action = {
        VerifyEmail: "VERIFY_EMAIL",
    };
})(FlowActionEmailVerifyEmail || (FlowActionEmailVerifyEmail = {}));
export var FlowActionFlowBooleanCondition;
(function (FlowActionFlowBooleanCondition) {
    FlowActionFlowBooleanCondition.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowBooleanCondition.Action = {
        BooleanCondition: "BOOLEAN_CONDITION",
    };
})(FlowActionFlowBooleanCondition || (FlowActionFlowBooleanCondition = {}));
export var FlowActionFlowDelayFlow;
(function (FlowActionFlowDelayFlow) {
    FlowActionFlowDelayFlow.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowDelayFlow.Action = {
        DelayFlow: "DELAY_FLOW",
    };
})(FlowActionFlowDelayFlow || (FlowActionFlowDelayFlow = {}));
export var FlowActionFlowDelayFlowParams;
(function (FlowActionFlowDelayFlowParams) {
    FlowActionFlowDelayFlowParams.Units = {
        Seconds: "SECONDS",
        Minutes: "MINUTES",
        Hours: "HOURS",
        Days: "DAYS",
    };
})(FlowActionFlowDelayFlowParams || (FlowActionFlowDelayFlowParams = {}));
export var FlowActionFlowDoNothing;
(function (FlowActionFlowDoNothing) {
    FlowActionFlowDoNothing.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowDoNothing.Action = {
        DoNothing: "DO_NOTHING",
    };
})(FlowActionFlowDoNothing || (FlowActionFlowDoNothing = {}));
export var FlowActionFlowErrorMessage;
(function (FlowActionFlowErrorMessage) {
    FlowActionFlowErrorMessage.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowErrorMessage.Action = {
        ErrorMessage: "ERROR_MESSAGE",
    };
})(FlowActionFlowErrorMessage || (FlowActionFlowErrorMessage = {}));
export var FlowActionFlowMapValue;
(function (FlowActionFlowMapValue) {
    FlowActionFlowMapValue.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowMapValue.Action = {
        MapValue: "MAP_VALUE",
    };
})(FlowActionFlowMapValue || (FlowActionFlowMapValue = {}));
export var FlowActionFlowReturnJson;
(function (FlowActionFlowReturnJson) {
    FlowActionFlowReturnJson.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowReturnJson.Action = {
        ReturnJson: "RETURN_JSON",
    };
})(FlowActionFlowReturnJson || (FlowActionFlowReturnJson = {}));
export var FlowActionFlowStoreVars;
(function (FlowActionFlowStoreVars) {
    FlowActionFlowStoreVars.Type = {
        Flow: "FLOW",
    };
    FlowActionFlowStoreVars.Action = {
        StoreVars: "STORE_VARS",
    };
})(FlowActionFlowStoreVars || (FlowActionFlowStoreVars = {}));
export var FlowActionGoogleSheetsAddRow;
(function (FlowActionGoogleSheetsAddRow) {
    FlowActionGoogleSheetsAddRow.Type = {
        GoogleSheets: "GOOGLE_SHEETS",
    };
    FlowActionGoogleSheetsAddRow.Action = {
        AddRow: "ADD_ROW",
    };
})(FlowActionGoogleSheetsAddRow || (FlowActionGoogleSheetsAddRow = {}));
export var FlowActionHttpSendRequest;
(function (FlowActionHttpSendRequest) {
    FlowActionHttpSendRequest.Type = {
        Http: "HTTP",
    };
    FlowActionHttpSendRequest.Action = {
        SendRequest: "SEND_REQUEST",
    };
})(FlowActionHttpSendRequest || (FlowActionHttpSendRequest = {}));
export var FlowActionHttpSendRequestParams;
(function (FlowActionHttpSendRequestParams) {
    FlowActionHttpSendRequestParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
        Patch: "PATCH",
        Delete: "DELETE",
    };
    FlowActionHttpSendRequestParams.ContentType = {
        Json: "JSON",
        Form: "FORM",
        Xml: "XML",
    };
})(FlowActionHttpSendRequestParams || (FlowActionHttpSendRequestParams = {}));
export var FlowActionHubspotEnrollContact;
(function (FlowActionHubspotEnrollContact) {
    FlowActionHubspotEnrollContact.Type = {
        Hubspot: "HUBSPOT",
    };
    FlowActionHubspotEnrollContact.Action = {
        EnrollContact: "ENROLL_CONTACT",
    };
})(FlowActionHubspotEnrollContact || (FlowActionHubspotEnrollContact = {}));
export var FlowActionHubspotGetContact;
(function (FlowActionHubspotGetContact) {
    FlowActionHubspotGetContact.Type = {
        Hubspot: "HUBSPOT",
    };
    FlowActionHubspotGetContact.Action = {
        GetContact: "GET_CONTACT",
    };
})(FlowActionHubspotGetContact || (FlowActionHubspotGetContact = {}));
export var FlowActionHubspotUpsertContact;
(function (FlowActionHubspotUpsertContact) {
    FlowActionHubspotUpsertContact.Type = {
        Hubspot: "HUBSPOT",
    };
    FlowActionHubspotUpsertContact.Action = {
        UpsertContact: "UPSERT_CONTACT",
    };
})(FlowActionHubspotUpsertContact || (FlowActionHubspotUpsertContact = {}));
export var FlowActionJsonCreateJson;
(function (FlowActionJsonCreateJson) {
    FlowActionJsonCreateJson.Type = {
        Json: "JSON",
    };
    FlowActionJsonCreateJson.Action = {
        CreateJson: "CREATE_JSON",
    };
})(FlowActionJsonCreateJson || (FlowActionJsonCreateJson = {}));
export var FlowActionJsonParseJson;
(function (FlowActionJsonParseJson) {
    FlowActionJsonParseJson.Type = {
        Json: "JSON",
    };
    FlowActionJsonParseJson.Action = {
        ParseJson: "PARSE_JSON",
    };
})(FlowActionJsonParseJson || (FlowActionJsonParseJson = {}));
export var FlowActionJsonSerializeJson;
(function (FlowActionJsonSerializeJson) {
    FlowActionJsonSerializeJson.Type = {
        Json: "JSON",
    };
    FlowActionJsonSerializeJson.Action = {
        SerializeJson: "SERIALIZE_JSON",
    };
})(FlowActionJsonSerializeJson || (FlowActionJsonSerializeJson = {}));
export var FlowActionJwtDecodeJwt;
(function (FlowActionJwtDecodeJwt) {
    FlowActionJwtDecodeJwt.Type = {
        Jwt: "JWT",
    };
    FlowActionJwtDecodeJwt.Action = {
        DecodeJwt: "DECODE_JWT",
    };
})(FlowActionJwtDecodeJwt || (FlowActionJwtDecodeJwt = {}));
export var FlowActionJwtSignJwt;
(function (FlowActionJwtSignJwt) {
    FlowActionJwtSignJwt.Type = {
        Jwt: "JWT",
    };
    FlowActionJwtSignJwt.Action = {
        SignJwt: "SIGN_JWT",
    };
})(FlowActionJwtSignJwt || (FlowActionJwtSignJwt = {}));
export var FlowActionJwtVerifyJwt;
(function (FlowActionJwtVerifyJwt) {
    FlowActionJwtVerifyJwt.Type = {
        Jwt: "JWT",
    };
    FlowActionJwtVerifyJwt.Action = {
        VerifyJwt: "VERIFY_JWT",
    };
})(FlowActionJwtVerifyJwt || (FlowActionJwtVerifyJwt = {}));
export var FlowActionMailchimpUpsertMember;
(function (FlowActionMailchimpUpsertMember) {
    FlowActionMailchimpUpsertMember.Type = {
        Mailchimp: "MAILCHIMP",
    };
    FlowActionMailchimpUpsertMember.Action = {
        UpsertMember: "UPSERT_MEMBER",
    };
})(FlowActionMailchimpUpsertMember || (FlowActionMailchimpUpsertMember = {}));
export var FlowActionMailjetSendEmail;
(function (FlowActionMailjetSendEmail) {
    FlowActionMailjetSendEmail.Type = {
        Mailjet: "MAILJET",
    };
    FlowActionMailjetSendEmail.Action = {
        SendEmail: "SEND_EMAIL",
    };
})(FlowActionMailjetSendEmail || (FlowActionMailjetSendEmail = {}));
export var FlowActionOtpGenerateCode;
(function (FlowActionOtpGenerateCode) {
    FlowActionOtpGenerateCode.Type = {
        Otp: "OTP",
    };
    FlowActionOtpGenerateCode.Action = {
        GenerateCode: "GENERATE_CODE",
    };
})(FlowActionOtpGenerateCode || (FlowActionOtpGenerateCode = {}));
export var FlowActionOtpVerifyCode;
(function (FlowActionOtpVerifyCode) {
    FlowActionOtpVerifyCode.Type = {
        Otp: "OTP",
    };
    FlowActionOtpVerifyCode.Action = {
        VerifyCode: "VERIFY_CODE",
    };
})(FlowActionOtpVerifyCode || (FlowActionOtpVerifyCode = {}));
export var FlowActionPipedriveAddDeal;
(function (FlowActionPipedriveAddDeal) {
    FlowActionPipedriveAddDeal.Type = {
        Pipedrive: "PIPEDRIVE",
    };
    FlowActionPipedriveAddDeal.Action = {
        AddDeal: "ADD_DEAL",
    };
})(FlowActionPipedriveAddDeal || (FlowActionPipedriveAddDeal = {}));
export var FlowActionPipedriveAddOrganization;
(function (FlowActionPipedriveAddOrganization) {
    FlowActionPipedriveAddOrganization.Type = {
        Pipedrive: "PIPEDRIVE",
    };
    FlowActionPipedriveAddOrganization.Action = {
        AddOrganization: "ADD_ORGANIZATION",
    };
})(FlowActionPipedriveAddOrganization || (FlowActionPipedriveAddOrganization = {}));
export var FlowActionPipedriveAddPerson;
(function (FlowActionPipedriveAddPerson) {
    FlowActionPipedriveAddPerson.Type = {
        Pipedrive: "PIPEDRIVE",
    };
    FlowActionPipedriveAddPerson.Action = {
        AddPerson: "ADD_PERSON",
    };
})(FlowActionPipedriveAddPerson || (FlowActionPipedriveAddPerson = {}));
export var FlowActionSalesforceCreateLead;
(function (FlowActionSalesforceCreateLead) {
    FlowActionSalesforceCreateLead.Type = {
        Salesforce: "SALESFORCE",
    };
    FlowActionSalesforceCreateLead.Action = {
        CreateLead: "CREATE_LEAD",
    };
})(FlowActionSalesforceCreateLead || (FlowActionSalesforceCreateLead = {}));
export var FlowActionSalesforceGetLead;
(function (FlowActionSalesforceGetLead) {
    FlowActionSalesforceGetLead.Type = {
        Salesforce: "SALESFORCE",
    };
    FlowActionSalesforceGetLead.Action = {
        GetLead: "GET_LEAD",
    };
})(FlowActionSalesforceGetLead || (FlowActionSalesforceGetLead = {}));
export var FlowActionSalesforceSearchLeads;
(function (FlowActionSalesforceSearchLeads) {
    FlowActionSalesforceSearchLeads.Type = {
        Salesforce: "SALESFORCE",
    };
    FlowActionSalesforceSearchLeads.Action = {
        SearchLeads: "SEARCH_LEADS",
    };
})(FlowActionSalesforceSearchLeads || (FlowActionSalesforceSearchLeads = {}));
export var FlowActionSalesforceSearchLeadsParams;
(function (FlowActionSalesforceSearchLeadsParams) {
    FlowActionSalesforceSearchLeadsParams.SearchField = {
        Email: "email",
        Name: "name",
        Phone: "phone",
        All: "all",
    };
})(FlowActionSalesforceSearchLeadsParams || (FlowActionSalesforceSearchLeadsParams = {}));
export var FlowActionSalesforceUpdateLead;
(function (FlowActionSalesforceUpdateLead) {
    FlowActionSalesforceUpdateLead.Type = {
        Salesforce: "SALESFORCE",
    };
    FlowActionSalesforceUpdateLead.Action = {
        UpdateLead: "UPDATE_LEAD",
    };
})(FlowActionSalesforceUpdateLead || (FlowActionSalesforceUpdateLead = {}));
export var FlowActionSendgridSendEmail;
(function (FlowActionSendgridSendEmail) {
    FlowActionSendgridSendEmail.Type = {
        Sendgrid: "SENDGRID",
    };
    FlowActionSendgridSendEmail.Action = {
        SendEmail: "SEND_EMAIL",
    };
})(FlowActionSendgridSendEmail || (FlowActionSendgridSendEmail = {}));
export var FlowActionSlackPostMessage;
(function (FlowActionSlackPostMessage) {
    FlowActionSlackPostMessage.Type = {
        Slack: "SLACK",
    };
    FlowActionSlackPostMessage.Action = {
        PostMessage: "POST_MESSAGE",
    };
})(FlowActionSlackPostMessage || (FlowActionSlackPostMessage = {}));
export var FlowActionSlackPostMessageParamsAttachment;
(function (FlowActionSlackPostMessageParamsAttachment) {
    FlowActionSlackPostMessageParamsAttachment.Color = {
        Good: "GOOD",
        Warning: "WARNING",
        Danger: "DANGER",
    };
})(FlowActionSlackPostMessageParamsAttachment || (FlowActionSlackPostMessageParamsAttachment = {}));
export var FlowActionStripeAddTaxId;
(function (FlowActionStripeAddTaxId) {
    FlowActionStripeAddTaxId.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeAddTaxId.Action = {
        AddTaxId: "ADD_TAX_ID",
    };
})(FlowActionStripeAddTaxId || (FlowActionStripeAddTaxId = {}));
export var FlowActionStripeCreateCustomer;
(function (FlowActionStripeCreateCustomer) {
    FlowActionStripeCreateCustomer.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeCreateCustomer.Action = {
        CreateCustomer: "CREATE_CUSTOMER",
    };
})(FlowActionStripeCreateCustomer || (FlowActionStripeCreateCustomer = {}));
export var FlowActionStripeCreatePortalSession;
(function (FlowActionStripeCreatePortalSession) {
    FlowActionStripeCreatePortalSession.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeCreatePortalSession.Action = {
        CreatePortalSession: "CREATE_PORTAL_SESSION",
    };
})(FlowActionStripeCreatePortalSession || (FlowActionStripeCreatePortalSession = {}));
export var FlowActionStripeDeleteTaxId;
(function (FlowActionStripeDeleteTaxId) {
    FlowActionStripeDeleteTaxId.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeDeleteTaxId.Action = {
        DeleteTaxId: "DELETE_TAX_ID",
    };
})(FlowActionStripeDeleteTaxId || (FlowActionStripeDeleteTaxId = {}));
export var FlowActionStripeFindCustomers;
(function (FlowActionStripeFindCustomers) {
    FlowActionStripeFindCustomers.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeFindCustomers.Action = {
        FindCustomers: "FIND_CUSTOMERS",
    };
})(FlowActionStripeFindCustomers || (FlowActionStripeFindCustomers = {}));
export var FlowActionStripeGetCustomer;
(function (FlowActionStripeGetCustomer) {
    FlowActionStripeGetCustomer.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeGetCustomer.Action = {
        GetCustomer: "GET_CUSTOMER",
    };
})(FlowActionStripeGetCustomer || (FlowActionStripeGetCustomer = {}));
export var FlowActionStripeUpdateCustomer;
(function (FlowActionStripeUpdateCustomer) {
    FlowActionStripeUpdateCustomer.Type = {
        Stripe: "STRIPE",
    };
    FlowActionStripeUpdateCustomer.Action = {
        UpdateCustomer: "UPDATE_CUSTOMER",
    };
})(FlowActionStripeUpdateCustomer || (FlowActionStripeUpdateCustomer = {}));
export var FlowActionTelegramSendMessage;
(function (FlowActionTelegramSendMessage) {
    FlowActionTelegramSendMessage.Type = {
        Telegram: "TELEGRAM",
    };
    FlowActionTelegramSendMessage.Action = {
        SendMessage: "SEND_MESSAGE",
    };
})(FlowActionTelegramSendMessage || (FlowActionTelegramSendMessage = {}));
export var FlowActionTwilioMakeCall;
(function (FlowActionTwilioMakeCall) {
    FlowActionTwilioMakeCall.Type = {
        Twilio: "TWILIO",
    };
    FlowActionTwilioMakeCall.Action = {
        MakeCall: "MAKE_CALL",
    };
})(FlowActionTwilioMakeCall || (FlowActionTwilioMakeCall = {}));
export var FlowActionTwilioSendSms;
(function (FlowActionTwilioSendSms) {
    FlowActionTwilioSendSms.Type = {
        Twilio: "TWILIO",
    };
    FlowActionTwilioSendSms.Action = {
        SendSms: "SEND_SMS",
    };
})(FlowActionTwilioSendSms || (FlowActionTwilioSendSms = {}));
export var FlowActionWhatsappSendMessage;
(function (FlowActionWhatsappSendMessage) {
    FlowActionWhatsappSendMessage.Type = {
        Whatsapp: "WHATSAPP",
    };
    FlowActionWhatsappSendMessage.Action = {
        SendMessage: "SEND_MESSAGE",
    };
})(FlowActionWhatsappSendMessage || (FlowActionWhatsappSendMessage = {}));
export var FlowActionWhatsappSendMessageParams;
(function (FlowActionWhatsappSendMessageParams) {
    FlowActionWhatsappSendMessageParams.Type = {
        Audio: "AUDIO",
        Contacts: "CONTACTS",
        Document: "DOCUMENT",
        Image: "IMAGE",
        Interactive: "INTERACTIVE",
        Location: "LOCATION",
        Sticker: "STICKER",
        Template: "TEMPLATE",
        Text: "TEXT",
    };
})(FlowActionWhatsappSendMessageParams || (FlowActionWhatsappSendMessageParams = {}));
export var FlowActionXmlParseXml;
(function (FlowActionXmlParseXml) {
    FlowActionXmlParseXml.Type = {
        Xml: "XML",
    };
    FlowActionXmlParseXml.Action = {
        ParseXml: "PARSE_XML",
    };
})(FlowActionXmlParseXml || (FlowActionXmlParseXml = {}));
export var FlowActionXmlSerializeXml;
(function (FlowActionXmlSerializeXml) {
    FlowActionXmlSerializeXml.Type = {
        Xml: "XML",
    };
    FlowActionXmlSerializeXml.Action = {
        SerializeXml: "SERIALIZE_XML",
    };
})(FlowActionXmlSerializeXml || (FlowActionXmlSerializeXml = {}));
export var FlowActionZapierTriggerWebhook;
(function (FlowActionZapierTriggerWebhook) {
    FlowActionZapierTriggerWebhook.Type = {
        Zapier: "ZAPIER",
    };
    FlowActionZapierTriggerWebhook.Action = {
        TriggerWebhook: "TRIGGER_WEBHOOK",
    };
})(FlowActionZapierTriggerWebhook || (FlowActionZapierTriggerWebhook = {}));
export var FlowActionZapierTriggerWebhookParams;
(function (FlowActionZapierTriggerWebhookParams) {
    FlowActionZapierTriggerWebhookParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
    };
})(FlowActionZapierTriggerWebhookParams || (FlowActionZapierTriggerWebhookParams = {}));
export const FlowsVaultConnectioSetupJwtAlgorithmEnum = {
    Hs256: "HS256",
    Hs384: "HS384",
    Hs512: "HS512",
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Es256: "ES256",
    Es384: "ES384",
    Es512: "ES512",
    Ps256: "PS256",
    Ps384: "PS384",
    Ps512: "PS512",
};
export const FlowsVaultConnectioSetupTypeApiKeyEnum = {
    ApiKey: "API_KEY",
};
export const FlowsVaultConnectioSetupTypeBearerEnum = {
    Bearer: "BEARER",
};
export const FlowsVaultConnectioSetupTypeJwtEnum = {
    Jwt: "JWT",
};
export const FlowsVaultConnectioSetupTypeKeyPairEnum = {
    KeyPair: "KEY_PAIR",
};
export const FlowsVaultConnectioSetupTypeOauthAppEnum = {
    OauthApp: "OAUTH_APP",
};
export const FlowsVaultConnectioSetupTypeOauthCodeEnum = {
    OauthCode: "OAUTH_CODE",
};
export const FlowsVaultConnectioSetupTypeOauthJwtEnum = {
    OauthJwt: "OAUTH_JWT",
};
export const FlowsVaultConnectioSetupTypeTokenEnum = {
    Token: "TOKEN",
};
export const FlowsVaultConnectioSetupTypeWebhookEnum = {
    Webhook: "WEBHOOK",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdActivecampaignEnum = {
    Activecampaign: "ACTIVECAMPAIGN",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdAirtableEnum = {
    Airtable: "AIRTABLE",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdAuth0Enum = {
    Auth0: "AUTH0",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdBigqueryEnum = {
    Bigquery: "BIGQUERY",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdClearbitEnum = {
    Clearbit: "CLEARBIT",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdDocusignEnum = {
    Docusign: "DOCUSIGN",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdGoogleSheetsEnum = {
    GoogleSheets: "GOOGLE_SHEETS",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdHttpEnum = {
    Http: "HTTP",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdHubspotEnum = {
    Hubspot: "HUBSPOT",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdJwtEnum = {
    Jwt: "JWT",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdMailchimpEnum = {
    Mailchimp: "MAILCHIMP",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdMailjetEnum = {
    Mailjet: "MAILJET",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdPipedriveEnum = {
    Pipedrive: "PIPEDRIVE",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdSalesforceEnum = {
    Salesforce: "SALESFORCE",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdSendgridEnum = {
    Sendgrid: "SENDGRID",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdSlackEnum = {
    Slack: "SLACK",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdStripeEnum = {
    Stripe: "STRIPE",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdTelegramEnum = {
    Telegram: "TELEGRAM",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdTwilioEnum = {
    Twilio: "TWILIO",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdWhatsappEnum = {
    Whatsapp: "WHATSAPP",
};
/** Flows Vault Connection app identifier. */
export const FlowsVaultConnectionAppIdZapierEnum = {
    Zapier: "ZAPIER",
};
export const FlowsVaultConnectionHttpApiKeySetupInEnum = {
    Header: "HEADER",
    Query: "QUERY",
};
export const FlowsVaultConnectionSetupTypeApiKeyEnum = {
    ApiKey: "API_KEY",
};
export const FlowsVaultConnectionSetupTypeBasicAuthEnum = {
    BasicAuth: "BASIC_AUTH",
};
export const FlowsVaultConnectionSetupTypeOauthClientCredentialsEnum = {
    OauthClientCredentials: "OAUTH_CLIENT_CREDENTIALS",
};
export var ForbiddenSchema;
(function (ForbiddenSchema) {
    ForbiddenSchema.Error_ = {
        Forbidden: "Forbidden",
    };
})(ForbiddenSchema || (ForbiddenSchema = {}));
export const FormBlockImageConfigPositionEnum = {
    Left: "LEFT",
    Center: "CENTER",
    Right: "RIGHT",
};
export const FormBlockResendButtonConfigTextAlignmentEnum = {
    Left: "LEFT",
    Center: "CENTER",
    Right: "RIGHT",
};
export const FormBlockTypeDividerConst = {
    Divider: "DIVIDER",
};
export const FormBlockTypeHtmlConst = {
    Html: "HTML",
};
export const FormBlockTypeImageConst = {
    Image: "IMAGE",
};
export const FormBlockTypeJumpButtonConst = {
    JumpButton: "JUMP_BUTTON",
};
export const FormBlockTypeNextButtonConst = {
    NextButton: "NEXT_BUTTON",
};
export const FormBlockTypePreviousButtonConst = {
    PreviousButton: "PREVIOUS_BUTTON",
};
export const FormBlockTypeResendButtonConst = {
    ResendButton: "RESEND_BUTTON",
};
export const FormBlockTypeRichTextConst = {
    RichText: "RICH_TEXT",
};
export const FormComponentCategoryBlockConst = {
    Block: "BLOCK",
};
export const FormComponentCategoryFieldConst = {
    Field: "FIELD",
};
export const FormComponentCategoryWidgetConst = {
    Widget: "WIDGET",
};
export const FormEndingNodeId = {
    Ending: "$ending",
};
export const FormFieldDateConfigFormatEnum = {
    Date: "DATE",
    Time: "TIME",
};
export const FormFieldFileConfigCategoryEnum = {
    Audio: "AUDIO",
    Video: "VIDEO",
    Image: "IMAGE",
    Document: "DOCUMENT",
    Archive: "ARCHIVE",
};
export const FormFieldFileConfigStorageTypeEnum = {
    Managed: "MANAGED",
    Custom: "CUSTOM",
};
export const FormFieldPasswordConfigHashEnum = {
    None: "NONE",
    Md5: "MD5",
    Sha1: "SHA1",
    Sha256: "SHA256",
    Sha512: "SHA512",
};
export const FormFieldPaymentConfigChargeOneOffCurrencyEnum = {
    Aud: "AUD",
    Cad: "CAD",
    Chf: "CHF",
    Eur: "EUR",
    Gbp: "GBP",
    Inr: "INR",
    Mxn: "MXN",
    Sek: "SEK",
    Usd: "USD",
};
export const FormFieldPaymentConfigChargeTypeOneOffConst = {
    OneOff: "ONE_OFF",
};
export const FormFieldPaymentConfigChargeTypeSubscriptionConst = {
    Subscription: "SUBSCRIPTION",
};
export const FormFieldPaymentConfigProviderEnum = {
    Stripe: "STRIPE",
};
export const FormFieldTypeBooleanConst = {
    Boolean: "BOOLEAN",
};
export const FormFieldTypeCardsConst = {
    Cards: "CARDS",
};
export const FormFieldTypeChoiceConst = {
    Choice: "CHOICE",
};
export const FormFieldTypeCustomConst = {
    Custom: "CUSTOM",
};
export const FormFieldTypeDateConst = {
    Date: "DATE",
};
export const FormFieldTypeDropdownConst = {
    Dropdown: "DROPDOWN",
};
export const FormFieldTypeEmailConst = {
    Email: "EMAIL",
};
export const FormFieldTypeFileConst = {
    File: "FILE",
};
export const FormFieldTypeLegalConst = {
    Legal: "LEGAL",
};
export const FormFieldTypeNumberConst = {
    Number: "NUMBER",
};
export const FormFieldTypePasswordConst = {
    Password: "PASSWORD",
};
export const FormFieldTypePaymentConst = {
    Payment: "PAYMENT",
};
export const FormFieldTypeSocialConst = {
    Social: "SOCIAL",
};
export const FormFieldTypeTelConst = {
    Tel: "TEL",
};
export const FormFieldTypeTextConst = {
    Text: "TEXT",
};
export const FormFieldTypeUrlConst = {
    Url: "URL",
};
export const FormNodeTypeFlowConst = {
    Flow: "FLOW",
};
export const FormNodeTypeRouterConst = {
    Router: "ROUTER",
};
export const FormNodeTypeStepConst = {
    Step: "STEP",
};
export const FormWidgetTypeAuth0VerifiableCredentialsConst = {
    Auth0VerifiableCredentials: "AUTH0_VERIFIABLE_CREDENTIALS",
};
export const FormWidgetTypeGMapsAddressConst = {
    GmapsAddress: "GMAPS_ADDRESS",
};
export const FormWidgetTypeRecaptchaConst = {
    Recaptcha: "RECAPTCHA",
};
export const FormsRequestParametersHydrateEnum = {
    FlowCount: "flow_count",
    Links: "links",
};
export const GetFlowExecutionRequestParametersHydrateEnum = {
    Debug: "debug",
};
export const GetFlowRequestParametersHydrateEnum = {
    FormCount: "form_count",
    Forms: "forms",
};
/** Type of the member. */
export const GroupMemberTypeEnum = {
    User: "user",
    Group: "group",
};
/** Type of the group. */
export const GroupTypeEnum = {
    Connection: "connection",
    Organization: "organization",
    Tenant: "tenant",
};
/** Optional. Specifies which factor the user must enroll with.<br />Note: Parameter can only be used with Universal Login; it cannot be used with Classic Login or custom MFA pages. */
export const GuardianEnrollmentFactorEnum = {
    PushNotification: "push-notification",
    Phone: "phone",
    Email: "email",
    Otp: "otp",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
};
/** Status of this enrollment. Can be `pending` or `confirmed`. */
export const GuardianEnrollmentStatus = {
    Pending: "pending",
    Confirmed: "confirmed",
};
/** Factor name. Can be `sms`, `push-notification`, `email`, `duo` `otp` `webauthn-roaming`, `webauthn-platform`, or `recovery-code`. */
export const GuardianFactorNameEnum = {
    PushNotification: "push-notification",
    Sms: "sms",
    Email: "email",
    Duo: "duo",
    Otp: "otp",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
    RecoveryCode: "recovery-code",
};
export const GuardianFactorPhoneFactorMessageTypeEnum = {
    Sms: "sms",
    Voice: "voice",
};
export const GuardianFactorsProviderPushNotificationProviderDataEnum = {
    Guardian: "guardian",
    Sns: "sns",
    Direct: "direct",
};
export const GuardianFactorsProviderSmsProviderEnum = {
    Auth0: "auth0",
    Twilio: "twilio",
    PhoneMessageHook: "phone-message-hook",
};
/** Retrieves hooks that match the trigger */
export const HookTriggerIdEnum = {
    CredentialsExchange: "credentials-exchange",
    PreUserRegistration: "pre-user-registration",
    PostUserRegistration: "post-user-registration",
    PostChangePassword: "post-change-password",
    SendPhoneMessage: "send-phone-message",
};
/** Identity provider name of the identity (e.g. `google-oauth2`). */
export const IdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** Identity provider name of the identity. Only `auth0` is supported */
export const IdentityProviderOnlyAuth0Enum = {
    Auth0: "auth0",
};
/** feature_type is the type of the integration. */
export const IntegrationFeatureTypeEnum = {
    Unspecified: "unspecified",
    Action: "action",
    SocialConnection: "social_connection",
    LogStream: "log_stream",
    SsoIntegration: "sso_integration",
    SmsProvider: "sms_provider",
};
export const IntegrationRequiredParamTypeEnum = {
    Unspecified: "UNSPECIFIED",
    String: "STRING",
};
/** Format of the file. Must be `json` or `csv`. */
export const JobFileFormatEnum = {
    Json: "json",
    Csv: "csv",
};
export const ListFlowsRequestParametersHydrateEnum = {
    FormCount: "form_count",
};
export const LogStreamDatadogEnum = {
    Datadog: "datadog",
};
/** Datadog region */
export const LogStreamDatadogRegionEnum = {
    Us: "us",
    Eu: "eu",
    Us3: "us3",
    Us5: "us5",
};
export const LogStreamEventBridgeEnum = {
    Eventbridge: "eventbridge",
};
/** The region in which the EventBridge event source will be created */
export const LogStreamEventBridgeSinkRegionEnum = {
    AfSouth1: "af-south-1",
    ApEast1: "ap-east-1",
    ApEast2: "ap-east-2",
    ApNortheast1: "ap-northeast-1",
    ApNortheast2: "ap-northeast-2",
    ApNortheast3: "ap-northeast-3",
    ApSouth1: "ap-south-1",
    ApSouth2: "ap-south-2",
    ApSoutheast1: "ap-southeast-1",
    ApSoutheast2: "ap-southeast-2",
    ApSoutheast3: "ap-southeast-3",
    ApSoutheast4: "ap-southeast-4",
    ApSoutheast5: "ap-southeast-5",
    ApSoutheast6: "ap-southeast-6",
    ApSoutheast7: "ap-southeast-7",
    CaCentral1: "ca-central-1",
    CaWest1: "ca-west-1",
    EuCentral1: "eu-central-1",
    EuCentral2: "eu-central-2",
    EuNorth1: "eu-north-1",
    EuSouth1: "eu-south-1",
    EuSouth2: "eu-south-2",
    EuWest1: "eu-west-1",
    EuWest2: "eu-west-2",
    EuWest3: "eu-west-3",
    IlCentral1: "il-central-1",
    MeCentral1: "me-central-1",
    MeSouth1: "me-south-1",
    MxCentral1: "mx-central-1",
    SaEast1: "sa-east-1",
    UsGovEast1: "us-gov-east-1",
    UsGovWest1: "us-gov-west-1",
    UsEast1: "us-east-1",
    UsEast2: "us-east-2",
    UsWest1: "us-west-1",
    UsWest2: "us-west-2",
};
export const LogStreamEventGridEnum = {
    Eventgrid: "eventgrid",
};
/** Azure Region Name */
export const LogStreamEventGridRegionEnum = {
    Australiacentral: "australiacentral",
    Australiaeast: "australiaeast",
    Australiasoutheast: "australiasoutheast",
    Brazilsouth: "brazilsouth",
    Canadacentral: "canadacentral",
    Canadaeast: "canadaeast",
    Centralindia: "centralindia",
    Centralus: "centralus",
    Eastasia: "eastasia",
    Eastus: "eastus",
    Eastus2: "eastus2",
    Francecentral: "francecentral",
    Germanywestcentral: "germanywestcentral",
    Japaneast: "japaneast",
    Japanwest: "japanwest",
    Koreacentral: "koreacentral",
    Koreasouth: "koreasouth",
    Northcentralus: "northcentralus",
    Northeurope: "northeurope",
    Norwayeast: "norwayeast",
    Southafricanorth: "southafricanorth",
    Southcentralus: "southcentralus",
    Southeastasia: "southeastasia",
    Southindia: "southindia",
    Swedencentral: "swedencentral",
    Switzerlandnorth: "switzerlandnorth",
    Uaenorth: "uaenorth",
    Uksouth: "uksouth",
    Ukwest: "ukwest",
    Westcentralus: "westcentralus",
    Westeurope: "westeurope",
    Westindia: "westindia",
    Westus: "westus",
    Westus2: "westus2",
};
/** Category group name */
export const LogStreamFilterGroupNameEnum = {
    AuthLoginFail: "auth.login.fail",
    AuthLoginNotification: "auth.login.notification",
    AuthLoginSuccess: "auth.login.success",
    AuthLogoutFail: "auth.logout.fail",
    AuthLogoutSuccess: "auth.logout.success",
    AuthSignupFail: "auth.signup.fail",
    AuthSignupSuccess: "auth.signup.success",
    AuthSilentAuthFail: "auth.silent_auth.fail",
    AuthSilentAuthSuccess: "auth.silent_auth.success",
    AuthTokenExchangeFail: "auth.token_exchange.fail",
    AuthTokenExchangeSuccess: "auth.token_exchange.success",
    ManagementFail: "management.fail",
    ManagementSuccess: "management.success",
    ScimEvent: "scim.event",
    SystemNotification: "system.notification",
    UserFail: "user.fail",
    UserNotification: "user.notification",
    UserSuccess: "user.success",
    Actions: "actions",
    Other: "other",
};
/** Filter type. Currently `category` is the only valid type. */
export const LogStreamFilterTypeEnum = {
    Category: "category",
};
/** HTTP JSON format */
export const LogStreamHttpContentFormatEnum = {
    Jsonarray: "JSONARRAY",
    Jsonlines: "JSONLINES",
    Jsonobject: "JSONOBJECT",
};
export const LogStreamHttpEnum = {
    Http: "http",
};
export const LogStreamMixpanelEnum = {
    Mixpanel: "mixpanel",
};
/** Mixpanel Region */
export const LogStreamMixpanelRegionEnum = {
    Us: "us",
    Eu: "eu",
};
export const LogStreamPiiAlgorithmEnum = {
    Xxhash: "xxhash",
};
export const LogStreamPiiLogFieldsEnum = {
    FirstName: "first_name",
    LastName: "last_name",
    Username: "username",
    Email: "email",
    Phone: "phone",
    Address: "address",
};
export const LogStreamPiiMethodEnum = {
    Mask: "mask",
    Hash: "hash",
};
export const LogStreamSegmentEnum = {
    Segment: "segment",
};
export const LogStreamSplunkEnum = {
    Splunk: "splunk",
};
/** The status of the log stream. Possible values: `active`, `paused`, `suspended` */
export const LogStreamStatusEnum = {
    Active: "active",
    Paused: "paused",
    Suspended: "suspended",
};
export const LogStreamSumoEnum = {
    Sumo: "sumo",
};
/** The MFA policy */
export const MfaPolicyEnum = {
    AllApplications: "all-applications",
    ConfidenceScore: "confidence-score",
};
/** Identifies the origin of the request as the Management API (management), Authentication API (authentication), Dynamic Client Registration API (dynamic_client_registration), or any (tenant) */
export const NetworkAclRuleScopeEnum = {
    Management: "management",
    Authentication: "authentication",
    Tenant: "tenant",
    DynamicClientRegistration: "dynamic_client_registration",
};
/** Access level for the organization (e.g., "none", "full"). */
export const OrganizationAccessLevelEnum = {
    None: "none",
    Readonly: "readonly",
    Limited: "limited",
    Full: "full",
};
/** Access level for the organization (e.g., "none", "full"). */
export const OrganizationAccessLevelEnumWithNull = {
    None: "none",
    Readonly: "readonly",
    Limited: "limited",
    Full: "full",
};
/** The verification status of the discovery domain. */
export const OrganizationDiscoveryDomainStatus = {
    Pending: "pending",
    Verified: "verified",
};
/** Defines whether organizations can be used with client credentials exchanges for this grant. */
export const OrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Name of the prompt. */
export const PartialGroupsEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordless: "login-passwordless",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    CustomizedConsent: "customized-consent",
    Passkeys: "passkeys",
};
export const PasswordCharacterTypeEnum = {
    Uppercase: "uppercase",
    Lowercase: "lowercase",
    Number: "number",
    Special: "special",
};
/** When enabled, passwords must contain at least 3 out of 4 character types. Can only be enabled when all 4 character types are specified */
export const PasswordCharacterTypeRulePolicyEnum = {
    All: "all",
    ThreeOfFour: "three_of_four",
};
/** Default dictionary to use for password validation. Options: "en_10k" (10,000 common words) or "en_100k" (100,000 common words) */
export const PasswordDefaultDictionariesEnum = {
    En10K: "en_10k",
    En100K: "en_100k",
};
/** Controls whether identical consecutive characters are allowed in passwords */
export const PasswordIdenticalCharactersPolicyEnum = {
    Allow: "allow",
    Block: "block",
};
/** Controls whether passwords that exceed the maximum length are truncated or rejected */
export const PasswordMaxLengthExceededPolicyEnum = {
    Truncate: "truncate",
    Error: "error",
};
/** Controls whether sequential characters are allowed in passwords */
export const PasswordSequentialCharactersPolicyEnum = {
    Allow: "allow",
    Block: "block",
};
/** This depicts the type of notifications this provider can receive. */
export const PhoneProviderChannelEnum = {
    Phone: "phone",
};
/** The delivery method for the notification */
export const PhoneProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
/** Name of the phone notification provider */
export const PhoneProviderNameEnum = {
    Twilio: "twilio",
    Custom: "custom",
};
export const PhoneTemplateNotificationTypeEnum = {
    OtpVerify: "otp_verify",
    OtpEnroll: "otp_enroll",
    ChangePassword: "change_password",
    BlockedAccount: "blocked_account",
    PasswordBreach: "password_breach",
};
/** Selected Signing Algorithm */
export const PostConnectionKeysAlgEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Ps256: "PS256",
    Ps384: "PS384",
    Es256: "ES256",
    Es384: "ES384",
};
/** Applies to phone authentication methods only. The preferred communication method. */
export const PreferredAuthenticationMethodEnum = {
    Voice: "voice",
    Sms: "sms",
};
/** Name of the prompt */
export const PromptGroupNameEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordless: "login-passwordless",
    LoginEmailVerification: "login-email-verification",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    PhoneIdentifierEnrollment: "phone-identifier-enrollment",
    PhoneIdentifierChallenge: "phone-identifier-challenge",
    EmailIdentifierChallenge: "email-identifier-challenge",
    ResetPassword: "reset-password",
    CustomForm: "custom-form",
    Consent: "consent",
    CustomizedConsent: "customized-consent",
    Logout: "logout",
    MfaPush: "mfa-push",
    MfaOtp: "mfa-otp",
    MfaVoice: "mfa-voice",
    MfaPhone: "mfa-phone",
    MfaWebauthn: "mfa-webauthn",
    MfaSms: "mfa-sms",
    MfaEmail: "mfa-email",
    MfaRecoveryCode: "mfa-recovery-code",
    Mfa: "mfa",
    Status: "status",
    DeviceFlow: "device-flow",
    EmailVerification: "email-verification",
    EmailOtpChallenge: "email-otp-challenge",
    Organizations: "organizations",
    Invitation: "invitation",
    Common: "common",
    Passkeys: "passkeys",
    Captcha: "captcha",
    BruteForceProtection: "brute-force-protection",
    AsyncApprovalFlow: "async-approval-flow",
};
/** Language to update. */
export const PromptLanguageEnum = {
    Am: "am",
    Ar: "ar",
    ArEg: "ar-EG",
    ArSa: "ar-SA",
    Az: "az",
    Bg: "bg",
    Bn: "bn",
    Bs: "bs",
    CaEs: "ca-ES",
    Cnr: "cnr",
    Cs: "cs",
    Cy: "cy",
    Da: "da",
    De: "de",
    El: "el",
    En: "en",
    EnCa: "en-CA",
    Es: "es",
    Es419: "es-419",
    EsAr: "es-AR",
    EsMx: "es-MX",
    Et: "et",
    EuEs: "eu-ES",
    Fa: "fa",
    Fi: "fi",
    Fr: "fr",
    FrCa: "fr-CA",
    FrFr: "fr-FR",
    GlEs: "gl-ES",
    Gu: "gu",
    He: "he",
    Hi: "hi",
    Hr: "hr",
    Hu: "hu",
    Hy: "hy",
    Id: "id",
    Is: "is",
    It: "it",
    Ja: "ja",
    Ka: "ka",
    Kk: "kk",
    Kn: "kn",
    Ko: "ko",
    Lt: "lt",
    Lv: "lv",
    Mk: "mk",
    Ml: "ml",
    Mn: "mn",
    Mr: "mr",
    Ms: "ms",
    My: "my",
    Nb: "nb",
    Nl: "nl",
    Nn: "nn",
    No: "no",
    Pa: "pa",
    Pl: "pl",
    Pt: "pt",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    Ro: "ro",
    Ru: "ru",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Sq: "sq",
    Sr: "sr",
    Sv: "sv",
    Sw: "sw",
    Ta: "ta",
    Te: "te",
    Th: "th",
    Tl: "tl",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Zgh: "zgh",
    ZhCn: "zh-CN",
    ZhHk: "zh-HK",
    ZhMo: "zh-MO",
    ZhTw: "zh-TW",
};
/** Algorithm which will be used with the credential. Can be one of RS256, RS384, PS256. If not specified, RS256 will be used. Applies to `public_key` credential type. */
export const PublicKeyCredentialAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Ps256: "PS256",
};
/** Credential type. Supported types: public_key. */
export const PublicKeyCredentialTypeEnum = {
    PublicKey: "public_key",
};
/** Refresh token expiration types, one of: expiring, non-expiring */
export const RefreshTokenExpirationTypeEnum = {
    Expiring: "expiring",
    NonExpiring: "non-expiring",
};
/** Refresh token rotation types, one of: rotating, non-rotating */
export const RefreshTokenRotationTypeEnum = {
    Rotating: "rotating",
    NonRotating: "non-rotating",
};
export const ResourceServerConsentPolicyEnum = {
    TransactionalAuthorizationWithMfa: "transactional-authorization-with-mfa",
};
/** Intended mechanism for Proof-of-Possession */
export const ResourceServerProofOfPossessionMechanismEnum = {
    Mtls: "mtls",
    Dpop: "dpop",
};
/** Specifies which client types require Proof-of-Possession */
export const ResourceServerProofOfPossessionRequiredForEnum = {
    PublicClients: "public_clients",
    AllClients: "all_clients",
};
/** Defines the client flows policy for the resource server */
export const ResourceServerSubjectTypeAuthorizationClientPolicyEnum = {
    DenyAll: "deny_all",
    RequireClientGrant: "require_client_grant",
};
/** Defines the user flows policy for the resource server */
export const ResourceServerSubjectTypeAuthorizationUserPolicyEnum = {
    AllowAll: "allow_all",
    DenyAll: "deny_all",
    RequireClientGrant: "require_client_grant",
};
/** Dialect of access tokens that should be issued. `access_token` is a JWT containing standard Auth0 claims; `rfc9068_profile` is a JWT conforming to the IETF JWT Access Token Profile. `access_token_authz` and `rfc9068_profile_authz` additionally include RBAC permissions claims. */
export const ResourceServerTokenDialectResponseEnum = {
    AccessToken: "access_token",
    AccessTokenAuthz: "access_token_authz",
    Rfc9068Profile: "rfc9068_profile",
    Rfc9068ProfileAuthz: "rfc9068_profile_authz",
};
/** Dialect of issued access token. `access_token` is a JWT containing standard Auth0 claims; `rfc9068_profile` is a JWT conforming to the IETF JWT Access Token Profile. `access_token_authz` and `rfc9068_profile_authz` additionally include RBAC permissions claims. */
export const ResourceServerTokenDialectSchemaEnum = {
    AccessToken: "access_token",
    AccessTokenAuthz: "access_token_authz",
    Rfc9068Profile: "rfc9068_profile",
    Rfc9068ProfileAuthz: "rfc9068_profile_authz",
};
/** Algorithm used to encrypt the token. */
export const ResourceServerTokenEncryptionAlgorithmEnum = {
    RsaOaep256: "RSA-OAEP-256",
    RsaOaep384: "RSA-OAEP-384",
    RsaOaep512: "RSA-OAEP-512",
};
/** Format of the encrypted JWT payload. */
export const ResourceServerTokenEncryptionFormatEnum = {
    CompactNestedJwe: "compact-nested-jwe",
};
/** Selected Signing Algorithm */
export const RotateConnectionKeysSigningAlgEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Ps256: "PS256",
    Ps384: "PS384",
    Es256: "ES256",
    Es384: "ES384",
};
/** Name of the screen */
export const ScreenGroupNameEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordlessEmailCode: "login-passwordless-email-code",
    LoginPasswordlessEmailLink: "login-passwordless-email-link",
    LoginPasswordlessSmsOtp: "login-passwordless-sms-otp",
    LoginEmailVerification: "login-email-verification",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    PhoneIdentifierEnrollment: "phone-identifier-enrollment",
    PhoneIdentifierChallenge: "phone-identifier-challenge",
    EmailIdentifierChallenge: "email-identifier-challenge",
    ResetPasswordRequest: "reset-password-request",
    ResetPasswordEmail: "reset-password-email",
    ResetPassword: "reset-password",
    ResetPasswordSuccess: "reset-password-success",
    ResetPasswordError: "reset-password-error",
    ResetPasswordMfaEmailChallenge: "reset-password-mfa-email-challenge",
    ResetPasswordMfaOtpChallenge: "reset-password-mfa-otp-challenge",
    ResetPasswordMfaPhoneChallenge: "reset-password-mfa-phone-challenge",
    ResetPasswordMfaPushChallengePush: "reset-password-mfa-push-challenge-push",
    ResetPasswordMfaRecoveryCodeChallenge: "reset-password-mfa-recovery-code-challenge",
    ResetPasswordMfaSmsChallenge: "reset-password-mfa-sms-challenge",
    ResetPasswordMfaVoiceChallenge: "reset-password-mfa-voice-challenge",
    ResetPasswordMfaWebauthnPlatformChallenge: "reset-password-mfa-webauthn-platform-challenge",
    ResetPasswordMfaWebauthnRoamingChallenge: "reset-password-mfa-webauthn-roaming-challenge",
    CustomForm: "custom-form",
    Consent: "consent",
    CustomizedConsent: "customized-consent",
    Logout: "logout",
    LogoutComplete: "logout-complete",
    LogoutAborted: "logout-aborted",
    MfaPushWelcome: "mfa-push-welcome",
    MfaPushEnrollmentQr: "mfa-push-enrollment-qr",
    MfaPushEnrollmentCode: "mfa-push-enrollment-code",
    MfaPushSuccess: "mfa-push-success",
    MfaPushChallengePush: "mfa-push-challenge-push",
    MfaPushList: "mfa-push-list",
    MfaOtpEnrollmentQr: "mfa-otp-enrollment-qr",
    MfaOtpEnrollmentCode: "mfa-otp-enrollment-code",
    MfaOtpChallenge: "mfa-otp-challenge",
    MfaVoiceEnrollment: "mfa-voice-enrollment",
    MfaVoiceChallenge: "mfa-voice-challenge",
    MfaPhoneChallenge: "mfa-phone-challenge",
    MfaPhoneEnrollment: "mfa-phone-enrollment",
    MfaWebauthnPlatformEnrollment: "mfa-webauthn-platform-enrollment",
    MfaWebauthnRoamingEnrollment: "mfa-webauthn-roaming-enrollment",
    MfaWebauthnPlatformChallenge: "mfa-webauthn-platform-challenge",
    MfaWebauthnRoamingChallenge: "mfa-webauthn-roaming-challenge",
    MfaWebauthnChangeKeyNickname: "mfa-webauthn-change-key-nickname",
    MfaWebauthnEnrollmentSuccess: "mfa-webauthn-enrollment-success",
    MfaWebauthnError: "mfa-webauthn-error",
    MfaWebauthnNotAvailableError: "mfa-webauthn-not-available-error",
    MfaCountryCodes: "mfa-country-codes",
    MfaSmsEnrollment: "mfa-sms-enrollment",
    MfaSmsChallenge: "mfa-sms-challenge",
    MfaSmsList: "mfa-sms-list",
    MfaEmailChallenge: "mfa-email-challenge",
    MfaEmailList: "mfa-email-list",
    MfaRecoveryCodeEnrollment: "mfa-recovery-code-enrollment",
    MfaRecoveryCodeChallengeNewCode: "mfa-recovery-code-challenge-new-code",
    MfaRecoveryCodeChallenge: "mfa-recovery-code-challenge",
    MfaDetectBrowserCapabilities: "mfa-detect-browser-capabilities",
    MfaEnrollResult: "mfa-enroll-result",
    MfaLoginOptions: "mfa-login-options",
    MfaBeginEnrollOptions: "mfa-begin-enroll-options",
    Status: "status",
    DeviceCodeActivation: "device-code-activation",
    DeviceCodeActivationAllowed: "device-code-activation-allowed",
    DeviceCodeActivationDenied: "device-code-activation-denied",
    DeviceCodeConfirmation: "device-code-confirmation",
    EmailVerificationResult: "email-verification-result",
    EmailOtpChallenge: "email-otp-challenge",
    OrganizationSelection: "organization-selection",
    OrganizationPicker: "organization-picker",
    PreLoginOrganizationPicker: "pre-login-organization-picker",
    AcceptInvitation: "accept-invitation",
    RedeemTicket: "redeem-ticket",
    PasskeyEnrollment: "passkey-enrollment",
    PasskeyEnrollmentLocal: "passkey-enrollment-local",
    InterstitialCaptcha: "interstitial-captcha",
    BruteForceProtectionUnblock: "brute-force-protection-unblock",
    BruteForceProtectionUnblockSuccess: "brute-force-protection-unblock-success",
    BruteForceProtectionUnblockFailure: "brute-force-protection-unblock-failure",
    AsyncApprovalError: "async-approval-error",
    AsyncApprovalAccepted: "async-approval-accepted",
    AsyncApprovalDenied: "async-approval-denied",
    AsyncApprovalWrongUser: "async-approval-wrong-user",
};
/** The version of the search engine */
export const SearchEngineVersionsEnum = {
    V1: "v1",
    V2: "v2",
    V3: "v3",
};
export const SelfServiceProfileAllowedStrategyEnum = {
    Oidc: "oidc",
    Samlp: "samlp",
    Waad: "waad",
    GoogleApps: "google-apps",
    Adfs: "adfs",
    Okta: "okta",
    Auth0Samlp: "auth0-samlp",
    OktaSamlp: "okta-samlp",
    KeycloakSamlp: "keycloak-samlp",
    Pingfederate: "pingfederate",
};
/** The language of the custom text. */
export const SelfServiceProfileCustomTextLanguageEnum = {
    En: "en",
};
/** The page where the custom text is shown. */
export const SelfServiceProfileCustomTextPageEnum = {
    GetStarted: "get-started",
};
/** Whether the end user should complete the domain verification step. Possible values are 'none' (the step is not shown to the user), 'optional' (the user may add a domain alias in the domain verification step) or 'required' (the user must add a domain alias in order to enable the connection). Defaults to 'none'. */
export const SelfServiceProfileSsoTicketDomainVerificationEnum = {
    None: "none",
    Optional: "optional",
    Required: "required",
};
/** The protocol used to connect to the the default application */
export const SelfServiceProfileSsoTicketIdpInitiatedClientProtocolEnum = {
    Samlp: "samlp",
    Wsfed: "wsfed",
    Oauth2: "oauth2",
};
export const SelfServiceProfileSsoTicketProvisioningScopeEnum = {
    GetUsers: "get:users",
    PostUsers: "post:users",
    PutUsers: "put:users",
    PatchUsers: "patch:users",
    DeleteUsers: "delete:users",
    GetGroups: "get:groups",
    PostGroups: "post:groups",
    PutGroups: "put:groups",
    PatchGroups: "patch:groups",
    DeleteGroups: "delete:groups",
};
/** [Private Early Access] The persistence mode of the session cookie. When set to "non-persistent" (ephemeral), the cookie will be deleted when the browser is closed. When set to "persistent", the cookie will be stored until it expires or is deleted by the user. */
export const SessionCookieMetadataModeEnum = {
    NonPersistent: "non-persistent",
    Persistent: "persistent",
};
/** Behavior of the session cookie */
export const SessionCookieModeEnum = {
    Persistent: "persistent",
    NonPersistent: "non-persistent",
};
/** Algorithm used to sign JWTs. Can be `HS256` (default) or `RS256`. `PS256` available via addon. */
export const SigningAlgorithmEnum = {
    Hs256: "HS256",
    Rs256: "RS256",
    Rs512: "RS512",
    Ps256: "PS256",
};
export const SignupStatusEnum = {
    Required: "required",
    Optional: "optional",
    Inactive: "inactive",
};
export const SupportedLocales = {
    Am: "am",
    Ar: "ar",
    ArEg: "ar-EG",
    ArSa: "ar-SA",
    Az: "az",
    Bg: "bg",
    Bn: "bn",
    Bs: "bs",
    CaEs: "ca-ES",
    Cnr: "cnr",
    Cs: "cs",
    Cy: "cy",
    Da: "da",
    De: "de",
    El: "el",
    En: "en",
    EnCa: "en-CA",
    Es: "es",
    Es419: "es-419",
    EsAr: "es-AR",
    EsMx: "es-MX",
    Et: "et",
    EuEs: "eu-ES",
    Fa: "fa",
    Fi: "fi",
    Fr: "fr",
    FrCa: "fr-CA",
    FrFr: "fr-FR",
    GlEs: "gl-ES",
    Gu: "gu",
    He: "he",
    Hi: "hi",
    Hr: "hr",
    Hu: "hu",
    Hy: "hy",
    Id: "id",
    Is: "is",
    It: "it",
    Ja: "ja",
    Ka: "ka",
    Kk: "kk",
    Kn: "kn",
    Ko: "ko",
    Lt: "lt",
    Lv: "lv",
    Mk: "mk",
    Ml: "ml",
    Mn: "mn",
    Mr: "mr",
    Ms: "ms",
    My: "my",
    Nb: "nb",
    Nl: "nl",
    Nn: "nn",
    No: "no",
    Pa: "pa",
    Pl: "pl",
    Pt: "pt",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    Ro: "ro",
    Ru: "ru",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Sq: "sq",
    Sr: "sr",
    Sv: "sv",
    Sw: "sw",
    Ta: "ta",
    Te: "te",
    Th: "th",
    Tl: "tl",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Zgh: "zgh",
    ZhCn: "zh-CN",
    ZhHk: "zh-HK",
    ZhMo: "zh-MO",
    ZhTw: "zh-TW",
};
export const SuspiciousIpThrottlingShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
/** Group synchronization configuration */
export const SynchronizeGroupsEnum = {
    All: "all",
    Off: "off",
    Selected: "selected",
};
/** Character set used to generate a User Code. Can be `base20` or `digits`. */
export const TenantSettingsDeviceFlowCharset = {
    Base20: "base20",
    Digits: "digits",
};
/** Sets the `third_party_security_mode` assigned to clients created via Dynamic Client Registration. `strict` applies enhanced security controls. `permissive` preserves <a href="https://auth0.com/docs/get-started/applications/third-party-applications/permissive-mode#dynamic-client-registration-in-permissive-mode">pre-existing behavior</a> and is only available to tenants with prior third-party client usage. */
export const TenantSettingsDynamicClientRegistrationSecurityMode = {
    Strict: "strict",
    Permissive: "permissive",
};
/** Profile that determines how the identity of the protected resource (i.e., API) can be specified in the OAuth endpoints when access is being requested. When set to audience (default), the audience parameter is used to specify the resource server. When set to compatibility, the audience parameter is still checked first, but if it not provided, then the resource parameter can be used to specify the resource server. */
export const TenantSettingsResourceParameterProfile = {
    Audience: "audience",
    Compatibility: "compatibility",
};
export const TenantSettingsSupportedLocalesEnum = {
    Am: "am",
    Ar: "ar",
    ArEg: "ar-EG",
    ArSa: "ar-SA",
    Az: "az",
    Bg: "bg",
    Bn: "bn",
    Bs: "bs",
    CaEs: "ca-ES",
    Cnr: "cnr",
    Cs: "cs",
    Cy: "cy",
    Da: "da",
    De: "de",
    El: "el",
    En: "en",
    EnCa: "en-CA",
    Es: "es",
    Es419: "es-419",
    EsAr: "es-AR",
    EsMx: "es-MX",
    Et: "et",
    EuEs: "eu-ES",
    Fa: "fa",
    Fi: "fi",
    Fr: "fr",
    FrCa: "fr-CA",
    FrFr: "fr-FR",
    GlEs: "gl-ES",
    Gu: "gu",
    He: "he",
    Hi: "hi",
    Hr: "hr",
    Hu: "hu",
    Hy: "hy",
    Id: "id",
    Is: "is",
    It: "it",
    Ja: "ja",
    Ka: "ka",
    Kk: "kk",
    Kn: "kn",
    Ko: "ko",
    Lt: "lt",
    Lv: "lv",
    Mk: "mk",
    Ml: "ml",
    Mn: "mn",
    Mr: "mr",
    Ms: "ms",
    My: "my",
    Nb: "nb",
    Nl: "nl",
    Nn: "nn",
    No: "no",
    Pa: "pa",
    Pl: "pl",
    Pt: "pt",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    Ro: "ro",
    Ru: "ru",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Sq: "sq",
    Sr: "sr",
    Sv: "sv",
    Sw: "sw",
    Ta: "ta",
    Te: "te",
    Th: "th",
    Tl: "tl",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Zgh: "zgh",
    ZhCn: "zh-CN",
    ZhHk: "zh-HK",
    ZhMo: "zh-MO",
    ZhTw: "zh-TW",
};
/** The type of the profile, which controls how the profile will be executed when receiving a token exchange request. */
export const TokenExchangeProfileTypeEnum = {
    CustomAuthentication: "custom_authentication",
};
export var TooManyRequestsSchema;
(function (TooManyRequestsSchema) {
    TooManyRequestsSchema.Error_ = {
        TooManyRequests: "Too Many Requests",
    };
})(TooManyRequestsSchema || (TooManyRequestsSchema = {}));
export const TwilioProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
export var UnauthorizedSchema;
(function (UnauthorizedSchema) {
    UnauthorizedSchema.Error_ = {
        Unauthorized: "Unauthorized",
    };
})(UnauthorizedSchema || (UnauthorizedSchema = {}));
/** Which login experience to use. Can be `new` or `classic`. */
export const UniversalLoginExperienceEnum = {
    New: "new",
    Classic: "classic",
};
/** Controls identifier input presentation on the login flow. Defaults to "unified" for legacy tenants, "separate" for tenants created post-GA of this feature. */
export const UpdateBrandingLoginDisplayEnum = {
    Unified: "unified",
    Separate: "separate",
};
/** Controls the format used when displaying phone numbers. */
export const UpdateBrandingPhoneFormattingEnum = {
    Regional: "regional",
    International: "international",
};
/** Controls how phone numbers are masked when displayed back to users. */
export const UpdateBrandingPhoneMaskingEnum = {
    ShowAll: "show_all",
    HideCountryCode: "hide_country_code",
    MaskDigits: "mask_digits",
};
/** OIDC mapping for user ID */
export const UserAttributeProfileUserIdOidcMappingEnum = {
    Sub: "sub",
};
/** OIDC mapping override for this strategy */
export const UserAttributeProfileUserIdOidcStrategyOverrideMapping = {
    Sub: "sub",
    Oid: "oid",
    Email: "email",
};
export const UserAuthenticationMethodPropertiesEnum = {
    Totp: "totp",
    Push: "push",
    Sms: "sms",
    Voice: "voice",
};
/** Authentication method for this enrollment. Can be `authenticator`, `guardian`, `sms`, `webauthn-roaming`, or `webauthn-platform`. */
export const UserEnrollmentAuthMethodEnum = {
    Authenticator: "authenticator",
    Guardian: "guardian",
    Sms: "sms",
    WebauthnPlatform: "webauthn-platform",
    WebauthnRoaming: "webauthn-roaming",
};
/** Status of this enrollment. Can be `pending` or `confirmed`. */
export const UserEnrollmentStatusEnum = {
    Pending: "pending",
    Confirmed: "confirmed",
};
/** The type of identity provider */
export const UserIdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yandex: "yandex",
};
/** The multi-factor provider. Supported values 'duo' or 'google-authenticator' */
export const UserMultifactorProviderEnum = {
    Duo: "duo",
    GoogleAuthenticator: "google-authenticator",
};
export const VerificationMethodEnum = {
    Link: "link",
    Otp: "otp",
};
export const X509CertificateCredentialTypeEnum = {
    X509Cert: "x509_cert",
};
