// auth0/dist/cjs/management/api/resources/attackProtection/client/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttackProtectionClient = void 0;
const BaseClient_js_1 = require("../../../../BaseClient.js");
const Client_js_1 = require("../resources/botDetection/client/Client.js");
const Client_js_2 = require("../resources/breachedPasswordDetection/client/Client.js");
const Client_js_3 = require("../resources/bruteForceProtection/client/Client.js");
const Client_js_4 = require("../resources/captcha/client/Client.js");
const Client_js_5 = require("../resources/suspiciousIpThrottling/client/Client.js");
class AttackProtectionClient {
    constructor(options) {
        this._options = (0, BaseClient_js_1.normalizeClientOptionsWithAuth)(options);
    }
    get botDetection() {
        var _a;
        return ((_a = this._botDetection) !== null && _a !== void 0 ? _a : (this._botDetection = new Client_js_1.BotDetectionClient(this._options)));
    }
    get breachedPasswordDetection() {
        var _a;
        return ((_a = this._breachedPasswordDetection) !== null && _a !== void 0 ? _a : (this._breachedPasswordDetection = new Client_js_2.BreachedPasswordDetectionClient(this._options)));
    }
    get bruteForceProtection() {
        var _a;
        return ((_a = this._bruteForceProtection) !== null && _a !== void 0 ? _a : (this._bruteForceProtection = new Client_js_3.BruteForceProtectionClient(this._options)));
    }
    get captcha() {
        var _a;
        return ((_a = this._captcha) !== null && _a !== void 0 ? _a : (this._captcha = new Client_js_4.CaptchaClient(this._options)));
    }
    get suspiciousIpThrottling() {
        var _a;
        return ((_a = this._suspiciousIpThrottling) !== null && _a !== void 0 ? _a : (this._suspiciousIpThrottling = new Client_js_5.SuspiciousIpThrottlingClient(this._options)));
    }
}
exports.AttackProtectionClient = AttackProtectionClient;
