// auth0/dist/esm/management/api/resources/attackProtection/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
import { normalizeClientOptionsWithAuth } from "../../../../BaseClient.mjs";
import { BotDetectionClient } from "../resources/botDetection/client/Client.mjs";
import { BreachedPasswordDetectionClient } from "../resources/breachedPasswordDetection/client/Client.mjs";
import { BruteForceProtectionClient } from "../resources/bruteForceProtection/client/Client.mjs";
import { CaptchaClient } from "../resources/captcha/client/Client.mjs";
import { SuspiciousIpThrottlingClient } from "../resources/suspiciousIpThrottling/client/Client.mjs";
export class AttackProtectionClient {
    constructor(options) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    get botDetection() {
        var _a;
        return ((_a = this._botDetection) !== null && _a !== void 0 ? _a : (this._botDetection = new BotDetectionClient(this._options)));
    }
    get breachedPasswordDetection() {
        var _a;
        return ((_a = this._breachedPasswordDetection) !== null && _a !== void 0 ? _a : (this._breachedPasswordDetection = new BreachedPasswordDetectionClient(this._options)));
    }
    get bruteForceProtection() {
        var _a;
        return ((_a = this._bruteForceProtection) !== null && _a !== void 0 ? _a : (this._bruteForceProtection = new BruteForceProtectionClient(this._options)));
    }
    get captcha() {
        var _a;
        return ((_a = this._captcha) !== null && _a !== void 0 ? _a : (this._captcha = new CaptchaClient(this._options)));
    }
    get suspiciousIpThrottling() {
        var _a;
        return ((_a = this._suspiciousIpThrottling) !== null && _a !== void 0 ? _a : (this._suspiciousIpThrottling = new SuspiciousIpThrottlingClient(this._options)));
    }
}
