// auth0/dist/cjs/management/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifiableCredentials = exports.users = exports.userGrants = exports.userBlocks = exports.userAttributeProfiles = exports.tokenExchangeProfiles = exports.tickets = exports.tenants = exports.supplementalSignals = exports.stats = exports.sessions = exports.selfServiceProfiles = exports.rulesConfigs = exports.rules = exports.roles = exports.riskAssessments = exports.resourceServers = exports.refreshTokens = exports.prompts = exports.organizations = exports.networkAcls = exports.logStreams = exports.logs = exports.keys = exports.jobs = exports.hooks = exports.guardian = exports.groups = exports.forms = exports.flows = exports.eventStreams = exports.events = exports.emailTemplates = exports.emails = exports.deviceCredentials = exports.customDomains = exports.connections = exports.connectionProfiles = exports.clients = exports.clientGrants = exports.branding = exports.attackProtection = exports.anomaly = exports.actions = void 0;
exports.actions = __importStar(require("./actions/index.js"));
exports.anomaly = __importStar(require("./anomaly/index.js"));
exports.attackProtection = __importStar(require("./attackProtection/index.js"));
exports.branding = __importStar(require("./branding/index.js"));
exports.clientGrants = __importStar(require("./clientGrants/index.js"));
exports.clients = __importStar(require("./clients/index.js"));
exports.connectionProfiles = __importStar(require("./connectionProfiles/index.js"));
exports.connections = __importStar(require("./connections/index.js"));
exports.customDomains = __importStar(require("./customDomains/index.js"));
exports.deviceCredentials = __importStar(require("./deviceCredentials/index.js"));
exports.emails = __importStar(require("./emails/index.js"));
exports.emailTemplates = __importStar(require("./emailTemplates/index.js"));
exports.events = __importStar(require("./events/index.js"));
exports.eventStreams = __importStar(require("./eventStreams/index.js"));
__exportStar(require("./eventStreams/types/index.js"), exports);
exports.flows = __importStar(require("./flows/index.js"));
exports.forms = __importStar(require("./forms/index.js"));
exports.groups = __importStar(require("./groups/index.js"));
exports.guardian = __importStar(require("./guardian/index.js"));
exports.hooks = __importStar(require("./hooks/index.js"));
exports.jobs = __importStar(require("./jobs/index.js"));
exports.keys = __importStar(require("./keys/index.js"));
exports.logs = __importStar(require("./logs/index.js"));
exports.logStreams = __importStar(require("./logStreams/index.js"));
exports.networkAcls = __importStar(require("./networkAcls/index.js"));
exports.organizations = __importStar(require("./organizations/index.js"));
exports.prompts = __importStar(require("./prompts/index.js"));
exports.refreshTokens = __importStar(require("./refreshTokens/index.js"));
exports.resourceServers = __importStar(require("./resourceServers/index.js"));
exports.riskAssessments = __importStar(require("./riskAssessments/index.js"));
exports.roles = __importStar(require("./roles/index.js"));
exports.rules = __importStar(require("./rules/index.js"));
exports.rulesConfigs = __importStar(require("./rulesConfigs/index.js"));
exports.selfServiceProfiles = __importStar(require("./selfServiceProfiles/index.js"));
exports.sessions = __importStar(require("./sessions/index.js"));
exports.stats = __importStar(require("./stats/index.js"));
exports.supplementalSignals = __importStar(require("./supplementalSignals/index.js"));
exports.tenants = __importStar(require("./tenants/index.js"));
exports.tickets = __importStar(require("./tickets/index.js"));
exports.tokenExchangeProfiles = __importStar(require("./tokenExchangeProfiles/index.js"));
exports.userAttributeProfiles = __importStar(require("./userAttributeProfiles/index.js"));
exports.userBlocks = __importStar(require("./userBlocks/index.js"));
exports.userGrants = __importStar(require("./userGrants/index.js"));
exports.users = __importStar(require("./users/index.js"));
exports.verifiableCredentials = __importStar(require("./verifiableCredentials/index.js"));
