// auth0/dist/cjs/management/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.withAbortSignal = exports.withHeaders = exports.withRetries = exports.withTimeout = exports.CustomDomainHeader = exports.ManagementEnvironment = exports.ManagementClient = exports.ManagementTimeoutError = exports.ManagementError = exports.Management = void 0;
exports.Management = __importStar(require("./api/index.js"));
var index_js_1 = require("./errors/index.js");
Object.defineProperty(exports, "ManagementError", { enumerable: true, get: function () { return index_js_1.ManagementError; } });
Object.defineProperty(exports, "ManagementTimeoutError", { enumerable: true, get: function () { return index_js_1.ManagementTimeoutError; } });
var ManagementClient_js_1 = require("./wrapper/ManagementClient.js");
Object.defineProperty(exports, "ManagementClient", { enumerable: true, get: function () { return ManagementClient_js_1.ManagementClient; } });
var environments_js_1 = require("./environments.js");
Object.defineProperty(exports, "ManagementEnvironment", { enumerable: true, get: function () { return environments_js_1.ManagementEnvironment; } });
var request_options_js_1 = require("./request-options.js");
Object.defineProperty(exports, "CustomDomainHeader", { enumerable: true, get: function () { return request_options_js_1.CustomDomainHeader; } });
Object.defineProperty(exports, "withTimeout", { enumerable: true, get: function () { return request_options_js_1.withTimeout; } });
Object.defineProperty(exports, "withRetries", { enumerable: true, get: function () { return request_options_js_1.withRetries; } });
Object.defineProperty(exports, "withHeaders", { enumerable: true, get: function () { return request_options_js_1.withHeaders; } });
Object.defineProperty(exports, "withAbortSignal", { enumerable: true, get: function () { return request_options_js_1.withAbortSignal; } });
__exportStar(require("./exports.js"), exports);
