// auth0/dist/esm/management/Client.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { ActionsClient } from "./api/resources/actions/client/Client.mjs";
import { AnomalyClient } from "./api/resources/anomaly/client/Client.mjs";
import { AttackProtectionClient } from "./api/resources/attackProtection/client/Client.mjs";
import { BrandingClient } from "./api/resources/branding/client/Client.mjs";
import { ClientGrantsClient } from "./api/resources/clientGrants/client/Client.mjs";
import { ClientsClient } from "./api/resources/clients/client/Client.mjs";
import { ConnectionProfilesClient } from "./api/resources/connectionProfiles/client/Client.mjs";
import { ConnectionsClient } from "./api/resources/connections/client/Client.mjs";
import { CustomDomainsClient } from "./api/resources/customDomains/client/Client.mjs";
import { DeviceCredentialsClient } from "./api/resources/deviceCredentials/client/Client.mjs";
import { EmailsClient } from "./api/resources/emails/client/Client.mjs";
import { EmailTemplatesClient } from "./api/resources/emailTemplates/client/Client.mjs";
import { EventsClient } from "./api/resources/events/client/Client.mjs";
import { EventStreamsClient } from "./api/resources/eventStreams/client/Client.mjs";
import { FlowsClient } from "./api/resources/flows/client/Client.mjs";
import { FormsClient } from "./api/resources/forms/client/Client.mjs";
import { GroupsClient } from "./api/resources/groups/client/Client.mjs";
import { GuardianClient } from "./api/resources/guardian/client/Client.mjs";
import { HooksClient } from "./api/resources/hooks/client/Client.mjs";
import { JobsClient } from "./api/resources/jobs/client/Client.mjs";
import { KeysClient } from "./api/resources/keys/client/Client.mjs";
import { LogsClient } from "./api/resources/logs/client/Client.mjs";
import { LogStreamsClient } from "./api/resources/logStreams/client/Client.mjs";
import { NetworkAclsClient } from "./api/resources/networkAcls/client/Client.mjs";
import { OrganizationsClient } from "./api/resources/organizations/client/Client.mjs";
import { PromptsClient } from "./api/resources/prompts/client/Client.mjs";
import { RefreshTokensClient } from "./api/resources/refreshTokens/client/Client.mjs";
import { ResourceServersClient } from "./api/resources/resourceServers/client/Client.mjs";
import { RiskAssessmentsClient } from "./api/resources/riskAssessments/client/Client.mjs";
import { RolesClient } from "./api/resources/roles/client/Client.mjs";
import { RulesClient } from "./api/resources/rules/client/Client.mjs";
import { RulesConfigsClient } from "./api/resources/rulesConfigs/client/Client.mjs";
import { SelfServiceProfilesClient } from "./api/resources/selfServiceProfiles/client/Client.mjs";
import { SessionsClient } from "./api/resources/sessions/client/Client.mjs";
import { StatsClient } from "./api/resources/stats/client/Client.mjs";
import { SupplementalSignalsClient } from "./api/resources/supplementalSignals/client/Client.mjs";
import { TenantsClient } from "./api/resources/tenants/client/Client.mjs";
import { TicketsClient } from "./api/resources/tickets/client/Client.mjs";
import { TokenExchangeProfilesClient } from "./api/resources/tokenExchangeProfiles/client/Client.mjs";
import { UserAttributeProfilesClient } from "./api/resources/userAttributeProfiles/client/Client.mjs";
import { UserBlocksClient } from "./api/resources/userBlocks/client/Client.mjs";
import { UserGrantsClient } from "./api/resources/userGrants/client/Client.mjs";
import { UsersClient } from "./api/resources/users/client/Client.mjs";
import { VerifiableCredentialsClient } from "./api/resources/verifiableCredentials/client/Client.mjs";
import { normalizeClientOptionsWithAuth } from "./BaseClient.mjs";
import * as core from "./core/index.mjs";
export class ManagementClient {
    constructor(options) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    get actions() {
        var _a;
        return ((_a = this._actions) !== null && _a !== void 0 ? _a : (this._actions = new ActionsClient(this._options)));
    }
    get branding() {
        var _a;
        return ((_a = this._branding) !== null && _a !== void 0 ? _a : (this._branding = new BrandingClient(this._options)));
    }
    get clientGrants() {
        var _a;
        return ((_a = this._clientGrants) !== null && _a !== void 0 ? _a : (this._clientGrants = new ClientGrantsClient(this._options)));
    }
    get clients() {
        var _a;
        return ((_a = this._clients) !== null && _a !== void 0 ? _a : (this._clients = new ClientsClient(this._options)));
    }
    get connectionProfiles() {
        var _a;
        return ((_a = this._connectionProfiles) !== null && _a !== void 0 ? _a : (this._connectionProfiles = new ConnectionProfilesClient(this._options)));
    }
    get connections() {
        var _a;
        return ((_a = this._connections) !== null && _a !== void 0 ? _a : (this._connections = new ConnectionsClient(this._options)));
    }
    get customDomains() {
        var _a;
        return ((_a = this._customDomains) !== null && _a !== void 0 ? _a : (this._customDomains = new CustomDomainsClient(this._options)));
    }
    get deviceCredentials() {
        var _a;
        return ((_a = this._deviceCredentials) !== null && _a !== void 0 ? _a : (this._deviceCredentials = new DeviceCredentialsClient(this._options)));
    }
    get emailTemplates() {
        var _a;
        return ((_a = this._emailTemplates) !== null && _a !== void 0 ? _a : (this._emailTemplates = new EmailTemplatesClient(this._options)));
    }
    get eventStreams() {
        var _a;
        return ((_a = this._eventStreams) !== null && _a !== void 0 ? _a : (this._eventStreams = new EventStreamsClient(this._options)));
    }
    get events() {
        var _a;
        return ((_a = this._events) !== null && _a !== void 0 ? _a : (this._events = new EventsClient(this._options)));
    }
    get flows() {
        var _a;
        return ((_a = this._flows) !== null && _a !== void 0 ? _a : (this._flows = new FlowsClient(this._options)));
    }
    get forms() {
        var _a;
        return ((_a = this._forms) !== null && _a !== void 0 ? _a : (this._forms = new FormsClient(this._options)));
    }
    get userGrants() {
        var _a;
        return ((_a = this._userGrants) !== null && _a !== void 0 ? _a : (this._userGrants = new UserGrantsClient(this._options)));
    }
    get groups() {
        var _a;
        return ((_a = this._groups) !== null && _a !== void 0 ? _a : (this._groups = new GroupsClient(this._options)));
    }
    get hooks() {
        var _a;
        return ((_a = this._hooks) !== null && _a !== void 0 ? _a : (this._hooks = new HooksClient(this._options)));
    }
    get jobs() {
        var _a;
        return ((_a = this._jobs) !== null && _a !== void 0 ? _a : (this._jobs = new JobsClient(this._options)));
    }
    get logStreams() {
        var _a;
        return ((_a = this._logStreams) !== null && _a !== void 0 ? _a : (this._logStreams = new LogStreamsClient(this._options)));
    }
    get logs() {
        var _a;
        return ((_a = this._logs) !== null && _a !== void 0 ? _a : (this._logs = new LogsClient(this._options)));
    }
    get networkAcls() {
        var _a;
        return ((_a = this._networkAcls) !== null && _a !== void 0 ? _a : (this._networkAcls = new NetworkAclsClient(this._options)));
    }
    get organizations() {
        var _a;
        return ((_a = this._organizations) !== null && _a !== void 0 ? _a : (this._organizations = new OrganizationsClient(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new PromptsClient(this._options)));
    }
    get refreshTokens() {
        var _a;
        return ((_a = this._refreshTokens) !== null && _a !== void 0 ? _a : (this._refreshTokens = new RefreshTokensClient(this._options)));
    }
    get resourceServers() {
        var _a;
        return ((_a = this._resourceServers) !== null && _a !== void 0 ? _a : (this._resourceServers = new ResourceServersClient(this._options)));
    }
    get roles() {
        var _a;
        return ((_a = this._roles) !== null && _a !== void 0 ? _a : (this._roles = new RolesClient(this._options)));
    }
    get rules() {
        var _a;
        return ((_a = this._rules) !== null && _a !== void 0 ? _a : (this._rules = new RulesClient(this._options)));
    }
    get rulesConfigs() {
        var _a;
        return ((_a = this._rulesConfigs) !== null && _a !== void 0 ? _a : (this._rulesConfigs = new RulesConfigsClient(this._options)));
    }
    get selfServiceProfiles() {
        var _a;
        return ((_a = this._selfServiceProfiles) !== null && _a !== void 0 ? _a : (this._selfServiceProfiles = new SelfServiceProfilesClient(this._options)));
    }
    get sessions() {
        var _a;
        return ((_a = this._sessions) !== null && _a !== void 0 ? _a : (this._sessions = new SessionsClient(this._options)));
    }
    get stats() {
        var _a;
        return ((_a = this._stats) !== null && _a !== void 0 ? _a : (this._stats = new StatsClient(this._options)));
    }
    get supplementalSignals() {
        var _a;
        return ((_a = this._supplementalSignals) !== null && _a !== void 0 ? _a : (this._supplementalSignals = new SupplementalSignalsClient(this._options)));
    }
    get tickets() {
        var _a;
        return ((_a = this._tickets) !== null && _a !== void 0 ? _a : (this._tickets = new TicketsClient(this._options)));
    }
    get tokenExchangeProfiles() {
        var _a;
        return ((_a = this._tokenExchangeProfiles) !== null && _a !== void 0 ? _a : (this._tokenExchangeProfiles = new TokenExchangeProfilesClient(this._options)));
    }
    get userAttributeProfiles() {
        var _a;
        return ((_a = this._userAttributeProfiles) !== null && _a !== void 0 ? _a : (this._userAttributeProfiles = new UserAttributeProfilesClient(this._options)));
    }
    get userBlocks() {
        var _a;
        return ((_a = this._userBlocks) !== null && _a !== void 0 ? _a : (this._userBlocks = new UserBlocksClient(this._options)));
    }
    get users() {
        var _a;
        return ((_a = this._users) !== null && _a !== void 0 ? _a : (this._users = new UsersClient(this._options)));
    }
    get anomaly() {
        var _a;
        return ((_a = this._anomaly) !== null && _a !== void 0 ? _a : (this._anomaly = new AnomalyClient(this._options)));
    }
    get attackProtection() {
        var _a;
        return ((_a = this._attackProtection) !== null && _a !== void 0 ? _a : (this._attackProtection = new AttackProtectionClient(this._options)));
    }
    get emails() {
        var _a;
        return ((_a = this._emails) !== null && _a !== void 0 ? _a : (this._emails = new EmailsClient(this._options)));
    }
    get guardian() {
        var _a;
        return ((_a = this._guardian) !== null && _a !== void 0 ? _a : (this._guardian = new GuardianClient(this._options)));
    }
    get keys() {
        var _a;
        return ((_a = this._keys) !== null && _a !== void 0 ? _a : (this._keys = new KeysClient(this._options)));
    }
    get riskAssessments() {
        var _a;
        return ((_a = this._riskAssessments) !== null && _a !== void 0 ? _a : (this._riskAssessments = new RiskAssessmentsClient(this._options)));
    }
    get tenants() {
        var _a;
        return ((_a = this._tenants) !== null && _a !== void 0 ? _a : (this._tenants = new TenantsClient(this._options)));
    }
    get verifiableCredentials() {
        var _a;
        return ((_a = this._verifiableCredentials) !== null && _a !== void 0 ? _a : (this._verifiableCredentials = new VerifiableCredentialsClient(this._options)));
    }
    /**
     * Make a passthrough request using the SDK's configured auth, retry, logging, etc.
     * This is useful for making requests to endpoints not yet supported in the SDK.
     * The input can be a URL string, URL object, or Request object. Relative paths are resolved against the configured base URL.
     *
     * @param {Request | string | URL} input - The URL, path, or Request object.
     * @param {RequestInit} init - Standard fetch RequestInit options.
     * @param {core.PassthroughRequest.RequestOptions} requestOptions - Per-request overrides (timeout, retries, headers, abort signal).
     * @returns {Promise<Response>} A standard Response object.
     */
    fetch(input, init, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a;
            return core.makePassthroughRequest(input, init, {
                baseUrl: (_a = this._options.baseUrl) !== null && _a !== void 0 ? _a : this._options.environment,
                headers: this._options.headers,
                timeoutInSeconds: this._options.timeoutInSeconds,
                maxRetries: this._options.maxRetries,
                fetch: this._options.fetch,
                logging: this._options.logging,
                getAuthHeaders: () => __awaiter(this, void 0, void 0, function* () { return (yield this._options.authProvider.getAuthRequest()).headers; }),
            }, requestOptions);
        });
    }
}
