// auth0/dist/esm/management/api/resources/userBlocks/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { normalizeClientOptionsWithAuth } from "../../../../BaseClient.mjs";
import * as core from "../../../../core/index.mjs";
import { mergeHeaders } from "../../../../core/headers.mjs";
import * as environments from "../../../../environments.mjs";
import { handleNonStatusCodeError } from "../../../../errors/handleNonStatusCodeError.mjs";
import * as errors from "../../../../errors/index.mjs";
import * as Management from "../../../index.mjs";
export class UserBlocksClient {
    constructor(options) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    /**
     * Retrieve details of all <a href="https://auth0.com/docs/secure/attack-protection/brute-force-protection">Brute-force Protection</a> blocks for a user with the given identifier (username, phone number, or email).
     *
     * @param {Management.ListUserBlocksByIdentifierRequestParameters} request
     * @param {UserBlocksClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.userBlocks.listByIdentifier({
     *         identifier: "identifier",
     *         consider_brute_force_enablement: true
     *     })
     */
    listByIdentifier(request, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__listByIdentifier(request, requestOptions));
    }
    __listByIdentifier(request, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const { identifier, consider_brute_force_enablement: considerBruteForceEnablement } = request;
            const _queryParams = {
                identifier,
                consider_brute_force_enablement: considerBruteForceEnablement,
            };
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, "user-blocks"),
                method: "GET",
                headers: _headers,
                queryString: core.url
                    .queryBuilder()
                    .addMany(_queryParams)
                    .mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams)
                    .build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return {
                    data: _response.body,
                    rawResponse: _response.rawResponse,
                };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "GET", "/user-blocks");
        });
    }
    /**
     * Remove all <a href="https://auth0.com/docs/secure/attack-protection/brute-force-protection">Brute-force Protection</a> blocks for the user with the given identifier (username, phone number, or email).
     *
     * Note: This endpoint does not unblock users that were <a href="https://auth0.com/docs/user-profile#block-and-unblock-a-user">blocked by a tenant administrator</a>.
     *
     * @param {Management.DeleteUserBlocksByIdentifierRequestParameters} request
     * @param {UserBlocksClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.userBlocks.deleteByIdentifier({
     *         identifier: "identifier"
     *     })
     */
    deleteByIdentifier(request, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__deleteByIdentifier(request, requestOptions));
    }
    __deleteByIdentifier(request, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const { identifier } = request;
            const _queryParams = {
                identifier,
            };
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, "user-blocks"),
                method: "DELETE",
                headers: _headers,
                queryString: core.url
                    .queryBuilder()
                    .addMany(_queryParams)
                    .mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams)
                    .build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return { data: undefined, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "DELETE", "/user-blocks");
        });
    }
    /**
     * Retrieve details of all <a href="https://auth0.com/docs/secure/attack-protection/brute-force-protection">Brute-force Protection</a> blocks for the user with the given ID.
     *
     * @param {string} id - user_id of the user blocks to retrieve.
     * @param {Management.ListUserBlocksRequestParameters} request
     * @param {UserBlocksClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.NotFoundError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.userBlocks.list("id", {
     *         consider_brute_force_enablement: true
     *     })
     */
    list(id, request = {}, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__list(id, request, requestOptions));
    }
    __list(id_1) {
        return __awaiter(this, arguments, void 0, function* (id, request = {}, requestOptions) {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const { consider_brute_force_enablement: considerBruteForceEnablement } = request;
            const _queryParams = {
                consider_brute_force_enablement: considerBruteForceEnablement,
            };
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, `user-blocks/${core.url.encodePathParam(id)}`),
                method: "GET",
                headers: _headers,
                queryString: core.url
                    .queryBuilder()
                    .addMany(_queryParams)
                    .mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams)
                    .build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return {
                    data: _response.body,
                    rawResponse: _response.rawResponse,
                };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 404:
                        throw new Management.NotFoundError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "GET", "/user-blocks/{id}");
        });
    }
    /**
     * Remove all <a href="https://auth0.com/docs/secure/attack-protection/brute-force-protection">Brute-force Protection</a> blocks for the user with the given ID.
     *
     * Note: This endpoint does not unblock users that were <a href="https://auth0.com/docs/user-profile#block-and-unblock-a-user">blocked by a tenant administrator</a>.
     *
     * @param {string} id - The user_id of the user to update.
     * @param {UserBlocksClient.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @throws {@link Management.BadRequestError}
     * @throws {@link Management.UnauthorizedError}
     * @throws {@link Management.ForbiddenError}
     * @throws {@link Management.NotFoundError}
     * @throws {@link Management.TooManyRequestsError}
     *
     * @example
     *     await client.userBlocks.delete("id")
     */
    delete(id, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__delete(id, requestOptions));
    }
    __delete(id, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            let _headers = mergeHeaders(_authRequest.headers, (_a = this._options) === null || _a === void 0 ? void 0 : _a.headers, requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers);
            const _response = yield ((_b = this._options.fetcher) !== null && _b !== void 0 ? _b : core.fetcher)({
                url: core.url.join((_d = (_c = (yield core.Supplier.get(this._options.baseUrl))) !== null && _c !== void 0 ? _c : (yield core.Supplier.get(this._options.environment))) !== null && _d !== void 0 ? _d : environments.ManagementEnvironment.Default, `user-blocks/${core.url.encodePathParam(id)}`),
                method: "DELETE",
                headers: _headers,
                queryString: core.url.queryBuilder().mergeAdditional(requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.queryParams).build(),
                timeoutMs: ((_g = (_e = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) !== null && _e !== void 0 ? _e : (_f = this._options) === null || _f === void 0 ? void 0 : _f.timeoutInSeconds) !== null && _g !== void 0 ? _g : 60) * 1000,
                maxRetries: (_h = requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries) !== null && _h !== void 0 ? _h : (_j = this._options) === null || _j === void 0 ? void 0 : _j.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
                fetchFn: (_k = this._options) === null || _k === void 0 ? void 0 : _k.fetch,
                logging: this._options.logging,
            });
            if (_response.ok) {
                return { data: undefined, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                switch (_response.error.statusCode) {
                    case 400:
                        throw new Management.BadRequestError(_response.error.body, _response.rawResponse);
                    case 401:
                        throw new Management.UnauthorizedError(_response.error.body, _response.rawResponse);
                    case 403:
                        throw new Management.ForbiddenError(_response.error.body, _response.rawResponse);
                    case 404:
                        throw new Management.NotFoundError(_response.error.body, _response.rawResponse);
                    case 429:
                        throw new Management.TooManyRequestsError(_response.error.body, _response.rawResponse);
                    default:
                        throw new errors.ManagementError({
                            statusCode: _response.error.statusCode,
                            body: _response.error.body,
                            rawResponse: _response.rawResponse,
                        });
                }
            }
            return handleNonStatusCodeError(_response.error, _response.rawResponse, "DELETE", "/user-blocks/{id}");
        });
    }
}
