// auth0/dist/cjs/management/api/resources/users/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.sessions = exports.roles = exports.riskAssessments = exports.refreshToken = exports.permissions = exports.organizations = exports.multifactor = exports.logs = exports.identities = exports.groups = exports.federatedConnectionsTokensets = exports.enrollments = exports.connectedAccounts = exports.authenticators = exports.authenticationMethods = void 0;
exports.authenticationMethods = __importStar(require("./authenticationMethods/index.js"));
exports.authenticators = __importStar(require("./authenticators/index.js"));
exports.connectedAccounts = __importStar(require("./connectedAccounts/index.js"));
exports.enrollments = __importStar(require("./enrollments/index.js"));
exports.federatedConnectionsTokensets = __importStar(require("./federatedConnectionsTokensets/index.js"));
exports.groups = __importStar(require("./groups/index.js"));
exports.identities = __importStar(require("./identities/index.js"));
exports.logs = __importStar(require("./logs/index.js"));
exports.multifactor = __importStar(require("./multifactor/index.js"));
exports.organizations = __importStar(require("./organizations/index.js"));
exports.permissions = __importStar(require("./permissions/index.js"));
exports.refreshToken = __importStar(require("./refreshToken/index.js"));
exports.riskAssessments = __importStar(require("./riskAssessments/index.js"));
exports.roles = __importStar(require("./roles/index.js"));
exports.sessions = __importStar(require("./sessions/index.js"));
